{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "processing raw text: html 태그 제거 외",
      "provenance": [],
      "authorship_tag": "ABX9TyMg4x2nsdYgdr+/Z1v5t8Y6",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Seong-JiHyeon/NLTK/blob/master/processing_raw_text_html_tag_remove_etc.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "HIm-vPbzJ9Ei",
        "colab_type": "code",
        "outputId": "d8241ee9-9a00-4f70-b265-6393651c46ab",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 131
        }
      },
      "source": [
        "from google.colab import drive\n",
        "\n",
        "drive.mount('/content/gdrive')"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly\n",
            "\n",
            "Enter your authorization code:\n",
            "··········\n",
            "Mounted at /content/gdrive\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "PbHNr7oeKd7L",
        "colab_type": "code",
        "outputId": "72e14ae0-8131-4a4f-c164-3311d80f760c",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "cd /content/gdrive/My Drive/인공지능 수업 데이터 파일"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/content/gdrive/My Drive/인공지능 수업 데이터 파일\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "OvR_dAtobbEq",
        "colab_type": "text"
      },
      "source": [
        "**Corpus화 전의 상태**\n",
        "\n",
        "* Data preprocessing\n",
        "* Regular expression\n",
        "\n",
        ">(token = word)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9nYkdO9EcOR3",
        "colab_type": "text"
      },
      "source": [
        "* Lemmatization: 단어의 원형\n",
        "* steeming: 어근\n",
        "* part of speech tagging: multiple meaning\n",
        "* parsing\n",
        "* sentence breaking\n",
        "* word segmentation: 단어와 단어 사이 분리"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "vrokVwsOKrGT",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "from urllib import * #import request (더 성능이 뛰어남)\n",
        "url = 'http://www.gutenberg.org/files/2554/2554-h/2554-h.htm'\n",
        "response = request.urlopen(url)\n",
        "raw = response.read().decode('utf8')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "InjT2wc8dqCs",
        "colab_type": "code",
        "outputId": "2bba4d8b-06b6-4961-f5ef-8d56d08a1d44",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "type(raw)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "str"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9Y3JIc7Ydrpx",
        "colab_type": "code",
        "outputId": "8f9c088d-ec57-46fc-dd2e-1b1f906f60bd",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "raw[:50] #tag가 섞여 있음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'<?xml version=\"1.0\" encoding=\"utf-8\"?>\\r\\n\\r\\n<!DOCTYP'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 7
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "oZjT6V85dviy",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "url2 = 'http://www.gutenberg.org/files/2554/2554-0.txt'\n",
        "response2 = request.urlopen(url2)\n",
        "raw2 = response2.read().decode('utf8')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "NbKhipl6eDHL",
        "colab_type": "code",
        "outputId": "0c696ee3-7b15-4ecb-c2da-f326d57a1b08",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(raw2)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1176967"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "0P-HsEEdeE_J",
        "colab_type": "code",
        "outputId": "3fe56813-66c4-4983-fb40-bd5907020a8d",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "raw2[:50] #tag 없음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'\\ufeffThe Project Gutenberg EBook of Crime and Punishme'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "RORhg4aNeJrr",
        "colab_type": "code",
        "outputId": "d79cbbba-e565-42b7-f0db-42ffff94b877",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "raw.rfind(\"PART 1\") #뒤에서부터 찾기"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "-1"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Zlp6D3Nyeo3B",
        "colab_type": "code",
        "outputId": "757d15c3-cc4a-462b-aedf-cc2ad7a8f7f5",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "raw.find(\"The Project Gutenberg EBook of Crime and Punishme\")"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1394"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "vxtMtiyee_k8",
        "colab_type": "code",
        "outputId": "12ae85d9-b2d4-4331-daf4-f9014a52877d",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 345
        }
      },
      "source": [
        "import nltk\n",
        "nltk.clean_html(raw) #exception 처리됨 To remove HTML markup, use BeautifulSoup's get_text() function"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "error",
          "ename": "NotImplementedError",
          "evalue": "ignored",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mNotImplementedError\u001b[0m                       Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-18-677108f1fd7b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mnltk\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mnltk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclean_html\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mraw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
            "\u001b[0;32m/usr/local/lib/python3.6/dist-packages/nltk/util.py\u001b[0m in \u001b[0;36mclean_html\u001b[0;34m(html)\u001b[0m\n\u001b[1;32m    354\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    355\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mclean_html\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhtml\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 356\u001b[0;31m     \u001b[0;32mraise\u001b[0m \u001b[0mNotImplementedError\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0;34m\"To remove HTML markup, use BeautifulSoup's get_text() function\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    357\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    358\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mclean_url\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mNotImplementedError\u001b[0m: To remove HTML markup, use BeautifulSoup's get_text() function"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-ocMJc7PgAXT",
        "colab_type": "code",
        "outputId": "8e96fe33-6d1d-4193-ee51-aa17310798ef",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 56
        }
      },
      "source": [
        "!pip install beautifulsoup4"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.6/dist-packages (4.6.3)\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "P7iSmtiwgKty",
        "colab_type": "code",
        "outputId": "66def16c-34df-4f3b-8581-d736d135bc83",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 56
        }
      },
      "source": [
        "from bs4 import BeautifulSoup\n",
        "url = 'http://www.gutenberg.org/files/2554/2554-h/2554-h.htm'\n",
        "html = request.urlopen(url).read().decode('utf8')\n",
        "html[:100]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'<?xml version=\"1.0\" encoding=\"utf-8\"?>\\r\\n\\r\\n<!DOCTYPE html\\r\\n   PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//E'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bxELIPZAglCR",
        "colab_type": "code",
        "outputId": "b1f0206c-b985-4c4a-e433-8a7e3ea72222",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 56
        }
      },
      "source": [
        "raw = BeautifulSoup(html, 'html.parser').get_text()\n",
        "raw[:100]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'\\n\\n\\n\\n\\r\\n      Crime and Punishment, by Fyodor Dostoevsky\\r\\n    \\n\\r\\n\\r\\n    body { margin:5%; background:#f'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "g3t4xjOAlw8p",
        "colab_type": "text"
      },
      "source": [
        "\n",
        "\n",
        "*   token 단위 분리\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "jk3gVnIDl_Je",
        "colab_type": "code",
        "outputId": "bd56d050-0722-412b-c47e-7744506b7c3c",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "import nltk\n",
        "nltk.download()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "NLTK Downloader\n",
            "---------------------------------------------------------------------------\n",
            "    d) Download   l) List    u) Update   c) Config   h) Help   q) Quit\n",
            "---------------------------------------------------------------------------\n",
            "Downloader> d\n",
            "\n",
            "Download which package (l=list; x=cancel)?\n",
            "  Identifier> all\n",
            "    Downloading collection 'all'\n",
            "       | \n",
            "       | Downloading package abc to /root/nltk_data...\n",
            "       |   Unzipping corpora/abc.zip.\n",
            "       | Downloading package alpino to /root/nltk_data...\n",
            "       |   Unzipping corpora/alpino.zip.\n",
            "       | Downloading package biocreative_ppi to /root/nltk_data...\n",
            "       |   Unzipping corpora/biocreative_ppi.zip.\n",
            "       | Downloading package brown to /root/nltk_data...\n",
            "       |   Unzipping corpora/brown.zip.\n",
            "       | Downloading package brown_tei to /root/nltk_data...\n",
            "       |   Unzipping corpora/brown_tei.zip.\n",
            "       | Downloading package cess_cat to /root/nltk_data...\n",
            "       |   Unzipping corpora/cess_cat.zip.\n",
            "       | Downloading package cess_esp to /root/nltk_data...\n",
            "       |   Unzipping corpora/cess_esp.zip.\n",
            "       | Downloading package chat80 to /root/nltk_data...\n",
            "       |   Unzipping corpora/chat80.zip.\n",
            "       | Downloading package city_database to /root/nltk_data...\n",
            "       |   Unzipping corpora/city_database.zip.\n",
            "       | Downloading package cmudict to /root/nltk_data...\n",
            "       |   Unzipping corpora/cmudict.zip.\n",
            "       | Downloading package comparative_sentences to\n",
            "       |     /root/nltk_data...\n",
            "       |   Unzipping corpora/comparative_sentences.zip.\n",
            "       | Downloading package comtrans to /root/nltk_data...\n",
            "       | Downloading package conll2000 to /root/nltk_data...\n",
            "       |   Unzipping corpora/conll2000.zip.\n",
            "       | Downloading package conll2002 to /root/nltk_data...\n",
            "       |   Unzipping corpora/conll2002.zip.\n",
            "       | Downloading package conll2007 to /root/nltk_data...\n",
            "       | Downloading package crubadan to /root/nltk_data...\n",
            "       |   Unzipping corpora/crubadan.zip.\n",
            "       | Downloading package dependency_treebank to /root/nltk_data...\n",
            "       |   Unzipping corpora/dependency_treebank.zip.\n",
            "       | Downloading package dolch to /root/nltk_data...\n",
            "       |   Unzipping corpora/dolch.zip.\n",
            "       | Downloading package europarl_raw to /root/nltk_data...\n",
            "       |   Unzipping corpora/europarl_raw.zip.\n",
            "       | Downloading package floresta to /root/nltk_data...\n",
            "       |   Unzipping corpora/floresta.zip.\n",
            "       | Downloading package framenet_v15 to /root/nltk_data...\n",
            "       |   Unzipping corpora/framenet_v15.zip.\n",
            "       | Downloading package framenet_v17 to /root/nltk_data...\n",
            "       |   Unzipping corpora/framenet_v17.zip.\n",
            "       | Downloading package gazetteers to /root/nltk_data...\n",
            "       |   Unzipping corpora/gazetteers.zip.\n",
            "       | Downloading package genesis to /root/nltk_data...\n",
            "       |   Unzipping corpora/genesis.zip.\n",
            "       | Downloading package gutenberg to /root/nltk_data...\n",
            "       |   Unzipping corpora/gutenberg.zip.\n",
            "       | Downloading package ieer to /root/nltk_data...\n",
            "       |   Unzipping corpora/ieer.zip.\n",
            "       | Downloading package inaugural to /root/nltk_data...\n",
            "       |   Unzipping corpora/inaugural.zip.\n",
            "       | Downloading package indian to /root/nltk_data...\n",
            "       |   Unzipping corpora/indian.zip.\n",
            "       | Downloading package jeita to /root/nltk_data...\n",
            "       | Downloading package kimmo to /root/nltk_data...\n",
            "       |   Unzipping corpora/kimmo.zip.\n",
            "       | Downloading package knbc to /root/nltk_data...\n",
            "       | Downloading package lin_thesaurus to /root/nltk_data...\n",
            "       |   Unzipping corpora/lin_thesaurus.zip.\n",
            "       | Downloading package mac_morpho to /root/nltk_data...\n",
            "       |   Unzipping corpora/mac_morpho.zip.\n",
            "       | Downloading package machado to /root/nltk_data...\n",
            "       | Downloading package masc_tagged to /root/nltk_data...\n",
            "       | Downloading package moses_sample to /root/nltk_data...\n",
            "       |   Unzipping models/moses_sample.zip.\n",
            "       | Downloading package movie_reviews to /root/nltk_data...\n",
            "       |   Unzipping corpora/movie_reviews.zip.\n",
            "       | Downloading package names to /root/nltk_data...\n",
            "       |   Unzipping corpora/names.zip.\n",
            "       | Downloading package nombank.1.0 to /root/nltk_data...\n",
            "       | Downloading package nps_chat to /root/nltk_data...\n",
            "       |   Unzipping corpora/nps_chat.zip.\n",
            "       | Downloading package omw to /root/nltk_data...\n",
            "       |   Unzipping corpora/omw.zip.\n",
            "       | Downloading package opinion_lexicon to /root/nltk_data...\n",
            "       |   Unzipping corpora/opinion_lexicon.zip.\n",
            "       | Downloading package paradigms to /root/nltk_data...\n",
            "       |   Unzipping corpora/paradigms.zip.\n",
            "       | Downloading package pil to /root/nltk_data...\n",
            "       |   Unzipping corpora/pil.zip.\n",
            "       | Downloading package pl196x to /root/nltk_data...\n",
            "       |   Unzipping corpora/pl196x.zip.\n",
            "       | Downloading package ppattach to /root/nltk_data...\n",
            "       |   Unzipping corpora/ppattach.zip.\n",
            "       | Downloading package problem_reports to /root/nltk_data...\n",
            "       |   Unzipping corpora/problem_reports.zip.\n",
            "       | Downloading package propbank to /root/nltk_data...\n",
            "       | Downloading package ptb to /root/nltk_data...\n",
            "       |   Unzipping corpora/ptb.zip.\n",
            "       | Downloading package product_reviews_1 to /root/nltk_data...\n",
            "       |   Unzipping corpora/product_reviews_1.zip.\n",
            "       | Downloading package product_reviews_2 to /root/nltk_data...\n",
            "       |   Unzipping corpora/product_reviews_2.zip.\n",
            "       | Downloading package pros_cons to /root/nltk_data...\n",
            "       |   Unzipping corpora/pros_cons.zip.\n",
            "       | Downloading package qc to /root/nltk_data...\n",
            "       |   Unzipping corpora/qc.zip.\n",
            "       | Downloading package reuters to /root/nltk_data...\n",
            "       | Downloading package rte to /root/nltk_data...\n",
            "       |   Unzipping corpora/rte.zip.\n",
            "       | Downloading package semcor to /root/nltk_data...\n",
            "       | Downloading package senseval to /root/nltk_data...\n",
            "       |   Unzipping corpora/senseval.zip.\n",
            "       | Downloading package sentiwordnet to /root/nltk_data...\n",
            "       |   Unzipping corpora/sentiwordnet.zip.\n",
            "       | Downloading package sentence_polarity to /root/nltk_data...\n",
            "       |   Unzipping corpora/sentence_polarity.zip.\n",
            "       | Downloading package shakespeare to /root/nltk_data...\n",
            "       |   Unzipping corpora/shakespeare.zip.\n",
            "       | Downloading package sinica_treebank to /root/nltk_data...\n",
            "       |   Unzipping corpora/sinica_treebank.zip.\n",
            "       | Downloading package smultron to /root/nltk_data...\n",
            "       |   Unzipping corpora/smultron.zip.\n",
            "       | Downloading package state_union to /root/nltk_data...\n",
            "       |   Unzipping corpora/state_union.zip.\n",
            "       | Downloading package stopwords to /root/nltk_data...\n",
            "       |   Unzipping corpora/stopwords.zip.\n",
            "       | Downloading package subjectivity to /root/nltk_data...\n",
            "       |   Unzipping corpora/subjectivity.zip.\n",
            "       | Downloading package swadesh to /root/nltk_data...\n",
            "       |   Unzipping corpora/swadesh.zip.\n",
            "       | Downloading package switchboard to /root/nltk_data...\n",
            "       |   Unzipping corpora/switchboard.zip.\n",
            "       | Downloading package timit to /root/nltk_data...\n",
            "       |   Unzipping corpora/timit.zip.\n",
            "       | Downloading package toolbox to /root/nltk_data...\n",
            "       |   Unzipping corpora/toolbox.zip.\n",
            "       | Downloading package treebank to /root/nltk_data...\n",
            "       |   Unzipping corpora/treebank.zip.\n",
            "       | Downloading package twitter_samples to /root/nltk_data...\n",
            "       |   Unzipping corpora/twitter_samples.zip.\n",
            "       | Downloading package udhr to /root/nltk_data...\n",
            "       |   Unzipping corpora/udhr.zip.\n",
            "       | Downloading package udhr2 to /root/nltk_data...\n",
            "       |   Unzipping corpora/udhr2.zip.\n",
            "       | Downloading package unicode_samples to /root/nltk_data...\n",
            "       |   Unzipping corpora/unicode_samples.zip.\n",
            "       | Downloading package universal_treebanks_v20 to\n",
            "       |     /root/nltk_data...\n",
            "       | Downloading package verbnet to /root/nltk_data...\n",
            "       |   Unzipping corpora/verbnet.zip.\n",
            "       | Downloading package verbnet3 to /root/nltk_data...\n",
            "       |   Unzipping corpora/verbnet3.zip.\n",
            "       | Downloading package webtext to /root/nltk_data...\n",
            "       |   Unzipping corpora/webtext.zip.\n",
            "       | Downloading package wordnet to /root/nltk_data...\n",
            "       |   Unzipping corpora/wordnet.zip.\n",
            "       | Downloading package wordnet_ic to /root/nltk_data...\n",
            "       |   Unzipping corpora/wordnet_ic.zip.\n",
            "       | Downloading package words to /root/nltk_data...\n",
            "       |   Unzipping corpora/words.zip.\n",
            "       | Downloading package ycoe to /root/nltk_data...\n",
            "       |   Unzipping corpora/ycoe.zip.\n",
            "       | Downloading package rslp to /root/nltk_data...\n",
            "       |   Unzipping stemmers/rslp.zip.\n",
            "       | Downloading package maxent_treebank_pos_tagger to\n",
            "       |     /root/nltk_data...\n",
            "       |   Unzipping taggers/maxent_treebank_pos_tagger.zip.\n",
            "       | Downloading package universal_tagset to /root/nltk_data...\n",
            "       |   Unzipping taggers/universal_tagset.zip.\n",
            "       | Downloading package maxent_ne_chunker to /root/nltk_data...\n",
            "       |   Unzipping chunkers/maxent_ne_chunker.zip.\n",
            "       | Downloading package punkt to /root/nltk_data...\n",
            "       |   Unzipping tokenizers/punkt.zip.\n",
            "       | Downloading package book_grammars to /root/nltk_data...\n",
            "       |   Unzipping grammars/book_grammars.zip.\n",
            "       | Downloading package sample_grammars to /root/nltk_data...\n",
            "       |   Unzipping grammars/sample_grammars.zip.\n",
            "       | Downloading package spanish_grammars to /root/nltk_data...\n",
            "       |   Unzipping grammars/spanish_grammars.zip.\n",
            "       | Downloading package basque_grammars to /root/nltk_data...\n",
            "       |   Unzipping grammars/basque_grammars.zip.\n",
            "       | Downloading package large_grammars to /root/nltk_data...\n",
            "       |   Unzipping grammars/large_grammars.zip.\n",
            "       | Downloading package tagsets to /root/nltk_data...\n",
            "       |   Unzipping help/tagsets.zip.\n",
            "       | Downloading package snowball_data to /root/nltk_data...\n",
            "       | Downloading package bllip_wsj_no_aux to /root/nltk_data...\n",
            "       |   Unzipping models/bllip_wsj_no_aux.zip.\n",
            "       | Downloading package word2vec_sample to /root/nltk_data...\n",
            "       |   Unzipping models/word2vec_sample.zip.\n",
            "       | Downloading package panlex_swadesh to /root/nltk_data...\n",
            "       | Downloading package mte_teip5 to /root/nltk_data...\n",
            "       |   Unzipping corpora/mte_teip5.zip.\n",
            "       | Downloading package averaged_perceptron_tagger to\n",
            "       |     /root/nltk_data...\n",
            "       |   Unzipping taggers/averaged_perceptron_tagger.zip.\n",
            "       | Downloading package averaged_perceptron_tagger_ru to\n",
            "       |     /root/nltk_data...\n",
            "       |   Unzipping taggers/averaged_perceptron_tagger_ru.zip.\n",
            "       | Downloading package perluniprops to /root/nltk_data...\n",
            "       |   Unzipping misc/perluniprops.zip.\n",
            "       | Downloading package nonbreaking_prefixes to\n",
            "       |     /root/nltk_data...\n",
            "       |   Unzipping corpora/nonbreaking_prefixes.zip.\n",
            "       | Downloading package vader_lexicon to /root/nltk_data...\n",
            "       | Downloading package porter_test to /root/nltk_data...\n",
            "       |   Unzipping stemmers/porter_test.zip.\n",
            "       | Downloading package wmt15_eval to /root/nltk_data...\n",
            "       |   Unzipping models/wmt15_eval.zip.\n",
            "       | Downloading package mwa_ppdb to /root/nltk_data...\n",
            "       |   Unzipping misc/mwa_ppdb.zip.\n",
            "       | \n",
            "     Done downloading collection all\n",
            "\n",
            "---------------------------------------------------------------------------\n",
            "    d) Download   l) List    u) Update   c) Config   h) Help   q) Quit\n",
            "---------------------------------------------------------------------------\n",
            "Downloader> q\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "True"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 3
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "VM-w2F82gu0g",
        "colab_type": "code",
        "outputId": "52f630a7-f1a2-4679-d31b-03d5c3ef2734",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 577
        }
      },
      "source": [
        "tokens = nltk.word_tokenize(raw)\n",
        "tokens[:30]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['Crime',\n",
              " 'and',\n",
              " 'Punishment',\n",
              " ',',\n",
              " 'by',\n",
              " 'Fyodor',\n",
              " 'Dostoevsky',\n",
              " 'body',\n",
              " '{',\n",
              " 'margin:5',\n",
              " '%',\n",
              " ';',\n",
              " 'background',\n",
              " ':',\n",
              " '#',\n",
              " 'faebd0',\n",
              " ';',\n",
              " 'text-align',\n",
              " ':',\n",
              " 'justify',\n",
              " '}',\n",
              " 'P',\n",
              " '{',\n",
              " 'text-indent',\n",
              " ':',\n",
              " '1em',\n",
              " ';',\n",
              " 'margin-top',\n",
              " ':',\n",
              " '.25em']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "DBPSm0FvmdiD",
        "colab_type": "code",
        "outputId": "e279f6cd-3ba0-4133-e314-c65d09622c00",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "text = nltk.Text(tokens) #text객체로 변환하면 여러 메소드 사용 가능\n",
        "type(text)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "nltk.text.Text"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 27
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "-4bWnx8gm1k8",
        "colab_type": "code",
        "outputId": "5240a4b0-7a95-410b-d699-12e3d4ad3924",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 727
        }
      },
      "source": [
        "text[1024:1062]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['of',\n",
              " 'epilepsy',\n",
              " ',',\n",
              " 'from',\n",
              " 'which',\n",
              " 'he',\n",
              " 'suffered',\n",
              " 'for',\n",
              " 'the',\n",
              " 'rest',\n",
              " 'of',\n",
              " 'his',\n",
              " 'life',\n",
              " '.',\n",
              " 'The',\n",
              " 'fits',\n",
              " 'occurred',\n",
              " 'three',\n",
              " 'or',\n",
              " 'four',\n",
              " 'times',\n",
              " 'a',\n",
              " 'year',\n",
              " 'and',\n",
              " 'were',\n",
              " 'more',\n",
              " 'frequent',\n",
              " 'in',\n",
              " 'periods',\n",
              " 'of',\n",
              " 'great',\n",
              " 'strain',\n",
              " '.',\n",
              " 'In',\n",
              " '1859',\n",
              " 'he',\n",
              " 'was',\n",
              " 'allowed']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "QeXQkEYom40m",
        "colab_type": "code",
        "outputId": "30fda6ba-21ec-4863-e5fb-436beb238820",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 129
        }
      },
      "source": [
        "text.collocations() #연어"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Katerina Ivanovna; Pyotr Petrovitch; Pulcheria Alexandrovna; Avdotya\n",
            "Romanovna; Rodion Romanovitch; Marfa Petrovna; Sofya Semyonovna; old\n",
            "woman; Project Gutenberg-tm; Porfiry Petrovitch; Amalia Ivanovna;\n",
            "great deal; young man; Nikodim Fomitch; Ilya Petrovitch; Project\n",
            "Gutenberg; Andrey Semyonovitch; Hay Market; Dmitri Prokofitch; Good\n",
            "heavens\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "NDWuie_hm9tm",
        "colab_type": "code",
        "outputId": "f87b8428-8839-4c57-c5eb-5f5a65c7205f",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "text.concordance('gene') #단어가 쓰인 모든 경우의 문맥을 찾아줌(색인)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "No matches\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Rz8Pf9N0osrB",
        "colab_type": "code",
        "outputId": "df6f37c4-320d-4cce-a54a-356f68e8d5e7",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 224
        }
      },
      "source": [
        "!pip install feedparser"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Collecting feedparser\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/91/d8/7d37fec71ff7c9dbcdd80d2b48bcdd86d6af502156fc93846fb0102cb2c4/feedparser-5.2.1.tar.bz2 (192kB)\n",
            "\r\u001b[K     |█▊                              | 10kB 18.5MB/s eta 0:00:01\r\u001b[K     |███▍                            | 20kB 1.7MB/s eta 0:00:01\r\u001b[K     |█████▏                          | 30kB 2.3MB/s eta 0:00:01\r\u001b[K     |██████▉                         | 40kB 1.7MB/s eta 0:00:01\r\u001b[K     |████████▌                       | 51kB 1.9MB/s eta 0:00:01\r\u001b[K     |██████████▎                     | 61kB 2.2MB/s eta 0:00:01\r\u001b[K     |████████████                    | 71kB 2.4MB/s eta 0:00:01\r\u001b[K     |█████████████▋                  | 81kB 2.5MB/s eta 0:00:01\r\u001b[K     |███████████████▍                | 92kB 2.8MB/s eta 0:00:01\r\u001b[K     |█████████████████               | 102kB 2.8MB/s eta 0:00:01\r\u001b[K     |██████████████████▊             | 112kB 2.8MB/s eta 0:00:01\r\u001b[K     |████████████████████▌           | 122kB 2.8MB/s eta 0:00:01\r\u001b[K     |██████████████████████▏         | 133kB 2.8MB/s eta 0:00:01\r\u001b[K     |███████████████████████▉        | 143kB 2.8MB/s eta 0:00:01\r\u001b[K     |█████████████████████████▋      | 153kB 2.8MB/s eta 0:00:01\r\u001b[K     |███████████████████████████▎    | 163kB 2.8MB/s eta 0:00:01\r\u001b[K     |█████████████████████████████   | 174kB 2.8MB/s eta 0:00:01\r\u001b[K     |██████████████████████████████▊ | 184kB 2.8MB/s eta 0:00:01\r\u001b[K     |████████████████████████████████| 194kB 2.8MB/s \n",
            "\u001b[?25hBuilding wheels for collected packages: feedparser\n",
            "  Building wheel for feedparser (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for feedparser: filename=feedparser-5.2.1-cp36-none-any.whl size=44940 sha256=64805bb5fca7bdc119a2f0137edec3f686da0011ce20ca4dddfc00508f8dd5f6\n",
            "  Stored in directory: /root/.cache/pip/wheels/8c/69/b7/f52763c41c5471df57703a0ef718a32a5e81ee35dcf6d4f97f\n",
            "Successfully built feedparser\n",
            "Installing collected packages: feedparser\n",
            "Successfully installed feedparser-5.2.1\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "CBZfdRj0nkYY",
        "colab_type": "code",
        "outputId": "44b646cb-8de0-4ea3-a155-9df3d7b6c326",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "#blog 글\n",
        "import feedparser\n",
        "llog = feedparser.parse('http://languagelog.ldc.upenn.edu/nll/?feed=atom')\n",
        "llog['feed']['title']"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Language Log'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 33
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ZOs03eLio6OW",
        "colab_type": "code",
        "outputId": "91a095d4-454a-4eb0-96ff-ad616ac85150",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(llog.entries)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "13"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 34
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "2BU9CxEDptZn",
        "colab_type": "text"
      },
      "source": [
        "파싱 결과는 FeedParserDict 자료형 객체에 저장된다. FeedParserDict 는 피드 정보를 담은 feed 속성과 개별적인 항목을 담은 entries 속성을 가진다."
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6Gp6vfxSo_4g",
        "colab_type": "code",
        "outputId": "e1784259-a742-49c0-d7bb-49b9de9b0b50",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "post = llog.entries[2]\n",
        "post.title"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'A curse from the novel coronavirus epicenter'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 35
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "f-H5wmjqqKoE",
        "colab_type": "code",
        "outputId": "222b60b1-d1fa-46fa-bc99-ba96205fa45b",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "content = post.content[0].value\n",
        "content[:70]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'<p>The whole world is now thoroughly familiar with the name \"Wuhan\", w'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 36
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6f3YmKnPqZs_",
        "colab_type": "code",
        "outputId": "d3241919-fa72-4d42-b5f6-73233b122198",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "raw = BeautifulSoup(content).get_text()\n",
        "nltk.word_tokenize(raw)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['The',\n",
              " 'whole',\n",
              " 'world',\n",
              " 'is',\n",
              " 'now',\n",
              " 'thoroughly',\n",
              " 'familiar',\n",
              " 'with',\n",
              " 'the',\n",
              " 'name',\n",
              " '``',\n",
              " 'Wuhan',\n",
              " \"''\",\n",
              " ',',\n",
              " 'whereas',\n",
              " 'four',\n",
              " 'months',\n",
              " 'ago',\n",
              " ',',\n",
              " 'only',\n",
              " 'a',\n",
              " 'small',\n",
              " 'number',\n",
              " 'of',\n",
              " 'people',\n",
              " 'outside',\n",
              " 'of',\n",
              " 'China',\n",
              " 'would',\n",
              " 'have',\n",
              " 'heard',\n",
              " 'of',\n",
              " 'it',\n",
              " '.',\n",
              " 'Since',\n",
              " ',',\n",
              " 'two',\n",
              " 'days',\n",
              " 'ago',\n",
              " ',',\n",
              " 'I',\n",
              " 'posted',\n",
              " 'about',\n",
              " 'Dutch',\n",
              " 'curses',\n",
              " ',',\n",
              " 'many',\n",
              " 'of',\n",
              " 'which',\n",
              " 'just',\n",
              " 'so',\n",
              " 'happen',\n",
              " 'to',\n",
              " 'be',\n",
              " 'linked',\n",
              " 'to',\n",
              " 'diseases',\n",
              " ',',\n",
              " 'I',\n",
              " 'am',\n",
              " 'prompted',\n",
              " 'to',\n",
              " 'dust',\n",
              " 'off',\n",
              " 'an',\n",
              " 'old',\n",
              " 'post',\n",
              " 'that',\n",
              " 'is',\n",
              " 'about',\n",
              " 'a',\n",
              " 'colorful',\n",
              " 'curse',\n",
              " 'from',\n",
              " 'Wuhan',\n",
              " ',',\n",
              " 'which',\n",
              " ',',\n",
              " 'by',\n",
              " 'the',\n",
              " 'way',\n",
              " ',',\n",
              " 'is',\n",
              " 'famous',\n",
              " 'among',\n",
              " 'all',\n",
              " 'Chinese',\n",
              " 'cities',\n",
              " 'for',\n",
              " 'the',\n",
              " 'proclivity',\n",
              " 'of',\n",
              " 'its',\n",
              " 'inhabitants',\n",
              " 'to',\n",
              " 'indulge',\n",
              " 'in',\n",
              " 'sharp-tongued',\n",
              " 'imprecations',\n",
              " 'at',\n",
              " 'the',\n",
              " 'slightest',\n",
              " 'provocation',\n",
              " '.',\n",
              " 'I',\n",
              " 'myself',\n",
              " 'have',\n",
              " 'been',\n",
              " 'witness',\n",
              " 'to',\n",
              " 'their',\n",
              " 'talent',\n",
              " 'in',\n",
              " 'this',\n",
              " 'art',\n",
              " ',',\n",
              " 'at',\n",
              " 'which',\n",
              " 'the',\n",
              " 'women',\n",
              " 'are',\n",
              " 'especially',\n",
              " 'adept',\n",
              " '.',\n",
              " 'The',\n",
              " 'particular',\n",
              " 'Wuhan',\n",
              " 'curse',\n",
              " 'in',\n",
              " 'question',\n",
              " 'here',\n",
              " 'looks',\n",
              " 'innocent',\n",
              " 'enough',\n",
              " 'on',\n",
              " 'the',\n",
              " 'surface',\n",
              " ':',\n",
              " 'gè',\n",
              " 'bānmǎ',\n",
              " '个斑马',\n",
              " '(',\n",
              " '``',\n",
              " 'a',\n",
              " 'zebra',\n",
              " \"''\",\n",
              " ')',\n",
              " '.',\n",
              " 'But',\n",
              " 'when',\n",
              " 'we',\n",
              " 'probe',\n",
              " 'more',\n",
              " 'deeply',\n",
              " 'into',\n",
              " 'Wuhan',\n",
              " 'topolect',\n",
              " ',',\n",
              " 'we',\n",
              " 'find',\n",
              " 'that',\n",
              " 'it',\n",
              " 'actually',\n",
              " 'means',\n",
              " '``',\n",
              " 'a',\n",
              " 'motherf*cker',\n",
              " \"''\",\n",
              " 'and',\n",
              " 'has',\n",
              " 'nothing',\n",
              " 'whatsoever',\n",
              " 'to',\n",
              " 'do',\n",
              " 'with',\n",
              " 'zebras',\n",
              " 'or',\n",
              " 'even',\n",
              " 'with',\n",
              " 'horses',\n",
              " '.',\n",
              " 'How',\n",
              " 'that',\n",
              " 'happens',\n",
              " 'is',\n",
              " 'extremely',\n",
              " 'complicated',\n",
              " ',',\n",
              " 'and',\n",
              " 'I',\n",
              " 'have',\n",
              " 'spelled',\n",
              " 'it',\n",
              " 'all',\n",
              " 'out',\n",
              " 'in',\n",
              " 'this',\n",
              " 'post',\n",
              " ':',\n",
              " \"''\",\n",
              " 'Why',\n",
              " 'zebra',\n",
              " \"'\",\n",
              " '?',\n",
              " \"''\",\n",
              " '(',\n",
              " '5/29/16',\n",
              " ')',\n",
              " 'The',\n",
              " 'linguistics',\n",
              " 'of',\n",
              " 'cursing',\n",
              " '—',\n",
              " 'especially',\n",
              " 'in',\n",
              " 'Wuhan',\n",
              " '—',\n",
              " 'is',\n",
              " 'not',\n",
              " 'for',\n",
              " 'the',\n",
              " 'faint',\n",
              " 'of',\n",
              " 'heart',\n",
              " '.',\n",
              " 'Selected',\n",
              " 'readings',\n",
              " \"''\",\n",
              " 'Dutch',\n",
              " 'curses',\n",
              " \"''\",\n",
              " '(',\n",
              " '4/19/20',\n",
              " ')',\n",
              " '—',\n",
              " 'with',\n",
              " 'additional',\n",
              " 'references',\n",
              " 'to',\n",
              " 'cursing',\n",
              " 'in',\n",
              " 'the',\n",
              " '``',\n",
              " 'Selected',\n",
              " 'readings',\n",
              " \"''\",\n",
              " \"''\",\n",
              " 'The',\n",
              " 'impact',\n",
              " 'of',\n",
              " 'COVID-19',\n",
              " 'on',\n",
              " 'Russian',\n",
              " \"''\",\n",
              " '(',\n",
              " '4/18/20',\n",
              " ')',\n",
              " '—',\n",
              " 'with',\n",
              " 'additional',\n",
              " 'references',\n",
              " 'to',\n",
              " 'COVID-19',\n",
              " 'in',\n",
              " 'the',\n",
              " '``',\n",
              " 'Selected',\n",
              " 'readings',\n",
              " \"''\"]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "iXjMNxDjq_9m",
        "colab_type": "code",
        "outputId": "63dd68bc-a136-4981-9f1b-f6549d4018b6",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "import os\n",
        "os.getcwd()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'/content/gdrive/My Drive/인공지능 수업 데이터 파일'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 39
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bR7MbM5xrGFz",
        "colab_type": "code",
        "outputId": "e0fe30ae-550b-4344-9a25-bce54a5c06f0",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 111
        }
      },
      "source": [
        "os.listdir('.')"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['harvard_england.wav',\n",
              " 'gni-6-1-1.txt',\n",
              " 'purchases.csv',\n",
              " 'purchases.json',\n",
              " 'Genomics-Informatics-Corpus-master']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 40
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "JAwT5L2irJ9Z",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "os.chdir('Genomics-Informatics-Corpus-master/') #cd to child directory"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "BBSY07fXrSci",
        "colab_type": "code",
        "outputId": "89142cb8-5791-4ed0-8d34-45c4226a0055",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 223
        }
      },
      "source": [
        "os.listdir()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['no_backoff_only_genia.txt',\n",
              " 'genia_and_backoff.txt',\n",
              " 'README.md',\n",
              " 'no_genia_only_backoff.txt',\n",
              " 'code',\n",
              " 'raw_text',\n",
              " 'GI_tag',\n",
              " 'raw_text2',\n",
              " 'GNI Corpus 1.0',\n",
              " 'original_text',\n",
              " 'XML']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 42
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "cqP-T10qrdb9",
        "colab_type": "text"
      },
      "source": [
        "PDF, MSWord, Other binary formats to txt\n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "N2B9-EUzwGgm",
        "colab_type": "text"
      },
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n",
        "\n",
        "python 입출력 command\n"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "qklD65YDrY_8",
        "colab_type": "code",
        "outputId": "47c78b72-78a8-44e1-8b78-ebd89eecd682",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "s = input(\"Enter some text: \")"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Enter some text: On an exceptionally hot evening early in July\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "FZ05sgUdwW1f",
        "colab_type": "code",
        "outputId": "2a17837c-70d2-48c6-dcba-5ec488cee371",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "print(\"You typed \", len(nltk.word_tokenize(s)), \"words.\")"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "You typed  8 words.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "w6VEiOMkwi7s",
        "colab_type": "code",
        "outputId": "f9fae185-7af8-4b40-e2d4-1acc398c7f12",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "couplet = \"Shall I compare thee to a Summer's day?\"\\\n",
        "\"Thou are more lovely and more temperate: \" #혹은 괄호로 묶어도 됨\n",
        "print(couplet)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Shall I compare thee to a Summer's day?Thou are more lovely and more temperate: \n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "FFo-RL7CxIPg",
        "colab_type": "code",
        "outputId": "cdbe28f7-824c-44a4-da4a-fb05a9fc40d2",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 55
        }
      },
      "source": [
        "couplet = '''Rough winds do shake the darling buds of May,\n",
        "And Summer's lease hath all too short a date: '''\n",
        "print(couplet) #enter로 나눈 문장 그대로 프린트"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Rough winds do shake the darling buds of May,\n",
            "And Summer's lease hath all too short a date: \n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "fTiMJYUcxc54",
        "colab_type": "code",
        "outputId": "e9fcc483-9d30-43e0-f915-e5b391daa612",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "sent = 'colorless green ideas sleep furiously'\n",
        "for char in sent:\n",
        "  print(char, end=' ') #end의 default는 \\n이므로 그냥 print문을 쓰면 하나씩 줄바꿔 프린트(세로로)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "c o l o r l e s s   g r e e n   i d e a s   s l e e p   f u r i o u s l y "
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "PP8NGyX-x3Fc",
        "colab_type": "code",
        "outputId": "59768949-65dd-4b13-8af8-d22c3705517c",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "from nltk.corpus import gutenberg\n",
        "raw = gutenberg.raw('melville-moby_dick.txt')\n",
        "fdist = nltk.FreqDist(ch.lower() for ch in raw if ch.isalpha())\n",
        "fdist.most_common(5)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[('e', 117092), ('t', 87996), ('a', 77916), ('o', 69326), ('n', 65617)]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 48
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "mQOLX4hgyPT_",
        "colab_type": "code",
        "outputId": "dcecffb2-8cd5-486b-e551-05b0448e65bf",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 503
        }
      },
      "source": [
        "[char for (char, count) in fdist.most_common()]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['e',\n",
              " 't',\n",
              " 'a',\n",
              " 'o',\n",
              " 'n',\n",
              " 'i',\n",
              " 's',\n",
              " 'h',\n",
              " 'r',\n",
              " 'l',\n",
              " 'd',\n",
              " 'u',\n",
              " 'm',\n",
              " 'c',\n",
              " 'w',\n",
              " 'f',\n",
              " 'g',\n",
              " 'p',\n",
              " 'b',\n",
              " 'y',\n",
              " 'v',\n",
              " 'k',\n",
              " 'q',\n",
              " 'j',\n",
              " 'x',\n",
              " 'z']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 50
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qjGVJHMsyx8v",
        "colab_type": "text"
      },
      "source": [
        "* 유니코드\n",
        "\n",
        "`# -*- coding: utf-8 -*-`\n",
        "\n",
        "* 파이썬 인코딩 error handling"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "YNno8ilVf6DO",
        "colab_type": "text"
      },
      "source": [
        "\n",
        "\n",
        "---\n",
        "* Regular expression\n",
        "\n",
        "low level pattern 분석"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "55nhmcA8yYNS",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "import re\n",
        "wordlist = [w for w in nltk.corpus.words.words('en') if w.islower()]"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "YFv7AVXt0ul4",
        "colab_type": "code",
        "outputId": "d962a21f-7b07-447c-f4ab-eefb6350b8c4",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(wordlist)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "210687"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ThMmP9HN0w29",
        "colab_type": "code",
        "outputId": "cad9b939-46d9-4ea1-c021-f0fe0880fecb",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "[w for w in wordlist if re.search('^abc', w)] #abc로 시작하는 단어 찾기"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['abcoulomb']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "jR_Lki1v07ZL",
        "colab_type": "code",
        "outputId": "90c06f89-79d9-488a-ece8-11128207f5e1",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 316
        }
      },
      "source": [
        "[w for w in wordlist if re.search('omics$', w)] # omics로 끝나는 단어 찾기"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['aerodromics',\n",
              " 'agronomics',\n",
              " 'anthroponomics',\n",
              " 'astronomics',\n",
              " 'atomics',\n",
              " 'bionomics',\n",
              " 'economics',\n",
              " 'ethonomics',\n",
              " 'hydroeconomics',\n",
              " 'loxodromics',\n",
              " 'orthodromics',\n",
              " 'phoronomics',\n",
              " 'physiognomics',\n",
              " 'psychonomics',\n",
              " 'pyronomics',\n",
              " 'socionomics']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 7
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "hyUlqjgb1Fq0",
        "colab_type": "code",
        "outputId": "df474fcd-6eeb-4371-9726-c2c2122b21d0",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 764
        }
      },
      "source": [
        "[w for w in wordlist if re.search('[abc][abc][abc][abc][abc]', w)] #*1~5* 단어에서 1~5가 abc중 하나인 단어"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['abaca',\n",
              " 'abacate',\n",
              " 'abacay',\n",
              " 'abbacomes',\n",
              " 'abbacy',\n",
              " 'acacatechin',\n",
              " 'acacatechol',\n",
              " 'asarabacca',\n",
              " 'babacoote',\n",
              " 'bacaba',\n",
              " 'bacach',\n",
              " 'bacbakiri',\n",
              " 'bacca',\n",
              " 'baccaceous',\n",
              " 'baccae',\n",
              " 'baccalaurean',\n",
              " 'baccalaureate',\n",
              " 'baccara',\n",
              " 'baccarat',\n",
              " 'baccate',\n",
              " 'baccated',\n",
              " 'bombacaceous',\n",
              " 'cabaan',\n",
              " 'caback',\n",
              " 'cabbage',\n",
              " 'cabbagehead',\n",
              " 'cabbagewood',\n",
              " 'cabbagy',\n",
              " 'grabbable',\n",
              " 'historicocabbalistical',\n",
              " 'maccaboy',\n",
              " 'phytolaccaceous',\n",
              " 'piacaba',\n",
              " 'pseudoacaccia',\n",
              " 'sarabacan',\n",
              " 'scabbard',\n",
              " 'scabbardless',\n",
              " 'taccaceous',\n",
              " 'unscabbard',\n",
              " 'unscabbarded']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "s_2yhr_L1u2B",
        "colab_type": "code",
        "outputId": "9b46d022-bb11-4348-9cef-1be135d3eb15",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "[w for w in wordlist if re.search('aab+', w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['haab', 'staab']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "8ywnnRKE1_xR",
        "colab_type": "code",
        "outputId": "8fc6a6b9-99f7-4d67-ba1e-69b53d623030",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "[w for w in wordlist if re.search('^[ghi][mno][jlk][def]$', w)] #휴대폰 패드의 4653에 있는 알파벳"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['gold', 'golf', 'hold', 'hole']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "CrbKsRlf2feP",
        "colab_type": "code",
        "outputId": "31575311-82cd-4f29-83ee-4888f689c4a5",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 92
        }
      },
      "source": [
        "chat_words = sorted(set(w for w in nltk.corpus.nps_chat.words())) # 미해병대 chatting\n",
        "[w for w in chat_words if re.search('^m+i+n+e+$', w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['miiiiiiiiiiiiinnnnnnnnnnneeeeeeeeee',\n",
              " 'miiiiiinnnnnnnnnneeeeeeee',\n",
              " 'mine',\n",
              " 'mmmmmmmmiiiiiiiiinnnnnnnnneeeeeeee']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "TWbnrujR28pw",
        "colab_type": "code",
        "outputId": "6d511f63-4494-49f7-cc90-ec12fca91366",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 559
        }
      },
      "source": [
        "[w for w in chat_words if re.search('^[ha]+$', w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['a',\n",
              " 'aaaaaaaaaaaaaaaaa',\n",
              " 'aaahhhh',\n",
              " 'ah',\n",
              " 'ahah',\n",
              " 'ahahah',\n",
              " 'ahh',\n",
              " 'ahhahahaha',\n",
              " 'ahhh',\n",
              " 'ahhhh',\n",
              " 'ahhhhhh',\n",
              " 'ahhhhhhhhhhhhhh',\n",
              " 'h',\n",
              " 'ha',\n",
              " 'haaa',\n",
              " 'hah',\n",
              " 'haha',\n",
              " 'hahaaa',\n",
              " 'hahah',\n",
              " 'hahaha',\n",
              " 'hahahaa',\n",
              " 'hahahah',\n",
              " 'hahahaha',\n",
              " 'hahahahaaa',\n",
              " 'hahahahahaha',\n",
              " 'hahahahahahaha',\n",
              " 'hahahahahahahahahahahahahahahaha',\n",
              " 'hahahhahah',\n",
              " 'hahhahahaha']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "XtjMTBbc3IUu",
        "colab_type": "code",
        "outputId": "2083e694-4130-4f07-8869-2e6a6217ee59",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "wsj = sorted(set(nltk.corpus.treebank.words()))\n",
        "[w for w in wsj if re.search('^[0-9]+\\.+[0-9]+$',w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['0.0085',\n",
              " '0.05',\n",
              " '0.1',\n",
              " '0.16',\n",
              " '0.2',\n",
              " '0.25',\n",
              " '0.28',\n",
              " '0.3',\n",
              " '0.4',\n",
              " '0.5',\n",
              " '0.50',\n",
              " '0.54',\n",
              " '0.56',\n",
              " '0.60',\n",
              " '0.7',\n",
              " '0.82',\n",
              " '0.84',\n",
              " '0.9',\n",
              " '0.95',\n",
              " '0.99',\n",
              " '1.01',\n",
              " '1.1',\n",
              " '1.125',\n",
              " '1.14',\n",
              " '1.1650',\n",
              " '1.17',\n",
              " '1.18',\n",
              " '1.19',\n",
              " '1.2',\n",
              " '1.20',\n",
              " '1.24',\n",
              " '1.25',\n",
              " '1.26',\n",
              " '1.28',\n",
              " '1.35',\n",
              " '1.39',\n",
              " '1.4',\n",
              " '1.457',\n",
              " '1.46',\n",
              " '1.49',\n",
              " '1.5',\n",
              " '1.50',\n",
              " '1.55',\n",
              " '1.56',\n",
              " '1.5755',\n",
              " '1.5805',\n",
              " '1.6',\n",
              " '1.61',\n",
              " '1.637',\n",
              " '1.64',\n",
              " '1.65',\n",
              " '1.7',\n",
              " '1.75',\n",
              " '1.76',\n",
              " '1.8',\n",
              " '1.82',\n",
              " '1.8415',\n",
              " '1.85',\n",
              " '1.8500',\n",
              " '1.9',\n",
              " '1.916',\n",
              " '1.92',\n",
              " '10.19',\n",
              " '10.2',\n",
              " '10.5',\n",
              " '107.03',\n",
              " '107.9',\n",
              " '109.73',\n",
              " '11.10',\n",
              " '11.5',\n",
              " '11.57',\n",
              " '11.6',\n",
              " '11.72',\n",
              " '11.95',\n",
              " '112.9',\n",
              " '113.2',\n",
              " '116.3',\n",
              " '116.4',\n",
              " '116.7',\n",
              " '116.9',\n",
              " '118.6',\n",
              " '12.09',\n",
              " '12.5',\n",
              " '12.52',\n",
              " '12.68',\n",
              " '12.7',\n",
              " '12.82',\n",
              " '12.97',\n",
              " '120.7',\n",
              " '1206.26',\n",
              " '121.6',\n",
              " '126.1',\n",
              " '126.15',\n",
              " '127.03',\n",
              " '129.91',\n",
              " '13.1',\n",
              " '13.15',\n",
              " '13.5',\n",
              " '13.50',\n",
              " '13.625',\n",
              " '13.65',\n",
              " '13.73',\n",
              " '13.8',\n",
              " '13.90',\n",
              " '130.6',\n",
              " '130.7',\n",
              " '131.01',\n",
              " '132.9',\n",
              " '133.7',\n",
              " '133.8',\n",
              " '14.00',\n",
              " '14.13',\n",
              " '14.26',\n",
              " '14.28',\n",
              " '14.43',\n",
              " '14.5',\n",
              " '14.53',\n",
              " '14.54',\n",
              " '14.6',\n",
              " '14.75',\n",
              " '14.99',\n",
              " '141.9',\n",
              " '142.84',\n",
              " '142.85',\n",
              " '143.08',\n",
              " '143.80',\n",
              " '143.93',\n",
              " '148.9',\n",
              " '149.9',\n",
              " '15.5',\n",
              " '150.00',\n",
              " '153.3',\n",
              " '154.2',\n",
              " '16.05',\n",
              " '16.09',\n",
              " '16.125',\n",
              " '16.2',\n",
              " '16.5',\n",
              " '16.68',\n",
              " '16.7',\n",
              " '16.9',\n",
              " '169.9',\n",
              " '17.3',\n",
              " '17.4',\n",
              " '17.5',\n",
              " '17.95',\n",
              " '1738.1',\n",
              " '176.1',\n",
              " '18.3',\n",
              " '18.6',\n",
              " '18.95',\n",
              " '185.9',\n",
              " '188.84',\n",
              " '19.3',\n",
              " '19.50',\n",
              " '19.6',\n",
              " '19.94',\n",
              " '19.95',\n",
              " '191.9',\n",
              " '2.07',\n",
              " '2.1',\n",
              " '2.15',\n",
              " '2.19',\n",
              " '2.2',\n",
              " '2.25',\n",
              " '2.29',\n",
              " '2.3',\n",
              " '2.30',\n",
              " '2.35',\n",
              " '2.375',\n",
              " '2.4',\n",
              " '2.42',\n",
              " '2.44',\n",
              " '2.46',\n",
              " '2.47',\n",
              " '2.5',\n",
              " '2.50',\n",
              " '2.6',\n",
              " '2.62',\n",
              " '2.65',\n",
              " '2.7',\n",
              " '2.75',\n",
              " '2.8',\n",
              " '2.80',\n",
              " '2.87',\n",
              " '2.875',\n",
              " '2.9',\n",
              " '2.95',\n",
              " '20.07',\n",
              " '20.5',\n",
              " '21.1',\n",
              " '21.9',\n",
              " '2141.7',\n",
              " '2160.1',\n",
              " '2163.2',\n",
              " '22.75',\n",
              " '220.45',\n",
              " '221.4',\n",
              " '225.6',\n",
              " '23.25',\n",
              " '23.4',\n",
              " '23.5',\n",
              " '23.72',\n",
              " '234.4',\n",
              " '236.74',\n",
              " '236.79',\n",
              " '24.95',\n",
              " '25.50',\n",
              " '25.6',\n",
              " '251.2',\n",
              " '26.2',\n",
              " '26.5',\n",
              " '26.8',\n",
              " '263.07',\n",
              " '2645.90',\n",
              " '2691.19',\n",
              " '27.1',\n",
              " '27.4',\n",
              " '273.5',\n",
              " '278.7',\n",
              " '28.25',\n",
              " '28.36',\n",
              " '28.4',\n",
              " '28.5',\n",
              " '28.53',\n",
              " '28.6',\n",
              " '29.3',\n",
              " '29.4',\n",
              " '29.9',\n",
              " '292.32',\n",
              " '3.01',\n",
              " '3.04',\n",
              " '3.1',\n",
              " '3.16',\n",
              " '3.18',\n",
              " '3.19',\n",
              " '3.2',\n",
              " '3.20',\n",
              " '3.23',\n",
              " '3.253',\n",
              " '3.28',\n",
              " '3.3',\n",
              " '3.35',\n",
              " '3.375',\n",
              " '3.4',\n",
              " '3.42',\n",
              " '3.43',\n",
              " '3.5',\n",
              " '3.55',\n",
              " '3.6',\n",
              " '3.61',\n",
              " '3.625',\n",
              " '3.7',\n",
              " '3.75',\n",
              " '3.8',\n",
              " '3.80',\n",
              " '3.9',\n",
              " '30.6',\n",
              " '30.9',\n",
              " '319.75',\n",
              " '32.8',\n",
              " '334.5',\n",
              " '34.625',\n",
              " '341.20',\n",
              " '3436.58',\n",
              " '35.2',\n",
              " '35.7',\n",
              " '352.7',\n",
              " '352.9',\n",
              " '35500.64',\n",
              " '35564.43',\n",
              " '36.9',\n",
              " '361.8',\n",
              " '3648.82',\n",
              " '37.3',\n",
              " '37.5',\n",
              " '372.14',\n",
              " '372.9',\n",
              " '374.19',\n",
              " '374.20',\n",
              " '377.60',\n",
              " '38.3',\n",
              " '38.375',\n",
              " '38.5',\n",
              " '38.875',\n",
              " '387.8',\n",
              " '4.1',\n",
              " '4.10',\n",
              " '4.2',\n",
              " '4.25',\n",
              " '4.3',\n",
              " '4.4',\n",
              " '4.5',\n",
              " '4.55',\n",
              " '4.6',\n",
              " '4.7',\n",
              " '4.75',\n",
              " '4.8',\n",
              " '4.875',\n",
              " '4.898',\n",
              " '4.9',\n",
              " '40.21',\n",
              " '41.60',\n",
              " '415.6',\n",
              " '415.8',\n",
              " '42.1',\n",
              " '42.5',\n",
              " '422.5',\n",
              " '43.875',\n",
              " '434.4',\n",
              " '436.01',\n",
              " '446.62',\n",
              " '449.04',\n",
              " '45.2',\n",
              " '45.3',\n",
              " '45.75',\n",
              " '456.64',\n",
              " '46.1',\n",
              " '47.1',\n",
              " '47.125',\n",
              " '47.5',\n",
              " '47.6',\n",
              " '49.9',\n",
              " '494.50',\n",
              " '497.34',\n",
              " '5.1',\n",
              " '5.2180',\n",
              " '5.276',\n",
              " '5.29',\n",
              " '5.3',\n",
              " '5.39',\n",
              " '5.4',\n",
              " '5.435',\n",
              " '5.5',\n",
              " '5.57',\n",
              " '5.6',\n",
              " '5.63',\n",
              " '5.7',\n",
              " '5.70',\n",
              " '5.8',\n",
              " '5.82',\n",
              " '5.9',\n",
              " '5.92',\n",
              " '50.1',\n",
              " '50.38',\n",
              " '50.45',\n",
              " '51.25',\n",
              " '51.6',\n",
              " '55.1',\n",
              " '566.54',\n",
              " '57.50',\n",
              " '57.6',\n",
              " '57.7',\n",
              " '58.64',\n",
              " '59.6',\n",
              " '59.9',\n",
              " '6.03',\n",
              " '6.1',\n",
              " '6.20',\n",
              " '6.21',\n",
              " '6.25',\n",
              " '6.4',\n",
              " '6.40',\n",
              " '6.44',\n",
              " '6.5',\n",
              " '6.50',\n",
              " '6.53',\n",
              " '6.6',\n",
              " '6.7',\n",
              " '6.70',\n",
              " '6.79',\n",
              " '6.84',\n",
              " '6.9',\n",
              " '60.36',\n",
              " '618.1',\n",
              " '62.1',\n",
              " '62.5',\n",
              " '62.625',\n",
              " '63.79',\n",
              " '630.9',\n",
              " '64.5',\n",
              " '66.5',\n",
              " '7.15',\n",
              " '7.2',\n",
              " '7.20',\n",
              " '7.272',\n",
              " '7.3',\n",
              " '7.4',\n",
              " '7.40',\n",
              " '7.422',\n",
              " '7.45',\n",
              " '7.458',\n",
              " '7.5',\n",
              " '7.50',\n",
              " '7.52',\n",
              " '7.55',\n",
              " '7.60',\n",
              " '7.62',\n",
              " '7.63',\n",
              " '7.65',\n",
              " '7.74',\n",
              " '7.78',\n",
              " '7.79',\n",
              " '7.8',\n",
              " '7.80',\n",
              " '7.84',\n",
              " '7.88',\n",
              " '7.90',\n",
              " '7.95',\n",
              " '70.2',\n",
              " '70.7',\n",
              " '705.6',\n",
              " '72.7',\n",
              " '734.9',\n",
              " '737.5',\n",
              " '77.56',\n",
              " '77.6',\n",
              " '77.70',\n",
              " '8.04',\n",
              " '8.06',\n",
              " '8.07',\n",
              " '8.1',\n",
              " '8.12',\n",
              " '8.14',\n",
              " '8.15',\n",
              " '8.19',\n",
              " '8.2',\n",
              " '8.22',\n",
              " '8.25',\n",
              " '8.30',\n",
              " '8.35',\n",
              " '8.45',\n",
              " '8.467',\n",
              " '8.47',\n",
              " '8.48',\n",
              " '8.5',\n",
              " '8.50',\n",
              " '8.53',\n",
              " '8.55',\n",
              " '8.56',\n",
              " '8.575',\n",
              " '8.60',\n",
              " '8.64',\n",
              " '8.65',\n",
              " '8.70',\n",
              " '8.75',\n",
              " '8.9',\n",
              " '80.50',\n",
              " '80.8',\n",
              " '81.8',\n",
              " '811.9',\n",
              " '83.4',\n",
              " '84.29',\n",
              " '84.9',\n",
              " '85.1',\n",
              " '85.7',\n",
              " '86.12',\n",
              " '87.5',\n",
              " '88.32',\n",
              " '89.7',\n",
              " '89.9',\n",
              " '9.3',\n",
              " '9.32',\n",
              " '9.37',\n",
              " '9.45',\n",
              " '9.5',\n",
              " '9.625',\n",
              " '9.75',\n",
              " '9.8',\n",
              " '9.82',\n",
              " '9.9',\n",
              " '92.9',\n",
              " '93.3',\n",
              " '93.9',\n",
              " '94.2',\n",
              " '94.8',\n",
              " '95.09',\n",
              " '96.4',\n",
              " '98.3',\n",
              " '99.1',\n",
              " '99.3']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "fm99lURL3dR-",
        "colab_type": "code",
        "outputId": "5682a756-cd05-47df-e36b-f31759b5c233",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "[w for w in wsj if re.search('^[A-Z]+\\$$', w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['C$', 'US$']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "jG4bOGTp3mZd",
        "colab_type": "code",
        "outputId": "ee188265-9b00-4570-eaed-ade0370bc546",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "[w for w in wsj if re.search('^[0-9]{4}$',w)] #[0-9] 4번 반복"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['1614',\n",
              " '1637',\n",
              " '1787',\n",
              " '1901',\n",
              " '1903',\n",
              " '1917',\n",
              " '1925',\n",
              " '1929',\n",
              " '1933',\n",
              " '1934',\n",
              " '1948',\n",
              " '1953',\n",
              " '1955',\n",
              " '1956',\n",
              " '1961',\n",
              " '1965',\n",
              " '1966',\n",
              " '1967',\n",
              " '1968',\n",
              " '1969',\n",
              " '1970',\n",
              " '1971',\n",
              " '1972',\n",
              " '1973',\n",
              " '1975',\n",
              " '1976',\n",
              " '1977',\n",
              " '1979',\n",
              " '1980',\n",
              " '1981',\n",
              " '1982',\n",
              " '1983',\n",
              " '1984',\n",
              " '1985',\n",
              " '1986',\n",
              " '1987',\n",
              " '1988',\n",
              " '1989',\n",
              " '1990',\n",
              " '1991',\n",
              " '1992',\n",
              " '1993',\n",
              " '1994',\n",
              " '1995',\n",
              " '1996',\n",
              " '1997',\n",
              " '1998',\n",
              " '1999',\n",
              " '2000',\n",
              " '2005',\n",
              " '2009',\n",
              " '2017',\n",
              " '2019',\n",
              " '2029',\n",
              " '3057',\n",
              " '8300']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "4evd5TsK3xbS",
        "colab_type": "code",
        "outputId": "85c340c4-f6a6-4121-8eca-e9618f726f14",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 596
        }
      },
      "source": [
        "[w for w in wsj if re.search('^[0-9]+-[a-z]{3,5}$',w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['10-day',\n",
              " '10-lap',\n",
              " '10-year',\n",
              " '100-share',\n",
              " '12-point',\n",
              " '12-year',\n",
              " '14-hour',\n",
              " '15-day',\n",
              " '150-point',\n",
              " '190-point',\n",
              " '20-point',\n",
              " '20-stock',\n",
              " '21-month',\n",
              " '237-seat',\n",
              " '240-page',\n",
              " '27-year',\n",
              " '30-day',\n",
              " '30-point',\n",
              " '30-share',\n",
              " '30-year',\n",
              " '300-day',\n",
              " '36-day',\n",
              " '36-store',\n",
              " '42-year',\n",
              " '50-state',\n",
              " '500-stock',\n",
              " '52-week',\n",
              " '69-point',\n",
              " '84-month',\n",
              " '87-store',\n",
              " '90-day']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "k_ecD6QG4Aq5",
        "colab_type": "code",
        "outputId": "021c4728-256c-421c-dad3-358023be2e0a",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 111
        }
      },
      "source": [
        "[w for w in wsj if re.search('^[a-z]{5,}-[a-z]{2,3}-[a-z]{,6}$',w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['black-and-white',\n",
              " 'bread-and-butter',\n",
              " 'father-in-law',\n",
              " 'machine-gun-toting',\n",
              " 'savings-and-loan']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 19
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "65bAVp2G4TA8",
        "colab_type": "code",
        "outputId": "01d5a349-3a29-4034-d3d5-fc0998db13f5",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "[w for w in wsj if re.search('(ed|ing)$',w)]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['62%-owned',\n",
              " 'Absorbed',\n",
              " 'According',\n",
              " 'Adopting',\n",
              " 'Advanced',\n",
              " 'Advancing',\n",
              " 'Alfred',\n",
              " 'Allied',\n",
              " 'Annualized',\n",
              " 'Anything',\n",
              " 'Arbitrage-related',\n",
              " 'Arbitraging',\n",
              " 'Asked',\n",
              " 'Assuming',\n",
              " 'Atlanta-based',\n",
              " 'Baking',\n",
              " 'Banking',\n",
              " 'Beginning',\n",
              " 'Beijing',\n",
              " 'Being',\n",
              " 'Bermuda-based',\n",
              " 'Betting',\n",
              " 'Boeing',\n",
              " 'Broadcasting',\n",
              " 'Bucking',\n",
              " 'Buying',\n",
              " 'Calif.-based',\n",
              " 'Change-ringing',\n",
              " 'Citing',\n",
              " 'Concerned',\n",
              " 'Confronted',\n",
              " 'Conn.based',\n",
              " 'Consolidated',\n",
              " 'Continued',\n",
              " 'Continuing',\n",
              " 'Declining',\n",
              " 'Defending',\n",
              " 'Depending',\n",
              " 'Designated',\n",
              " 'Determining',\n",
              " 'Developed',\n",
              " 'Died',\n",
              " 'During',\n",
              " 'Encouraged',\n",
              " 'Encouraging',\n",
              " 'English-speaking',\n",
              " 'Estimated',\n",
              " 'Everything',\n",
              " 'Excluding',\n",
              " 'Exxon-owned',\n",
              " 'Faulding',\n",
              " 'Fed',\n",
              " 'Feeding',\n",
              " 'Filling',\n",
              " 'Filmed',\n",
              " 'Financing',\n",
              " 'Following',\n",
              " 'Founded',\n",
              " 'Fracturing',\n",
              " 'Francisco-based',\n",
              " 'Fred',\n",
              " 'Funded',\n",
              " 'Funding',\n",
              " 'Generalized',\n",
              " 'Germany-based',\n",
              " 'Getting',\n",
              " 'Guaranteed',\n",
              " 'Having',\n",
              " 'Heating',\n",
              " 'Heightened',\n",
              " 'Holding',\n",
              " 'Housing',\n",
              " 'Illuminating',\n",
              " 'Indeed',\n",
              " 'Indexing',\n",
              " 'Irving',\n",
              " 'Jersey-based',\n",
              " 'Judging',\n",
              " 'Knowing',\n",
              " 'Learning',\n",
              " 'Legislating',\n",
              " 'Leming',\n",
              " 'Limited',\n",
              " 'London-based',\n",
              " 'Manfred',\n",
              " 'Manufacturing',\n",
              " 'Melamed',\n",
              " 'Miami-based',\n",
              " 'Mich.-based',\n",
              " 'Mining',\n",
              " 'Minneapolis-based',\n",
              " 'Mo.-based',\n",
              " 'Mortgage-Backed',\n",
              " 'Moving',\n",
              " 'Muzzling',\n",
              " 'N.J.-based',\n",
              " 'NBC-owned',\n",
              " 'NIH-appointed',\n",
              " 'Named',\n",
              " 'No-Smoking',\n",
              " 'Observing',\n",
              " 'Offering',\n",
              " 'Ohio-based',\n",
              " 'Orleans-based',\n",
              " 'Packaging',\n",
              " 'Performing',\n",
              " 'Philadelphia-based',\n",
              " 'Posted',\n",
              " 'Provided',\n",
              " 'Publishing',\n",
              " 'Purchasing',\n",
              " 'Rated',\n",
              " 'Reached',\n",
              " 'Red',\n",
              " 'Red-blooded',\n",
              " 'Reducing',\n",
              " 'Reed',\n",
              " 'Regarded',\n",
              " 'Rekindled',\n",
              " 'Related',\n",
              " 'Ringing',\n",
              " 'Rolling',\n",
              " 'Sacramento-based',\n",
              " 'Scoring',\n",
              " 'Seattle-based',\n",
              " 'Seed',\n",
              " 'Skilled',\n",
              " 'Smelting',\n",
              " 'Something',\n",
              " 'Spending',\n",
              " 'Standardized',\n",
              " 'Standing',\n",
              " 'Starting',\n",
              " 'Sterling',\n",
              " 'Taking',\n",
              " 'Texas-based',\n",
              " 'Toronto-based',\n",
              " 'Traded',\n",
              " 'Trading',\n",
              " 'Troubled',\n",
              " 'U.N.-supervised',\n",
              " 'U.S.-backed',\n",
              " 'United',\n",
              " 'Used',\n",
              " 'Varying',\n",
              " 'Washington-based',\n",
              " 'Whiting',\n",
              " 'Wilfred',\n",
              " 'Winning',\n",
              " 'Xiaoping',\n",
              " 'York-based',\n",
              " 'Zayed',\n",
              " 'abandoned',\n",
              " 'abating',\n",
              " 'abolishing',\n",
              " 'abortion-related',\n",
              " 'abounding',\n",
              " 'abridging',\n",
              " 'absorbed',\n",
              " 'acceded',\n",
              " 'accelerated',\n",
              " 'accepted',\n",
              " 'accepting',\n",
              " 'according',\n",
              " 'accounted',\n",
              " 'accounting',\n",
              " 'accrued',\n",
              " 'accumulated',\n",
              " 'accused',\n",
              " 'accusing',\n",
              " 'achieved',\n",
              " 'achieving',\n",
              " 'acknowledging',\n",
              " 'acquired',\n",
              " 'acquiring',\n",
              " 'acquisition-minded',\n",
              " 'acted',\n",
              " 'acting',\n",
              " 'adapted',\n",
              " 'adapting',\n",
              " 'added',\n",
              " 'adding',\n",
              " 'addressing',\n",
              " 'adjusted',\n",
              " 'adjusting',\n",
              " 'admitted',\n",
              " 'admitting',\n",
              " 'adopted',\n",
              " 'advanced',\n",
              " 'advancing',\n",
              " 'advertised',\n",
              " 'advertising',\n",
              " 'advised',\n",
              " 'advocated',\n",
              " 'advocating',\n",
              " 'affecting',\n",
              " 'afflicted',\n",
              " 'aggravated',\n",
              " 'agreed',\n",
              " 'agreeing',\n",
              " 'ailing',\n",
              " 'aimed',\n",
              " 'aiming',\n",
              " 'aired',\n",
              " 'airline-related',\n",
              " 'alarmed',\n",
              " 'alienated',\n",
              " 'alleged',\n",
              " 'alleging',\n",
              " 'allocated',\n",
              " 'allowed',\n",
              " 'altered',\n",
              " 'altering',\n",
              " 'amended',\n",
              " 'amending',\n",
              " 'amounted',\n",
              " 'amusing',\n",
              " 'angered',\n",
              " 'announced',\n",
              " 'annoyed',\n",
              " 'annualized',\n",
              " 'answered',\n",
              " 'anti-dumping',\n",
              " 'anticipated',\n",
              " 'anticipating',\n",
              " 'anything',\n",
              " 'apologizing',\n",
              " 'appealing',\n",
              " 'appeared',\n",
              " 'appearing',\n",
              " 'applied',\n",
              " 'appointed',\n",
              " 'approached',\n",
              " 'appropriated',\n",
              " 'approved',\n",
              " 'arched',\n",
              " 'argued',\n",
              " 'arguing',\n",
              " 'arising',\n",
              " 'armed',\n",
              " 'arranged',\n",
              " 'arrested',\n",
              " 'arrived',\n",
              " 'asbestos-related',\n",
              " 'asked',\n",
              " 'asking',\n",
              " 'assassinated',\n",
              " 'assembled',\n",
              " 'asserted',\n",
              " 'asserting',\n",
              " 'assessed',\n",
              " 'assigned',\n",
              " 'assisted',\n",
              " 'associated',\n",
              " 'assumed',\n",
              " 'assuming',\n",
              " 'assured',\n",
              " 'attached',\n",
              " 'attacking',\n",
              " 'attempted',\n",
              " 'attempting',\n",
              " 'attended',\n",
              " 'attending',\n",
              " 'attracted',\n",
              " 'attracting',\n",
              " 'attributed',\n",
              " 'auctioned',\n",
              " 'authorized',\n",
              " 'authorizing',\n",
              " 'automated',\n",
              " 'automotive-lighting',\n",
              " 'averaged',\n",
              " 'averted',\n",
              " 'avoiding',\n",
              " 'awarded',\n",
              " 'awarding',\n",
              " 'backed',\n",
              " 'backing',\n",
              " 'balanced',\n",
              " 'bald-faced',\n",
              " 'balkanized',\n",
              " 'balked',\n",
              " 'balloting',\n",
              " 'bank-backed',\n",
              " 'banking',\n",
              " 'banned',\n",
              " 'banning',\n",
              " 'barking',\n",
              " 'barred',\n",
              " 'based',\n",
              " 'battered',\n",
              " 'battery-operated',\n",
              " 'batting',\n",
              " 'bearing',\n",
              " 'becoming',\n",
              " 'bedding',\n",
              " 'befuddled',\n",
              " 'beginning',\n",
              " 'behaving',\n",
              " 'beheading',\n",
              " 'being',\n",
              " 'beleaguered',\n",
              " 'believed',\n",
              " 'bell-ringing',\n",
              " 'belonging',\n",
              " 'benefited',\n",
              " 'best-selling',\n",
              " 'betting',\n",
              " 'bickering',\n",
              " 'bidding',\n",
              " 'billed',\n",
              " 'billing',\n",
              " 'blamed',\n",
              " 'bled',\n",
              " 'blessing',\n",
              " 'blighted',\n",
              " 'blocked',\n",
              " 'blurred',\n",
              " 'boarding',\n",
              " 'bolstered',\n",
              " 'bombarding',\n",
              " 'booked',\n",
              " 'booming',\n",
              " 'boosted',\n",
              " 'boosting',\n",
              " 'borrowed',\n",
              " 'borrowing',\n",
              " 'botched',\n",
              " 'bothered',\n",
              " 'bounced',\n",
              " 'bowed',\n",
              " 'breaking',\n",
              " 'breathed',\n",
              " 'breathtaking',\n",
              " 'breed',\n",
              " 'bribed',\n",
              " 'bribing',\n",
              " 'briefing',\n",
              " 'brightened',\n",
              " 'bring',\n",
              " 'bringing',\n",
              " 'broad-based',\n",
              " 'broadcasting',\n",
              " 'broadened',\n",
              " 'brokering',\n",
              " 'brushed',\n",
              " 'budding',\n",
              " 'building',\n",
              " 'bundling',\n",
              " 'buoyed',\n",
              " 'burned',\n",
              " 'buying',\n",
              " 'calculated',\n",
              " 'called',\n",
              " 'calling',\n",
              " 'campaigning',\n",
              " 'cancer-causing',\n",
              " 'capitalized',\n",
              " 'capped',\n",
              " 'captivating',\n",
              " 'cared',\n",
              " 'carried',\n",
              " 'carrying',\n",
              " 'cascading',\n",
              " 'casting',\n",
              " 'caused',\n",
              " 'causing',\n",
              " 'cautioned',\n",
              " 'ceiling',\n",
              " 'centralized',\n",
              " 'certified',\n",
              " 'chaired',\n",
              " 'challenging',\n",
              " 'championing',\n",
              " 'change-ringing',\n",
              " 'changed',\n",
              " 'changing',\n",
              " 'characterized',\n",
              " 'characterizing',\n",
              " 'charged',\n",
              " 'charging',\n",
              " 'chastised',\n",
              " 'cheating',\n",
              " 'checking',\n",
              " 'cheerleading',\n",
              " 'chilled',\n",
              " 'choosing',\n",
              " 'chopped',\n",
              " 'circulated',\n",
              " 'cited',\n",
              " 'citing',\n",
              " 'citizen-sparked',\n",
              " 'city-owned',\n",
              " 'claimed',\n",
              " 'claiming',\n",
              " 'clamped',\n",
              " 'clarified',\n",
              " 'clashed',\n",
              " 'classed',\n",
              " 'classified',\n",
              " 'cleaned',\n",
              " 'cleaner-burning',\n",
              " 'cleared',\n",
              " 'clearing',\n",
              " 'clicked',\n",
              " 'climbed',\n",
              " 'climbing',\n",
              " 'clipped',\n",
              " 'clobbered',\n",
              " 'closed',\n",
              " 'closing',\n",
              " 'clothing',\n",
              " 'clouding',\n",
              " 'cluttered',\n",
              " 'co-founded',\n",
              " 'coaching',\n",
              " 'coal-fired',\n",
              " 'coated',\n",
              " 'codified',\n",
              " 'collaborated',\n",
              " 'collapsed',\n",
              " 'collected',\n",
              " 'collecting',\n",
              " 'collective-bargaining',\n",
              " 'colored',\n",
              " 'combined',\n",
              " 'coming',\n",
              " 'commanded',\n",
              " 'commenting',\n",
              " 'committed',\n",
              " 'committing',\n",
              " 'compared',\n",
              " 'compelling',\n",
              " 'competed',\n",
              " 'competing',\n",
              " 'compiled',\n",
              " 'complained',\n",
              " 'complaining',\n",
              " 'completed',\n",
              " 'completing',\n",
              " 'complicated',\n",
              " 'composed',\n",
              " 'composting',\n",
              " 'compressed',\n",
              " 'computer-aided',\n",
              " 'computer-assisted',\n",
              " 'computer-generated',\n",
              " 'computerized',\n",
              " 'computing',\n",
              " 'conceding',\n",
              " 'concentrated',\n",
              " 'concentrating',\n",
              " 'concerned',\n",
              " 'concluded',\n",
              " 'condemned',\n",
              " 'condemning',\n",
              " 'conducted',\n",
              " 'conducting',\n",
              " 'confined',\n",
              " 'confirmed',\n",
              " 'confused',\n",
              " 'connected',\n",
              " 'consented',\n",
              " 'considered',\n",
              " 'considering',\n",
              " 'consisting',\n",
              " 'construed',\n",
              " 'consulting',\n",
              " 'contacted',\n",
              " 'contained',\n",
              " 'containing',\n",
              " 'contesting',\n",
              " 'continued',\n",
              " 'continuing',\n",
              " 'contracted',\n",
              " 'contributed',\n",
              " 'contributing',\n",
              " 'controlled',\n",
              " 'controlling',\n",
              " 'converted',\n",
              " 'converting',\n",
              " 'convicted',\n",
              " 'convinced',\n",
              " 'cooled',\n",
              " 'cooperating',\n",
              " 'copied',\n",
              " 'copying',\n",
              " 'corn-buying',\n",
              " 'corrected',\n",
              " 'correcting',\n",
              " 'cost-cutting',\n",
              " 'cost-sharing',\n",
              " 'counseling',\n",
              " 'counting',\n",
              " 'coupled',\n",
              " 'court-ordered',\n",
              " 'covered',\n",
              " 'covering',\n",
              " 'cranked',\n",
              " 'crashing',\n",
              " 'created',\n",
              " 'creating',\n",
              " 'credit-rating',\n",
              " 'crippled',\n",
              " 'criticized',\n",
              " 'crossed',\n",
              " 'crossing',\n",
              " 'crowded',\n",
              " 'cruising',\n",
              " 'crushed',\n",
              " 'crying',\n",
              " 'cultivated',\n",
              " 'curbed',\n",
              " 'curbing',\n",
              " 'curled',\n",
              " 'current-carrying',\n",
              " 'curtailed',\n",
              " 'cushioned',\n",
              " 'customized',\n",
              " 'cutting',\n",
              " 'damaged',\n",
              " 'damaging',\n",
              " 'dancing',\n",
              " 'darned',\n",
              " 'dashed',\n",
              " 'dating',\n",
              " 'dead-eyed',\n",
              " 'dealing',\n",
              " 'decided',\n",
              " 'declared',\n",
              " 'declaring',\n",
              " 'declined',\n",
              " 'declining',\n",
              " 'decorated',\n",
              " 'decried',\n",
              " 'deducting',\n",
              " 'deemed',\n",
              " 'defeated',\n",
              " 'defended',\n",
              " 'defined',\n",
              " 'defying',\n",
              " 'delayed',\n",
              " 'deliberating',\n",
              " 'delisted',\n",
              " 'delivered',\n",
              " 'delivering',\n",
              " 'demanding',\n",
              " 'demonstrating',\n",
              " 'denied',\n",
              " 'denouncing',\n",
              " 'denying',\n",
              " 'depended',\n",
              " 'depending',\n",
              " 'depleted',\n",
              " 'depressed',\n",
              " 'deprived',\n",
              " 'derived',\n",
              " 'descending',\n",
              " 'described',\n",
              " 'deserving',\n",
              " 'designated',\n",
              " 'designed',\n",
              " 'designing',\n",
              " 'desired',\n",
              " 'despised',\n",
              " 'detailed',\n",
              " 'deteriorated',\n",
              " 'deteriorating',\n",
              " 'determined',\n",
              " 'deterring',\n",
              " 'devastating',\n",
              " 'developed',\n",
              " 'developing',\n",
              " 'devised',\n",
              " 'devoted',\n",
              " 'devouring',\n",
              " 'diagnosed',\n",
              " 'died',\n",
              " 'diluted',\n",
              " 'diming',\n",
              " 'diminished',\n",
              " 'directed',\n",
              " 'directing',\n",
              " 'disaffected',\n",
              " 'disagreed',\n",
              " 'disappointed',\n",
              " 'disappointing',\n",
              " 'disapproved',\n",
              " 'discarded',\n",
              " 'disciplined',\n",
              " 'disclosed',\n",
              " 'disclosing',\n",
              " 'discontinued',\n",
              " 'discontinuing',\n",
              " 'discouraging',\n",
              " 'discovered',\n",
              " 'discussed',\n",
              " 'discussing',\n",
              " 'disembodied',\n",
              " 'dismayed',\n",
              " 'dismissed',\n",
              " 'disposed',\n",
              " 'disputed',\n",
              " 'disseminating',\n",
              " 'distinguished',\n",
              " 'distorted',\n",
              " 'distributed',\n",
              " 'disturbing',\n",
              " 'diversified',\n",
              " 'diversifying',\n",
              " 'divided',\n",
              " 'dividing',\n",
              " 'documented',\n",
              " 'doing',\n",
              " 'doling',\n",
              " 'dollar-denominated',\n",
              " 'dominated',\n",
              " 'dominating',\n",
              " 'doubled',\n",
              " 'doubted',\n",
              " 'downgraded',\n",
              " 'downgrading',\n",
              " 'drafted',\n",
              " 'drawing',\n",
              " 'dreamed',\n",
              " 'dressed',\n",
              " 'drifted',\n",
              " 'drinking',\n",
              " 'driving',\n",
              " 'drooled',\n",
              " 'dropped',\n",
              " 'dubbed',\n",
              " 'duckling',\n",
              " 'dumbfounded',\n",
              " 'dumped',\n",
              " 'during',\n",
              " 'dwindling',\n",
              " 'earned',\n",
              " 'earning',\n",
              " 'eased',\n",
              " 'easing',\n",
              " 'eating',\n",
              " 'echoed',\n",
              " 'edged',\n",
              " 'editing',\n",
              " 'educated',\n",
              " 'elected',\n",
              " 'eliminated',\n",
              " 'eliminating',\n",
              " 'embarrassing',\n",
              " 'embroiled',\n",
              " 'emerged',\n",
              " 'emerging',\n",
              " 'emphasized',\n",
              " 'employed',\n",
              " 'empowered',\n",
              " 'enabled',\n",
              " 'enabling',\n",
              " 'enacted',\n",
              " 'encircling',\n",
              " 'enclosed',\n",
              " 'encouraging',\n",
              " 'encroaching',\n",
              " 'ended',\n",
              " 'ending',\n",
              " 'endorsed',\n",
              " 'engaged',\n",
              " 'engaging',\n",
              " 'engineered',\n",
              " 'engineering',\n",
              " 'enhanced',\n",
              " 'enjoyed',\n",
              " 'enjoying',\n",
              " 'enlarged',\n",
              " 'enraged',\n",
              " 'ensnarled',\n",
              " 'entangled',\n",
              " 'entered',\n",
              " 'entering',\n",
              " 'entertaining',\n",
              " 'enticed',\n",
              " 'entitled',\n",
              " 'entrenched',\n",
              " 'entrusted',\n",
              " 'equaling',\n",
              " 'equipped',\n",
              " 'escalated',\n",
              " 'escaped',\n",
              " 'established',\n",
              " 'establishing',\n",
              " 'estimated',\n",
              " 'evaluated',\n",
              " 'evaluating',\n",
              " 'evaporated',\n",
              " 'evening',\n",
              " 'everything',\n",
              " 'evoking',\n",
              " 'evolved',\n",
              " 'exacerbated',\n",
              " 'examined',\n",
              " 'exceed',\n",
              " 'exceeded',\n",
              " 'exceeding',\n",
              " 'exchanging',\n",
              " 'excited',\n",
              " 'exciting',\n",
              " 'executed',\n",
              " 'executing',\n",
              " 'exercised',\n",
              " 'exerting',\n",
              " 'exhausted',\n",
              " 'exhibited',\n",
              " 'existed',\n",
              " 'existing',\n",
              " 'expanded',\n",
              " 'expanding',\n",
              " 'expected',\n",
              " 'expecting',\n",
              " 'expedited',\n",
              " 'expelled',\n",
              " 'experienced',\n",
              " 'experiencing',\n",
              " 'expired',\n",
              " 'explained',\n",
              " 'explaining',\n",
              " 'exploded',\n",
              " 'export-oriented',\n",
              " 'exposed',\n",
              " 'expressed',\n",
              " 'expressing',\n",
              " 'expunged',\n",
              " 'extended',\n",
              " 'extending',\n",
              " 'exuded',\n",
              " 'eyeing',\n",
              " 'fabled',\n",
              " 'faced',\n",
              " 'facing',\n",
              " 'factoring',\n",
              " 'faded',\n",
              " 'failed',\n",
              " 'failing',\n",
              " 'fainting',\n",
              " 'falling',\n",
              " 'faltered',\n",
              " 'famed',\n",
              " 'family-planning',\n",
              " 'fared',\n",
              " 'fashioned',\n",
              " 'fast-growing',\n",
              " 'fastest-growing',\n",
              " 'fattened',\n",
              " 'favored',\n",
              " 'fawning',\n",
              " 'feared',\n",
              " 'featured',\n",
              " 'featuring',\n",
              " 'fed',\n",
              " 'feed',\n",
              " 'feeling',\n",
              " 'fetching',\n",
              " 'fielded',\n",
              " 'fighting',\n",
              " 'filed',\n",
              " 'filing',\n",
              " 'filled',\n",
              " 'filling',\n",
              " 'finalized',\n",
              " 'financed',\n",
              " 'financing',\n",
              " 'finding',\n",
              " 'fined',\n",
              " 'finished',\n",
              " 'fired',\n",
              " 'firmed',\n",
              " 'fixed',\n",
              " 'fizzled',\n",
              " 'fled',\n",
              " 'fledgling',\n",
              " 'fleeting',\n",
              " 'flirted',\n",
              " 'floated',\n",
              " 'flooded',\n",
              " 'focused',\n",
              " 'focusing',\n",
              " 'folded',\n",
              " 'followed',\n",
              " 'following',\n",
              " 'forced',\n",
              " 'forcing',\n",
              " 'forecasting',\n",
              " 'foreign-led',\n",
              " 'formed',\n",
              " 'forthcoming',\n",
              " 'founded',\n",
              " 'foundering',\n",
              " 'fretted',\n",
              " 'frightened',\n",
              " 'frustrating',\n",
              " 'fueled',\n",
              " 'fueling',\n",
              " 'full-fledged',\n",
              " 'fuming',\n",
              " 'functioning',\n",
              " 'funded',\n",
              " 'funding',\n",
              " 'fundraising',\n",
              " 'futures-related',\n",
              " 'gained',\n",
              " 'gaining',\n",
              " 'galling',\n",
              " 'galvanized',\n",
              " 'gambling',\n",
              " 'gauging',\n",
              " 'generated',\n",
              " 'getting',\n",
              " 'giving',\n",
              " 'going',\n",
              " 'good-hearted',\n",
              " 'good-natured',\n",
              " 'gored',\n",
              " 'government-certified',\n",
              " 'government-funded',\n",
              " 'government-owned',\n",
              " 'graduated',\n",
              " 'granted',\n",
              " 'granting',\n",
              " 'greed',\n",
              " 'gripping',\n",
              " 'growing',\n",
              " 'guaranteed',\n",
              " 'guarding',\n",
              " 'guided',\n",
              " 'gut-wrenching',\n",
              " 'hailed',\n",
              " 'hailing',\n",
              " 'halted',\n",
              " 'hampered',\n",
              " 'handed',\n",
              " 'handled',\n",
              " 'handling',\n",
              " 'happened',\n",
              " 'happening',\n",
              " 'hard-charging',\n",
              " 'hard-drinking',\n",
              " 'hard-hitting',\n",
              " 'harmed',\n",
              " 'harped',\n",
              " 'harvested',\n",
              " 'hauled',\n",
              " 'hauling',\n",
              " 'having',\n",
              " 'headed',\n",
              " 'heading',\n",
              " 'headlined',\n",
              " 'healing',\n",
              " 'hearing',\n",
              " 'heated',\n",
              " 'heating',\n",
              " 'hedging',\n",
              " 'heightened',\n",
              " 'helped',\n",
              " 'helping',\n",
              " 'high-flying',\n",
              " 'high-minded',\n",
              " 'high-polluting',\n",
              " 'high-priced',\n",
              " 'high-rolling',\n",
              " 'high-speed',\n",
              " 'higher-salaried',\n",
              " 'highest-pitched',\n",
              " 'hired',\n",
              " 'hitting',\n",
              " 'holding',\n",
              " 'hoped',\n",
              " 'hosted',\n",
              " 'housing',\n",
              " 'hugging',\n",
              " 'hundred',\n",
              " 'hunted',\n",
              " 'hurting',\n",
              " 'identified',\n",
              " 'ignored',\n",
              " 'ignoring',\n",
              " 'impaired',\n",
              " 'impeding',\n",
              " 'impending',\n",
              " 'implemented',\n",
              " 'implied',\n",
              " 'imported',\n",
              " 'imposed',\n",
              " 'imposing',\n",
              " 'impressed',\n",
              " 'improved',\n",
              " 'improving',\n",
              " 'incentive-backed',\n",
              " 'inched',\n",
              " 'inching',\n",
              " 'included',\n",
              " 'including',\n",
              " 'incorporated',\n",
              " 'increased',\n",
              " 'increasing',\n",
              " 'incurred',\n",
              " 'indeed',\n",
              " 'index-related',\n",
              " 'indicated',\n",
              " 'indicating',\n",
              " 'indulging',\n",
              " 'industrialized',\n",
              " 'industry-supported',\n",
              " 'inflated',\n",
              " 'influenced',\n",
              " 'influencing',\n",
              " 'infringed',\n",
              " 'inherited',\n",
              " 'initialing',\n",
              " 'initiated',\n",
              " 'initiating',\n",
              " 'injecting',\n",
              " 'injuring',\n",
              " 'inkling',\n",
              " 'inquiring',\n",
              " 'inserted',\n",
              " 'insider-trading',\n",
              " 'insinuating',\n",
              " 'insisted',\n",
              " 'inspired',\n",
              " 'installed',\n",
              " 'installing',\n",
              " 'instituted',\n",
              " 'instructed',\n",
              " 'insured',\n",
              " 'integrated',\n",
              " 'intended',\n",
              " 'intentioned',\n",
              " 'interest-bearing',\n",
              " 'interested',\n",
              " 'interesting',\n",
              " 'interrogated',\n",
              " 'interviewed',\n",
              " 'intriguing',\n",
              " 'introduced',\n",
              " 'introducing',\n",
              " 'invented',\n",
              " 'inverted',\n",
              " 'invested',\n",
              " 'investigating',\n",
              " 'investing',\n",
              " 'inviting',\n",
              " 'involved',\n",
              " 'involving',\n",
              " 'issued',\n",
              " 'issuing',\n",
              " 'jeopardizing',\n",
              " 'joined',\n",
              " 'joining',\n",
              " 'judged',\n",
              " 'jumped',\n",
              " 'jumping',\n",
              " 'justified',\n",
              " 'justifying',\n",
              " 'keeping',\n",
              " 'kicked',\n",
              " 'kidnapping',\n",
              " 'killed',\n",
              " 'killing',\n",
              " 'knitted',\n",
              " 'knocked',\n",
              " 'labeled',\n",
              " 'labeling',\n",
              " 'labor-backed',\n",
              " 'lacked',\n",
              " 'lagging',\n",
              " 'land-idling',\n",
              " 'landing',\n",
              " 'lasted',\n",
              " 'lasting',\n",
              " 'lauded',\n",
              " 'laughing',\n",
              " 'launched',\n",
              " 'lawmaking',\n",
              " 'laying',\n",
              " 'leading',\n",
              " 'learned',\n",
              " 'learning',\n",
              " 'leasing',\n",
              " 'leaving',\n",
              " 'led',\n",
              " 'lending',\n",
              " 'lengthened',\n",
              " 'lessening',\n",
              " 'letter-writing',\n",
              " 'letting',\n",
              " 'leveling',\n",
              " 'leveraged',\n",
              " 'leveraging',\n",
              " 'licensed',\n",
              " 'licensing',\n",
              " 'lifted',\n",
              " ...]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 20
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "sBXXsvQV4g5s",
        "colab_type": "code",
        "outputId": "62a03419-ef58-45bb-b305-122270a43102",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 316
        }
      },
      "source": [
        "word = 'supercalifragilisticexpialidocious'\n",
        "re.findall(r'[aeiou]', word)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['u',\n",
              " 'e',\n",
              " 'a',\n",
              " 'i',\n",
              " 'a',\n",
              " 'i',\n",
              " 'i',\n",
              " 'i',\n",
              " 'e',\n",
              " 'i',\n",
              " 'a',\n",
              " 'i',\n",
              " 'o',\n",
              " 'i',\n",
              " 'o',\n",
              " 'u']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "c4a3TxM643vV",
        "colab_type": "code",
        "outputId": "f86abac0-d9c0-4e03-d9d5-5b4434d9cd59",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(re.findall(r'[aeiou]', word))"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "16"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "MtsNImon48SG",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "# GNI 논문 각각의 책임저자 이메일 주소 출력 코드\n",
        "from nltk.corpus import *\n",
        "GNI = PlaintextCorpusReader('/content/gdrive/My Drive/인공지능 수업 데이터 파일/Genomics-Informatics-Corpus-master/GNI Corpus 1.0/raw/', '.*\\.txt', encoding='utf-8')"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "hYAaWef46ANB",
        "colab_type": "code",
        "outputId": "03f980e3-b861-498a-cebd-8b2a59218c59",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "for file in GNI.fileids():\n",
        "  raw = GNI.raw(file)\n",
        "  print(re.findall(r'[\\w._%+-]+@[\\w.-]+\\.[a-zA-Z]{2,4}',raw))"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "['yejun@catholic.ac.kr']\n",
            "['bluelabelman81@gmail.com']\n",
            "['yejun@catholic.ac.kr']\n",
            "['kimty@snu.ac.kr']\n",
            "['schoi@kangwon.ac.kr']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['yejun@catholic.ac.kr']\n",
            "['MinhoLee@catholic.ac.kr']\n",
            "['MinhoLee@catholic.ac.kr']\n",
            "['MinhoLee@catholic.ac.kr']\n",
            "['sureshkumar@msu.edu.my']\n",
            "['Mohamad.hajari@gmail.com', 'm-hajari@scu.ac.ir']\n",
            "['sskimb@ssu.ac.krTh']\n",
            "['jongil@snu.ac.kr']\n",
            "['lim.dj@korea.krTh']\n",
            "['sdh1214@gmail.com']\n",
            "['o@biophilos.org']\n",
            "['jslee2@hanyang.ac.krTe', 'choii@snu.ac.krTe']\n",
            "['yooncho33@hallym.ac.kr']\n",
            "['ynkim@dsmc.or.kr']\n",
            "['jwpark@hallym.ac.kr']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['o@biophilos.org']\n",
            "['xushua@picb.ac.cn']\n",
            "['tyroh@postech.edu']\n",
            "['sangdunchoi@ajou.ac.kr']\n",
            "['parkjh@sookmyung.ac.kr']\n",
            "['yhk0215@korea.ac.kr']\n",
            "['jsunha@yuhs.ac']\n",
            "['cij1224@ncc.re.kr', 'yslee2@ncc.re.kr']\n",
            "['yejun@catholic.ac.kr']\n",
            "['kwanskim@chungbuk.ac.kr']\n",
            "['paekwk@mest.go.kr', 'jeongheuilim@gmail.comT']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['murim.choi@yale.edu']\n",
            "['sikegawa@ims.u-tokyo.ac.jp']\n",
            "['jim97@dankook.ac.kr']\n",
            "['cha8764@yuhs.acTe']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['kspark@snu.ac.kr']\n",
            "['ohbs@khu.ac.kr']\n",
            "['sangdunchoi@ajou.ac.krTe', 'kjvenkat@jpl.nasa.gov']\n",
            "['tspark@stats.snu.ac.kr']\n",
            "['yoonsj@sookmyung.ac.kr']\n",
            "['neo@ewha.ac.kr']\n",
            "['neo@ewha.ac.kr']\n",
            "['dhlee@biosoft.kaist.ac.krTe', 'yongsung@kribb.re.krTe']\n",
            "['ohbs@khu.ac.krTe']\n",
            "['hojinc@kaist.ac.krTe']\n",
            "['hojinc@kaist.ac.krTe']\n",
            "['hgcho@pusan.ac.krTe']\n",
            "['yejun@catholic.ac.krTe', 'yejun@catholic.ac.kr']\n",
            "['hg@ajou.ac.kr']\n",
            "['khs307@pusan.ac.kr']\n",
            "['hhsong@catholic.ac.kr', 'yejun@catholic.ac.kr']\n",
            "['kbkwack@cha.ac.kr']\n",
            "['sskimb@ssu.ac.krTe']\n",
            "['leejy63@nih.go.kr']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['hanayi@korea.ac.kr']\n",
            "['seonlee@dau.ac.kr']\n",
            "['baejw@khu.ac.krMi']\n",
            "['kimye@korea.kr']\n",
            "['sskmb@ssu.ac.kr']\n",
            "['khs307@pusan.ac.kr']\n",
            "['kbj6181@cdc.go.kr', 'leejy63@nih.go.kr']\n",
            "['kbkwack@cha.ac.kr']\n",
            "['hss2003@snu.ac.kr']\n",
            "[]\n",
            "['sskimb@ssu.ac.kr']\n",
            "['jsjlee@ajou.ac.kr']\n",
            "['sehoon.lee119@gmail.com']\n",
            "['jlee@mdanderson.org']\n",
            "['yejun@catholic.ac.kr']\n",
            "['xing@biology.rutgers.edu']\n",
            "['jungkyoon@kaist.ac.kr']\n",
            "['TaeHyun.Hwang@UTSouthwestern.edu']\n",
            "['bhkang11@korea.kr']\n",
            "['lyra@catholic.ac.kr', 'yejun@catholic.ac.kr', 'lyra@catholic.ac.kr', 'yejun@catholic.ac.krCo']\n",
            "['parkjh@sookmyung.ac.kr']\n",
            "['pliang@brocku.ca']\n",
            "['yoonsj@sookmyung.ac.kr']\n",
            "['deepreds@kribb.re.krTh']\n",
            "['gobrian@kcch.re.kr']\n",
            "['jihyni@yuhs.ac']\n",
            "['mlee@sungshin.ac.kr']\n",
            "['ahnty@dankook.ac.krPr']\n",
            "['light@spo.go.kr', 'kimsy@kribb.re.krJo']\n",
            "['o@biophilos.org']\n",
            "['satishangral@gmail.com']\n",
            "['ssysmc@snu.ac.kr']\n",
            "['juhan@snu.ac.kr']\n",
            "['jwpark@hallym.ac.kr']\n",
            "['yonglee@snu.ac.kr']\n",
            "['Hyun.Kim@UTsouthwestern.edu']\n",
            "[]\n",
            "['sskimb@ssu.ac.kr']\n",
            "['wonk@mail.med.upenn.edu']\n",
            "['jieunlee@ncc.re.kr']\n",
            "['jbbae72@gmail.com']\n",
            "['bcho@kaist.ac.kr']\n",
            "['jongil@snu.ac.kr', 'jeongsun@snu.ac.krHy']\n",
            "['ohbs@khu.ac.kr']\n",
            "[]\n",
            "[]\n",
            "[]\n",
            "['yejun@catholic.ac.kr']\n",
            "['jkim@lsu.edu']\n",
            "['tyroh@postech.edu']\n",
            "['semikim@kribb.re.kr', 'jwl@snu.ac.kr']\n",
            "['wonki@dsmc.or.kr']\n",
            "['sejongoh@dankook.ac.kr']\n",
            "['yejun@catholic.ac.kreI']\n",
            "['zhangfeng@fudan.edu.cn']\n",
            "['neo@ewha.ac.kr']\n",
            "['lyra@catholic.ac.kr', 'yejun@catholic.ac.kr']\n",
            "['elusandy@gmail.com']\n",
            "['sskimb@ssu.ac.krTh']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['jungkyoon@kaist.ac.krTh']\n",
            "['zknightm@gmail.comT']\n",
            "['hyung@ewha.ac.krTh']\n",
            "['ssarda@umiacs.umd.edu', 'Sridhar@umiacs.umd.edu', 'ssarda@umiacs.umd.edu', 'Sridhar@umiacs.umd.eduC']\n",
            "['sejongoh@dankook.ac.krTh']\n",
            "['lypark@yonsei.ac.krTh']\n",
            "['Q.Nguyen@uws.edu.au']\n",
            "['kasohn@ajou.ac.krTh']\n",
            "['kbj6181@cdc.go.krTh']\n",
            "['jsunha@yuhs.ac']\n",
            "['sbcho@korea.kr']\n",
            "['shkwak@snu.ac.kr']\n",
            "['jongil@snu.ac.kr']\n",
            "['heebal@snu.ac.kr']\n",
            "['khs307@pusan.ac.krTh']\n",
            "['anayet_johny@yahoo.com']\n",
            "['satishangral@gmail.com']\n",
            "['snrbioinfo@gmail.com']\n",
            "['yejun@catholic.ac.krTh']\n",
            "['bckang@insilicogen.com']\n",
            "['joy@kribb.re.kr', 'bulee@kribb.re.krAl', 'joy@kribb.re.kr', 'bulee@kribb.re.kr']\n",
            "['yejun@catholic.ac.kreI']\n",
            "['bckim00@gmail.com']\n",
            "['schalr@schbc.ac.kr', 'hdshin@sogang.ac.krTh']\n",
            "['satishangral@gmail.com']\n",
            "['dsloy2003@gmail.comT']\n",
            "['axxa76@kribb.re.kr']\n",
            "['yejun@catholic.ac.kreI']\n",
            "['kimbd@snu.ac.kr']\n",
            "['kimnamsu@kangwon.ac.kr']\n",
            "['khs307@pusan.ac.kr']\n",
            "['yejun@catholic.ac.kr']\n",
            "['kornsorn.s@ku.ac.th']\n",
            "['schoi@kangwon.ac.kr']\n",
            "['swkim@yuhs.ac']\n",
            "['kspark@snu.ac.kr']\n",
            "['schoi@kangwon.ac.kr']\n",
            "['sks@scnu.kr', 'heebal@snu.ac.kr']\n",
            "['heebal@snu.ac.kr', 'ckh1219@korea.kr']\n",
            "['drvus@snmail.org', 'dipa.bits@gmail.com']\n",
            "['neo@ewha.ac.kr']\n",
            "[]\n",
            "['12091931@dankook.ac.kr']\n",
            "['yejun@catholic.ac.kr']\n",
            "['yslee@utmb.edu']\n",
            "['veritas@ajou.ac.kr', 'n@rna.krTh']\n",
            "['jungkyoon@kaist.ac.kr']\n",
            "['snrbioinfo@gmail.com']\n",
            "['anayet_johny@yahoo.com']\n",
            "['satishangral@gmail.com']\n",
            "['yejun@catholic.ac.kr']\n",
            "['tahmoores@um.ac.ir']\n",
            "['yongsung@kribb.re.kr']\n",
            "['sskimb@ssu.ac.krTh']\n",
            "['thkwon@jbnu.ac.kr', 'choii@snu.ac.kr']\n",
            "['chu@kribb.re.kr']\n",
            "['axxa76@nie.re.kr']\n",
            "['yejun@catholic.ac.kr']\n",
            "['yslee@utmb.edu', 'xibao@utmb.edu']\n",
            "['yejun@catholic.ac.kr']\n",
            "['satishangral@gmail.com']\n",
            "['snrbioinfo@gmail.com']\n",
            "['jelee9137@kist.re.kr']\n",
            "['mihassan@jmi.ac.inTh']\n",
            "['tspark@stats.snu.ac.kr']\n",
            "['tspark@stats.snu.ac.kr']\n",
            "['tspark@stats.snu.ac.kr']\n",
            "['won1@snu.ac.kr']\n",
            "['leesy@sejong.ac.kr']\n",
            "['buhm.han@amc.seoul.kr']\n",
            "['mira@eulji.ac.kr']\n",
            "['hsun@pusan.ac.kr']\n",
            "['yyoo@snu.ac.kr']\n",
            "['dhhwang@postech.ac.kr']\n",
            "['iychoi@catholic.ac.kr']\n",
            "['ksy@ncc.re.kr']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['kbj6181@cdc.go.krTh']\n",
            "['djjang@kfri.re.kr']\n",
            "['aditi504@korea.kr', 'variance@korea.kr']\n",
            "['Mohamad.hajari@gmail.com', 'm-hajari@scu.ac.ir', 'tahmasebi-ma@ajums.ac.irMo']\n",
            "['snrbioinfo@gmail.com']\n",
            "['sm.shahik@gmail.com']\n",
            "['sskimb@ssu.ac.kr']\n",
            "['kimsy@kribb.re.kr']\n",
            "['yejun@catholic.ac.kr']\n",
            "['murimchoi@snu.ac.kr']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n",
            "['yejun@catholic.ac.krTe']\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "6-fSc3Mk6i22",
        "colab_type": "code",
        "outputId": "9fe5f11c-823f-4929-aaa2-13533f3faad9",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 56
        }
      },
      "source": [
        "# GNI 단어들 중 모음이 두 개 연이어 나오는 경우의 단어 빈도수 측정\n",
        "words = sorted(set(GNI.words()))\n",
        "fd = nltk.FreqDist(x for word in words for x in re.findall(r'[aeiou]{2,}', word))\n",
        "fd.items()"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "dict_items([('ei', 344), ('ioi', 25), ('ea', 663), ('io', 1660), ('ou', 413), ('ua', 211), ('ia', 651), ('ie', 545), ('oo', 251), ('ee', 276), ('ui', 148), ('iu', 61), ('au', 149), ('ai', 300), ('eae', 14), ('ae', 77), ('eo', 239), ('eu', 169), ('oi', 176), ('ue', 190), ('oa', 92), ('aeo', 4), ('iai', 1), ('aa', 11), ('ao', 15), ('eau', 5), ('oui', 3), ('ioa', 7), ('ioe', 11), ('oe', 43), ('ii', 18), ('uou', 13), ('eou', 19), ('uu', 4), ('eiou', 2), ('uo', 23), ('aue', 4), ('eui', 3), ('oeu', 2), ('iou', 33), ('eio', 8), ('ieu', 2), ('ieae', 1), ('aie', 1), ('eie', 1), ('oai', 1), ('oea', 1), ('iao', 3), ('eoae', 1), ('ooi', 1), ('aea', 2), ('oua', 1), ('eea', 3), ('eei', 3), ('uaeo', 1), ('ueou', 1), ('iae', 2), ('oau', 3), ('oii', 1), ('oie', 4), ('oue', 1), ('iii', 1), ('oae', 1), ('eoa', 1), ('aeu', 1), ('uai', 1), ('uie', 2), ('iau', 1), ('uee', 1)])"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "V4H9uJLi7Be2",
        "colab_type": "code",
        "outputId": "425eaee4-0e5b-41e1-a097-28a64b1bdbe1",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 148
        }
      },
      "source": [
        "#rotokas 언어 조사\n",
        "rotokas_words = nltk.corpus.toolbox.words('rotokas.dic')\n",
        "cvs = [cv for w in rotokas_words for cv in re.findall(r'[ptksvr][aeiou]',w)]\n",
        "cfd = nltk.ConditionalFreqDist(cvs)\n",
        "cfd.tabulate() #sa, se, ti 발음이 전혀 없다는 것을 알 수 있음"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "    a   e   i   o   u \n",
            "k 418 148  94 420 173 \n",
            "p  83  31 105  34  51 \n",
            "r 187  63  84  89  79 \n",
            "s   0   0 100   2   1 \n",
            "t  47   8   0 148  37 \n",
            "v  93  27 105  48  49 \n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "bdPXkRhg75f1",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "def stem(word):\n",
        "  for suffix in ['ing', 'ly', 'ed', 'ious', 'ies', 'ive', 'es', 's', 'ment']:\n",
        "    if word.endswith(suffix):\n",
        "      return word[:-len(suffix)]\n",
        "  return word"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "8mbUK9878g87",
        "colab_type": "code",
        "outputId": "5cd8bbf2-8acc-40ff-dc24-908590e57aaa",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 92
        }
      },
      "source": [
        "from nltk.corpus import gutenberg, nps_chat #re.findall과는 다름\n",
        "moby = nltk.Text(gutenberg.words('melville-moby_dick.txt'))\n",
        "moby.findall(r'<a> (<.*>) <man>') # a ~ man"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "monied; nervous; dangerous; white; white; white; pious; queer; good;\n",
            "mature; white; Cape; great; wise; wise; butterless; white; fiendish;\n",
            "pale; furious; better; certain; complete; dismasted; younger; brave;\n",
            "brave; brave; brave\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "QSkTcirD9eAT",
        "colab_type": "code",
        "outputId": "df4e6e30-50e2-4695-c121-b039d20a240f",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "chat = nltk.Text(nps_chat.words())\n",
        "chat.findall(r'<.*> <.*> <bro>') # bro 앞의 두 단어 찾아줌 -> 괄호를 안쳤기 때문에 bro 까지 프린트"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "you rule bro; telling you bro; u twizted bro\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "AwMpM5h6-Abk",
        "colab_type": "code",
        "outputId": "fb5f5a73-2e5a-483c-8b0a-362dc5a4fd8a",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 55
        }
      },
      "source": [
        "chat.findall(r'<l.*>{3,}')"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "lol lol lol; lmao lol lol; lol lol lol; la la la la la; la la la; la\n",
            "la la; lovely lol lol love; lol lol lol.; la la la; la la la\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Jw7VDXwD_c5M",
        "colab_type": "text"
      },
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n",
        "* Normalization :lemmatization(사전에 기반), stemming(규칙에 기반)\n",
        "\n",
        ">spoken과 같은 경우, lemmatization은 speak를 반환하지만 stemming은 규칙이 없어 spoken을 그대로 반환"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "3GEKTPVzAv1v",
        "colab_type": "code",
        "outputId": "cfe7f2fc-60c1-48de-de47-d48c7eed26cc",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "wnl.lemmatize('genes')"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'gene'"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "uYfGq3qo-Hlo",
        "colab_type": "code",
        "outputId": "7d3d53a9-b327-400d-bd44-eac492ee97e7",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "wnl = nltk.WordNetLemmatizer()\n",
        "[wnl.lemmatize(t) for t in chat]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['now',\n",
              " 'im',\n",
              " 'left',\n",
              " 'with',\n",
              " 'this',\n",
              " 'gay',\n",
              " 'name',\n",
              " ':P',\n",
              " 'PART',\n",
              " 'hey',\n",
              " 'everyone',\n",
              " 'ah',\n",
              " 'well',\n",
              " 'NICK',\n",
              " ':',\n",
              " 'U7',\n",
              " 'U7',\n",
              " 'is',\n",
              " 'a',\n",
              " 'gay',\n",
              " 'name',\n",
              " '.',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'give',\n",
              " 'U121',\n",
              " 'a',\n",
              " 'golf',\n",
              " 'clap',\n",
              " '.',\n",
              " ':)',\n",
              " 'JOIN',\n",
              " 'hi',\n",
              " 'U59',\n",
              " '26',\n",
              " '/',\n",
              " 'm',\n",
              " '/',\n",
              " 'ky',\n",
              " 'woman',\n",
              " 'that',\n",
              " 'are',\n",
              " 'nice',\n",
              " 'please',\n",
              " 'pm',\n",
              " 'me',\n",
              " 'JOIN',\n",
              " 'PART',\n",
              " 'there',\n",
              " 'ya',\n",
              " 'go',\n",
              " 'U7',\n",
              " 'do',\n",
              " \"n't\",\n",
              " 'golf',\n",
              " 'clap',\n",
              " 'me',\n",
              " '.',\n",
              " 'fuck',\n",
              " 'you',\n",
              " 'U121',\n",
              " ':@',\n",
              " 'whats',\n",
              " 'everyone',\n",
              " 'up',\n",
              " 'to',\n",
              " '?',\n",
              " 'PART',\n",
              " 'PART',\n",
              " 'i',\n",
              " \"'ll\",\n",
              " 'thunder',\n",
              " 'clap',\n",
              " 'your',\n",
              " 'as',\n",
              " '.',\n",
              " 'PART',\n",
              " 'and',\n",
              " 'i',\n",
              " 'dont',\n",
              " 'even',\n",
              " 'know',\n",
              " 'what',\n",
              " 'that',\n",
              " 'mean',\n",
              " '.',\n",
              " 'that',\n",
              " 'sound',\n",
              " 'painful',\n",
              " 'any',\n",
              " 'ladis',\n",
              " 'wanna',\n",
              " 'chat',\n",
              " '?',\n",
              " '29',\n",
              " 'm',\n",
              " '26',\n",
              " '/',\n",
              " 'm',\n",
              " 'JOIN',\n",
              " 'my',\n",
              " 'cousin',\n",
              " 'drew',\n",
              " 'a',\n",
              " 'messed',\n",
              " 'up',\n",
              " 'pic',\n",
              " 'on',\n",
              " 'my',\n",
              " 'cast',\n",
              " 'PART',\n",
              " '24',\n",
              " '/',\n",
              " 'm',\n",
              " 'boo',\n",
              " '.',\n",
              " '26',\n",
              " '/',\n",
              " 'm',\n",
              " 'and',\n",
              " 'sexy',\n",
              " 'lol',\n",
              " 'U115',\n",
              " 'boo',\n",
              " '.',\n",
              " 'JOIN',\n",
              " 'PART',\n",
              " 'he',\n",
              " 'drew',\n",
              " 'a',\n",
              " 'girl',\n",
              " 'with',\n",
              " 'leg',\n",
              " 'spread',\n",
              " 'boo',\n",
              " '.',\n",
              " 'hope',\n",
              " 'he',\n",
              " 'didnt',\n",
              " 'draw',\n",
              " 'a',\n",
              " 'penis',\n",
              " 'PART',\n",
              " 'ewwwww',\n",
              " 'lol',\n",
              " '&',\n",
              " 'a',\n",
              " 'head',\n",
              " 'between',\n",
              " 'her',\n",
              " 'leg',\n",
              " 'JOIN',\n",
              " 'JOIN',\n",
              " 'sound',\n",
              " 'good',\n",
              " 'to',\n",
              " 'me',\n",
              " '.',\n",
              " 'r',\n",
              " 'u',\n",
              " 'serious',\n",
              " 'JOIN',\n",
              " 'PART',\n",
              " 'I',\n",
              " \"'ll\",\n",
              " 'take',\n",
              " 'one',\n",
              " ',',\n",
              " 'please',\n",
              " '.',\n",
              " '&',\n",
              " 'i',\n",
              " 'have',\n",
              " 'to',\n",
              " 'go',\n",
              " 'to',\n",
              " 'the',\n",
              " 'doc',\n",
              " 'tomorrow',\n",
              " 'ya',\n",
              " 'man',\n",
              " 'I',\n",
              " 'am',\n",
              " 'too',\n",
              " '..',\n",
              " 'Connected',\n",
              " 'to',\n",
              " '...',\n",
              " 'Slip',\n",
              " 'away',\n",
              " '...',\n",
              " 'Fade',\n",
              " 'away',\n",
              " '...',\n",
              " 'Days',\n",
              " 'away',\n",
              " 'I',\n",
              " '...',\n",
              " 'Still',\n",
              " 'feel',\n",
              " 'you',\n",
              " '...',\n",
              " 'Touching',\n",
              " 'me',\n",
              " '...',\n",
              " 'Changing',\n",
              " 'me',\n",
              " '...',\n",
              " 'Considerably',\n",
              " 'killing',\n",
              " 'me',\n",
              " '...',\n",
              " 'heeeey',\n",
              " '!',\n",
              " 'do',\n",
              " \"n't\",\n",
              " 'you',\n",
              " 'have',\n",
              " 'a',\n",
              " 'sharpie',\n",
              " '?',\n",
              " '26',\n",
              " '/',\n",
              " 'm',\n",
              " 'you',\n",
              " \"'re\",\n",
              " 'back',\n",
              " 'U115',\n",
              " 'U129',\n",
              " 'yep',\n",
              " 'U115',\n",
              " 'Any',\n",
              " 'lady',\n",
              " 'wanna',\n",
              " 'chat',\n",
              " 'with',\n",
              " '24',\n",
              " '/',\n",
              " 'm',\n",
              " 'hurry',\n",
              " 'lady',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'JOIN',\n",
              " 'not',\n",
              " 'fast',\n",
              " 'enough',\n",
              " 'U116',\n",
              " 'a',\n",
              " 'bowl',\n",
              " 'i',\n",
              " 'got',\n",
              " 'a',\n",
              " 'blunt',\n",
              " 'an',\n",
              " 'a',\n",
              " 'bong',\n",
              " '......',\n",
              " 'lol',\n",
              " 'JOIN',\n",
              " 'well',\n",
              " ',',\n",
              " 'glad',\n",
              " 'it',\n",
              " 'worked',\n",
              " 'out',\n",
              " 'my',\n",
              " 'chair',\n",
              " 'is',\n",
              " 'too',\n",
              " 'hard',\n",
              " '.',\n",
              " 'Anyone',\n",
              " 'from',\n",
              " 'Tennessee',\n",
              " 'in',\n",
              " 'here',\n",
              " '?',\n",
              " 'hey',\n",
              " 'lady',\n",
              " 'a',\n",
              " 'am',\n",
              " 'i',\n",
              " 'is',\n",
              " 'U68',\n",
              " 'back',\n",
              " 'yet',\n",
              " 'PART',\n",
              " 'hey',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'U121',\n",
              " 'is',\n",
              " 'missing',\n",
              " 'a',\n",
              " 'B',\n",
              " 'in',\n",
              " 'her',\n",
              " 'name',\n",
              " 'and',\n",
              " 'i',\n",
              " 'do',\n",
              " \"n't\",\n",
              " 'complain',\n",
              " 'about',\n",
              " 'thing',\n",
              " 'being',\n",
              " 'hard',\n",
              " 'very',\n",
              " 'often',\n",
              " '.',\n",
              " 'ok',\n",
              " 'yes',\n",
              " 'U30',\n",
              " 'fire',\n",
              " 'it',\n",
              " 'up',\n",
              " 'Any',\n",
              " 'woman',\n",
              " 'from',\n",
              " 'Nashville',\n",
              " 'in',\n",
              " 'here',\n",
              " '?',\n",
              " 'JOIN',\n",
              " 'PART',\n",
              " 'and',\n",
              " 'an',\n",
              " 'an',\n",
              " '\"',\n",
              " 'itch',\n",
              " '\"',\n",
              " 'JOIN',\n",
              " 'yo',\n",
              " ',',\n",
              " 'U133',\n",
              " 'or',\n",
              " 'a',\n",
              " '\"',\n",
              " 'ogan',\n",
              " '\"',\n",
              " 'are',\n",
              " 'you',\n",
              " 'a',\n",
              " 'male',\n",
              " '?',\n",
              " 'JOIN',\n",
              " 'JOIN',\n",
              " 'show',\n",
              " 'will',\n",
              " 'let',\n",
              " \"'s\",\n",
              " 'talk',\n",
              " '.',\n",
              " 'PART',\n",
              " ':)',\n",
              " 'haha',\n",
              " 'brb',\n",
              " 'opps',\n",
              " 'JOIN',\n",
              " 'PART',\n",
              " 'sho',\n",
              " '*',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'keep',\n",
              " 'U115',\n",
              " 's',\n",
              " 'place',\n",
              " 'nice',\n",
              " 'and',\n",
              " 'warm',\n",
              " '.',\n",
              " 'hey',\n",
              " 'any',\n",
              " 'guy',\n",
              " 'with',\n",
              " 'cam',\n",
              " 'wanna',\n",
              " 'play',\n",
              " '?',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'sits',\n",
              " 'on',\n",
              " 'U68',\n",
              " \"'s\",\n",
              " 'lap',\n",
              " '.',\n",
              " 'JOIN',\n",
              " 'JOIN',\n",
              " 'any',\n",
              " 'guyz',\n",
              " 'wanna',\n",
              " 'chat',\n",
              " 'hi',\n",
              " 'there',\n",
              " 'boo',\n",
              " ',',\n",
              " 'it',\n",
              " \"'s\",\n",
              " 'a',\n",
              " 'female',\n",
              " '.',\n",
              " 'hey',\n",
              " 'U126',\n",
              " 'PART',\n",
              " 'i',\n",
              " 'wonna',\n",
              " 'chat',\n",
              " 'PART',\n",
              " '24',\n",
              " 'f',\n",
              " 'nc',\n",
              " 'single',\n",
              " 'mom',\n",
              " 'where',\n",
              " 'did',\n",
              " 'everyone',\n",
              " 'gooo',\n",
              " '?',\n",
              " 'sure',\n",
              " 'U126',\n",
              " 'JOIN',\n",
              " 'what',\n",
              " 'did',\n",
              " 'you',\n",
              " 'but',\n",
              " 'on',\n",
              " 'e-bay',\n",
              " 'i',\n",
              " 'feel',\n",
              " 'like',\n",
              " 'im',\n",
              " 'in',\n",
              " 'the',\n",
              " 'wrong',\n",
              " 'room',\n",
              " 'yeee',\n",
              " 'haw',\n",
              " 'U30',\n",
              " 'im',\n",
              " 'considering',\n",
              " 'changing',\n",
              " 'my',\n",
              " 'nickname',\n",
              " 'to',\n",
              " '\"',\n",
              " 'ihavehotnips',\n",
              " '\"',\n",
              " 'JOIN',\n",
              " 'i',\n",
              " 'do',\n",
              " \"n't\",\n",
              " 'want',\n",
              " 'hot',\n",
              " 'pic',\n",
              " 'of',\n",
              " 'a',\n",
              " 'female',\n",
              " ',',\n",
              " 'I',\n",
              " 'can',\n",
              " 'look',\n",
              " 'in',\n",
              " 'a',\n",
              " 'mirror',\n",
              " '.',\n",
              " 'hi',\n",
              " 'U64',\n",
              " 'wb',\n",
              " 'U139',\n",
              " 'u',\n",
              " 'should',\n",
              " 'U44',\n",
              " ':)',\n",
              " 'PART',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'single',\n",
              " 'dad',\n",
              " 'here',\n",
              " 'JOIN',\n",
              " 'ty',\n",
              " 'U68',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'Hi',\n",
              " 'U139',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'hi',\n",
              " 'U138',\n",
              " 'HAHAHA',\n",
              " 'yw',\n",
              " 'U139',\n",
              " 'you',\n",
              " 'should',\n",
              " 'make',\n",
              " 'it',\n",
              " \"'\",\n",
              " 'iamahotnip',\n",
              " \"'\",\n",
              " ',',\n",
              " 'U44',\n",
              " 'alright',\n",
              " 'hi',\n",
              " 'U139',\n",
              " '.',\n",
              " 'you',\n",
              " \"'re\",\n",
              " 'fucking',\n",
              " 'hot',\n",
              " '.',\n",
              " 'i',\n",
              " 'thought',\n",
              " 'of',\n",
              " 'that',\n",
              " '!',\n",
              " 'hi',\n",
              " 'U126',\n",
              " ',',\n",
              " 'it',\n",
              " 'so',\n",
              " 'late',\n",
              " 'lmao',\n",
              " 'ahah',\n",
              " '\"',\n",
              " 'iamahotniplickme',\n",
              " '\"',\n",
              " 'PART',\n",
              " 'Hi',\n",
              " 'U121',\n",
              " 'in',\n",
              " 'ny',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'would',\n",
              " 'lol',\n",
              " '@',\n",
              " 'U121',\n",
              " '. . .',\n",
              " 'but',\n",
              " 'appearently',\n",
              " 'she',\n",
              " 'doe',\n",
              " 'not',\n",
              " 'like',\n",
              " 'that',\n",
              " '.',\n",
              " 'cya',\n",
              " 'later',\n",
              " 'guy',\n",
              " 'single',\n",
              " 'white',\n",
              " 'male',\n",
              " '26',\n",
              " 'ca',\n",
              " \"n't\",\n",
              " 'sleep',\n",
              " 'huh',\n",
              " '?',\n",
              " 'JOIN',\n",
              " 'hi',\n",
              " 'U139',\n",
              " 'bye',\n",
              " 'U20',\n",
              " 'well',\n",
              " 'U68',\n",
              " 'what',\n",
              " 'did',\n",
              " 'u',\n",
              " 'buy',\n",
              " 'yeah',\n",
              " 'i',\n",
              " 'know',\n",
              " ':)',\n",
              " 'U136',\n",
              " '..',\n",
              " 'get',\n",
              " 'the',\n",
              " 'hell',\n",
              " 'outta',\n",
              " 'my',\n",
              " 'freaking',\n",
              " 'PM',\n",
              " 'box',\n",
              " '..',\n",
              " 'Im',\n",
              " 'with',\n",
              " 'my',\n",
              " 'fiance',\n",
              " '!!!!!!!!!!!!!!!!',\n",
              " 'answer',\n",
              " 'for',\n",
              " 'U139',\n",
              " '...',\n",
              " 'hi',\n",
              " 'U101',\n",
              " ';)',\n",
              " 'I',\n",
              " 'like',\n",
              " 'it',\n",
              " 'when',\n",
              " 'you',\n",
              " 'do',\n",
              " 'it',\n",
              " ',',\n",
              " 'U83',\n",
              " 'iamahotnipwithpics',\n",
              " 'U20',\n",
              " 'go',\n",
              " 'plan',\n",
              " 'the',\n",
              " 'wedding',\n",
              " '!',\n",
              " ':P',\n",
              " 'first',\n",
              " 'warning',\n",
              " '!!!!!',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'uh',\n",
              " 'huh',\n",
              " 'A',\n",
              " 'gold',\n",
              " 'jeep',\n",
              " 'charm',\n",
              " 'for',\n",
              " 'my',\n",
              " 'necklace',\n",
              " 'U30',\n",
              " 'make',\n",
              " 'sure',\n",
              " 'u',\n",
              " 'buy',\n",
              " 'a',\n",
              " 'nice',\n",
              " 'ring',\n",
              " 'for',\n",
              " 'U6',\n",
              " 'lol',\n",
              " 'U7',\n",
              " 'Hi',\n",
              " 'U115',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'isnt',\n",
              " 'falling',\n",
              " 'for',\n",
              " 'that',\n",
              " 'one',\n",
              " 'again',\n",
              " '.',\n",
              " 'U136',\n",
              " '..',\n",
              " 'get',\n",
              " 'the',\n",
              " 'hell',\n",
              " 'in',\n",
              " 'my',\n",
              " 'freaking',\n",
              " 'PM',\n",
              " 'box',\n",
              " '..',\n",
              " 'I',\n",
              " 'have',\n",
              " 'no',\n",
              " 'fiance',\n",
              " '!!!!!!!',\n",
              " 'right',\n",
              " 'on',\n",
              " 'i',\n",
              " 'have',\n",
              " 'one',\n",
              " 'already',\n",
              " 'U7',\n",
              " '...',\n",
              " 'yayayayayyy',\n",
              " '!!',\n",
              " 'OOooOO',\n",
              " ':)',\n",
              " 'how',\n",
              " 'you',\n",
              " 'doin',\n",
              " 'U139',\n",
              " 'lmaoo',\n",
              " 'U121',\n",
              " \"'\",\n",
              " 'iamahotnipwithhotnippics',\n",
              " \"'\",\n",
              " 'how',\n",
              " 'many',\n",
              " 'kt',\n",
              " 'any',\n",
              " 'single',\n",
              " 'white',\n",
              " 'female',\n",
              " '?',\n",
              " 'off',\n",
              " 'off',\n",
              " '?',\n",
              " 'really',\n",
              " '?',\n",
              " 'Just',\n",
              " 'fine',\n",
              " 'thanks',\n",
              " ',',\n",
              " 'how',\n",
              " 'are',\n",
              " 'you',\n",
              " '?',\n",
              " 'that',\n",
              " 'sound',\n",
              " 'freakishly',\n",
              " 'like',\n",
              " 'dr',\n",
              " 'seuss',\n",
              " 'wow',\n",
              " '.',\n",
              " 'twice',\n",
              " ',',\n",
              " 'I',\n",
              " \"'m\",\n",
              " 'impressed',\n",
              " '.',\n",
              " 'PART',\n",
              " 'hiya',\n",
              " 'room',\n",
              " 'lmao',\n",
              " '!!!',\n",
              " 'im',\n",
              " 'doin',\n",
              " 'alright',\n",
              " 'thanks',\n",
              " 'omg',\n",
              " 'Finger',\n",
              " '..',\n",
              " 'Deep',\n",
              " 'within',\n",
              " 'the',\n",
              " 'borderline',\n",
              " '..',\n",
              " 'Show',\n",
              " 'me',\n",
              " 'that',\n",
              " 'you',\n",
              " 'love',\n",
              " 'me',\n",
              " ',',\n",
              " 'and',\n",
              " 'that',\n",
              " 'we',\n",
              " 'belong',\n",
              " 'together',\n",
              " '...',\n",
              " 'Relax',\n",
              " '..',\n",
              " 'Turn',\n",
              " 'around',\n",
              " '..',\n",
              " 'And',\n",
              " 'take',\n",
              " 'my',\n",
              " '..',\n",
              " 'Hand',\n",
              " '..',\n",
              " 'I',\n",
              " 'just',\n",
              " 'laughed',\n",
              " 'didnt',\n",
              " 'even',\n",
              " 'look',\n",
              " 'hang',\n",
              " 'on',\n",
              " 'finger',\n",
              " '?',\n",
              " 'yay',\n",
              " '!',\n",
              " 'PART',\n",
              " 'that',\n",
              " \"'s\",\n",
              " 'such',\n",
              " 'a',\n",
              " 'DIRTY',\n",
              " 'word',\n",
              " '.',\n",
              " 'oh',\n",
              " 'it',\n",
              " '14',\n",
              " 'k',\n",
              " 'U141',\n",
              " '...',\n",
              " 'get',\n",
              " 'outta',\n",
              " 'my',\n",
              " 'PM',\n",
              " 'Box',\n",
              " '..',\n",
              " 'didnt',\n",
              " 'ya',\n",
              " 'hear',\n",
              " '!!!!',\n",
              " 'PART',\n",
              " 'JOIN',\n",
              " 'geeshhh',\n",
              " 'lol',\n",
              " 'U6',\n",
              " 'PART',\n",
              " 'he',\n",
              " 'deaf',\n",
              " 'ppl',\n",
              " 'here',\n",
              " 'dont',\n",
              " 'get',\n",
              " 'it',\n",
              " 'U6',\n",
              " ',',\n",
              " '!',\n",
              " 'hug',\n",
              " 'U115',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'watch',\n",
              " 'U6',\n",
              " 'hug',\n",
              " 'the',\n",
              " \"stuffin'\",\n",
              " 'outta',\n",
              " 'U115',\n",
              " '....',\n",
              " 'you',\n",
              " 'can',\n",
              " 'forward',\n",
              " 'the',\n",
              " 'pervs',\n",
              " 'to',\n",
              " 'me',\n",
              " '.',\n",
              " '.',\n",
              " 'ACTION',\n",
              " 'hug',\n",
              " 'U6',\n",
              " '.',\n",
              " 'any',\n",
              " 'girl',\n",
              " 'wanna',\n",
              " 'chat',\n",
              " 'with',\n",
              " '24',\n",
              " '/',\n",
              " 'm',\n",
              " 'please',\n",
              " 'behave',\n",
              " 'baby',\n",
              " 'boy',\n",
              " '..',\n",
              " 'I',\n",
              " 'gottsa',\n",
              " 'go',\n",
              " 'now',\n",
              " 'Only',\n",
              " 'for',\n",
              " 'the',\n",
              " 'dirty',\n",
              " 'minded',\n",
              " ',',\n",
              " 'U121',\n",
              " '.',\n",
              " 'ima',\n",
              " 'behave',\n",
              " 'yes',\n",
              " 'U121',\n",
              " '??',\n",
              " 'PART',\n",
              " 'aww',\n",
              " 'U6',\n",
              " 'have',\n",
              " 'fun',\n",
              " 'forwads',\n",
              " 'away',\n",
              " 'from',\n",
              " 'U121',\n",
              " 'i',\n",
              " 'already',\n",
              " 'wrote',\n",
              " 'what',\n",
              " 'i',\n",
              " 'wanted',\n",
              " 'you',\n",
              " 'to',\n",
              " 'read',\n",
              " '.',\n",
              " 'there',\n",
              " 'nobody',\n",
              " 'here',\n",
              " 'i',\n",
              " 'wanna',\n",
              " 'misbeahve',\n",
              " 'with',\n",
              " 'lol',\n",
              " 'JOIN',\n",
              " 'so',\n",
              " 'read',\n",
              " 'it',\n",
              " '.',\n",
              " 'thanks',\n",
              " 'U7',\n",
              " '..',\n",
              " 'Im',\n",
              " 'happy',\n",
              " 'to',\n",
              " 'have',\n",
              " 'my',\n",
              " 'fiance',\n",
              " 'here',\n",
              " '!!',\n",
              " 'forwads',\n",
              " '?',\n",
              " 'JOIN',\n",
              " '22',\n",
              " '/',\n",
              " 'm',\n",
              " '/',\n",
              " 'wa',\n",
              " 'any',\n",
              " 'lady',\n",
              " 'want',\n",
              " 'to',\n",
              " 'chat',\n",
              " 'can',\n",
              " 'i',\n",
              " 'talk',\n",
              " 'to',\n",
              " 'him',\n",
              " '!!',\n",
              " 'lol',\n",
              " 'U121',\n",
              " '!!!',\n",
              " 'forward',\n",
              " 'too',\n",
              " 'lol',\n",
              " 'JOIN',\n",
              " 'ALL',\n",
              " 'PErvs',\n",
              " '...',\n",
              " 'redirect',\n",
              " 'to',\n",
              " 'U121',\n",
              " \"'s\",\n",
              " 'PM',\n",
              " 'VBox',\n",
              " '!!!',\n",
              " 'PART',\n",
              " 'I',\n",
              " \"'ll\",\n",
              " 'take',\n",
              " 'all',\n",
              " 'that',\n",
              " 'I',\n",
              " 'can',\n",
              " 'get',\n",
              " '.',\n",
              " 'JOIN',\n",
              " 'PART',\n",
              " 'who',\n",
              " 'is',\n",
              " 'he',\n",
              " 'U6',\n",
              " '?',\n",
              " 'i',\n",
              " 'hate',\n",
              " 'it',\n",
              " 'when',\n",
              " 'my',\n",
              " 'devotion',\n",
              " 'to',\n",
              " 'christianity',\n",
              " 'is',\n",
              " 'questioned',\n",
              " '.',\n",
              " 'U121',\n",
              " 'who',\n",
              " 'honey',\n",
              " '??',\n",
              " 'Depends',\n",
              " 'Hi',\n",
              " 'all',\n",
              " '31',\n",
              " '/',\n",
              " 'm',\n",
              " '/',\n",
              " 'CA',\n",
              " 'U115',\n",
              " 'is',\n",
              " 'having',\n",
              " 'my',\n",
              " 'baby',\n",
              " 'U20',\n",
              " 'yep',\n",
              " ...]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "gxGKlyKyA4v8",
        "colab_type": "code",
        "outputId": "29eea294-76c5-4ebb-e48e-f9e1f0d8ec78",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 185
        }
      },
      "source": [
        "from nltk.corpus import stopwords\n",
        "words = ['the', 'adenovirus', 'protein', 'can', 'activate', 'transcription', 'of', 'the', 'mouse', 'c-fos', 'gene', 'in', 'a', 'manner', 'that', 'depends', 'on']\n",
        "[x for x in words if not (x in stopwords.words('english'))]"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['adenovirus',\n",
              " 'protein',\n",
              " 'activate',\n",
              " 'transcription',\n",
              " 'mouse',\n",
              " 'c-fos',\n",
              " 'gene',\n",
              " 'manner',\n",
              " 'depends']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 39
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "K7-PM5aoBaXc",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "giRaw = GNI.raw()\n",
        "giWords = GNI.words()"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "PCTY-w9TBmNc",
        "colab_type": "code",
        "outputId": "24bf95d1-0a69-4aab-846f-c30bf6afc8df",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "for fileid in GNI.fileids():\n",
        "  numChars = len(GNI.raw(fileid))\n",
        "  words = GNI.raw(fileid).split(' ')\n",
        "  print(fileid, \":\", numChars, len(words))"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "gi-2017-15-3-81.txt : 1821 207\n",
            "gi-2017-15-3-82.txt : 10543 1544\n",
            "gi-2017-15-4-113.txt : 2078 244\n",
            "gi-2017-15-4-114.txt : 20616 2970\n",
            "gi-2017-15-4-123.txt : 11901 1800\n",
            "gi-2017-15-4-128.txt : 13307 2216\n",
            "gi-2017-15-4-136.txt : 14027 2166\n",
            "gi-2017-15-4-142.txt : 9544 1467\n",
            "gi-2017-15-4-147.txt : 17028 2720\n",
            "gi-2017-15-4-156.txt : 10396 1657\n",
            "gi-2017-15-4-162.txt : 17999 2887\n",
            "gi-2017-15-4-170.txt : 17355 2743\n",
            "gi-2017-15-4-178.txt : 10693 1699\n",
            "gi-2018-16-1-1.txt : 2100 274\n",
            "gi-2018-16-1-10.txt : 12710 2013\n",
            "gi-2018-16-1-14.txt : 15182 2494\n",
            "gi-2018-16-1-2.txt : 15385 2470\n",
            "gni-10-1.txt : 5530 690\n",
            "gni-10-106.txt : 10522 1683\n",
            "gni-10-110.txt : 12861 2011\n",
            "gni-10-117.txt : 15097 2492\n",
            "gni-10-123.txt : 15519 2649\n",
            "gni-10-128.txt : 8124 1223\n",
            "gni-10-133.txt : 33681 5248\n",
            "gni-10-145.txt : 22252 3317\n",
            "gni-10-153.txt : 36823 5760\n",
            "gni-10-16.txt : 5590 769\n",
            "gni-10-167.txt : 13191 2071\n",
            "gni-10-175.txt : 29621 4924\n",
            "gni-10-184.txt : 18074 2852\n",
            "gni-10-194.txt : 14164 2322\n",
            "gni-10-200.txt : 13403 2081\n",
            "gni-10-206.txt : 10039 1529\n",
            "gni-10-213.txt : 2887 420\n",
            "gni-10-214.txt : 17274 2689\n",
            "gni-10-220.txt : 17385 2966\n",
            "gni-10-226.txt : 20774 3348\n",
            "gni-10-23.txt : 5101 640\n",
            "gni-10-234.txt : 15307 2354\n",
            "gni-10-239.txt : 13303 2181\n",
            "gni-10-244.txt : 12353 1879\n",
            "gni-10-249.txt : 22612 3586\n",
            "gni-10-256.txt : 18197 2946\n",
            "gni-10-263.txt : 6896 1009\n",
            "gni-10-266.txt : 8990 1440\n",
            "gni-10-271.txt : 8112 1210\n",
            "gni-10-33.txt : 5372 661\n",
            "gni-10-40.txt : 5183 725\n",
            "gni-10-44.txt : 5269 710\n",
            "gni-10-51.txt : 5311 684\n",
            "gni-10-58.txt : 5139 695\n",
            "gni-10-65.txt : 4079 527\n",
            "gni-10-69.txt : 15071 2264\n",
            "gni-10-74.txt : 15076 2297\n",
            "gni-10-81.txt : 25334 4339\n",
            "gni-10-88.txt : 27043 4566\n",
            "gni-10-9.txt : 5308 692\n",
            "gni-10-99.txt : 14128 2124\n",
            "gni-11-1.txt : 2383 348\n",
            "gni-11-101.txt : 2155 299\n",
            "gni-11-102.txt : 39702 6100\n",
            "gni-11-114.txt : 17176 2572\n",
            "gni-11-121.txt : 21902 3402\n",
            "gni-11-129.txt : 11252 1773\n",
            "gni-11-135.txt : 9946 1616\n",
            "gni-11-142.txt : 13277 2084\n",
            "gni-11-149.txt : 12366 1899\n",
            "gni-11-15.txt : 23264 3752\n",
            "gni-11-155.txt : 18873 3091\n",
            "gni-11-161.txt : 3005 322\n",
            "gni-11-163.txt : 1802 244\n",
            "gni-11-164.txt : 31366 4677\n",
            "gni-11-174.txt : 20402 3289\n",
            "gni-11-180.txt : 16015 2373\n",
            "gni-11-186.txt : 17361 2518\n",
            "gni-11-191.txt : 35543 5703\n",
            "gni-11-2.txt : 12036 1818\n",
            "gni-11-200.txt : 29408 4527\n",
            "gni-11-211.txt : 20862 3235\n",
            "gni-11-218.txt : 22415 3454\n",
            "gni-11-224.txt : 14966 2276\n",
            "gni-11-230.txt : 32324 5104\n",
            "gni-11-239.txt : 12396 1949\n",
            "gni-11-24.txt : 19967 3262\n",
            "gni-11-245.txt : 22561 3583\n",
            "gni-11-254.txt : 21144 3472\n",
            "gni-11-263.txt : 23561 3904\n",
            "gni-11-272.txt : 11828 1839\n",
            "gni-11-277.txt : 10045 1565\n",
            "gni-11-282.txt : 13951 2145\n",
            "gni-11-289.txt : 8242 1212\n",
            "gni-11-34.txt : 9391 1492\n",
            "gni-11-38.txt : 11326 1894\n",
            "gni-11-46.txt : 11941 1904\n",
            "gni-11-52.txt : 7061 1058\n",
            "gni-11-55.txt : 9024 1406\n",
            "gni-11-58.txt : 939 78\n",
            "gni-11-59.txt : 2443 338\n",
            "gni-11-60.txt : 20928 3251\n",
            "gni-11-68.txt : 21136 3453\n",
            "gni-11-7.txt : 22548 3410\n",
            "gni-11-76.txt : 20193 3076\n",
            "gni-11-83.txt : 26117 4207\n",
            "gni-11-93.txt : 9601 1497\n",
            "gni-11-97.txt : 1144 108\n",
            "gni-11-98.txt : 1406 144\n",
            "gni-11-99.txt : 1516 169\n",
            "gni-12-1.txt : 2494 347\n",
            "gni-12-105.txt : 33065 5401\n",
            "gni-12-114.txt : 15610 2564\n",
            "gni-12-12.txt : 28343 4216\n",
            "gni-12-121.txt : 14177 2107\n",
            "gni-12-127.txt : 10871 1744\n",
            "gni-12-135.txt : 2598 326\n",
            "gni-12-136.txt : 21934 3393\n",
            "gni-12-145.txt : 16703 2752\n",
            "gni-12-151.txt : 18737 2996\n",
            "gni-12-156.txt : 24814 3792\n",
            "gni-12-165.txt : 13999 2327\n",
            "gni-12-171.txt : 13744 2186\n",
            "gni-12-181.txt : 12330 1917\n",
            "gni-12-187.txt : 17490 2757\n",
            "gni-12-195.txt : 18594 2820\n",
            "gni-12-2.txt : 31106 4566\n",
            "gni-12-203.txt : 7257 1158\n",
            "gni-12-208.txt : 25745 4436\n",
            "gni-12-21.txt : 32810 5272\n",
            "gni-12-216.txt : 16317 2608\n",
            "gni-12-225.txt : 12662 1963\n",
            "gni-12-231.txt : 10873 1733\n",
            "gni-12-236.txt : 9170 1395\n",
            "gni-12-240.txt : 20161 3264\n",
            "gni-12-247.txt : 13046 1986\n",
            "gni-12-254.txt : 17917 3005\n",
            "gni-12-261.txt : 16618 2643\n",
            "gni-12-268.txt : 21348 3204\n",
            "gni-12-276.txt : 16615 2732\n",
            "gni-12-283.txt : 16442 2574\n",
            "gni-12-289.txt : 8652 1300\n",
            "gni-12-35.txt : 16633 2439\n",
            "gni-12-42.txt : 19059 2919\n",
            "gni-12-49.txt : 2519 315\n",
            "gni-12-50.txt : 20942 3368\n",
            "gni-12-58.txt : 11112 1712\n",
            "gni-12-64.txt : 15929 2573\n",
            "gni-12-71.txt : 13549 2287\n",
            "gni-12-76.txt : 5755 789\n",
            "gni-12-79.txt : 3033 405\n",
            "gni-12-80.txt : 19408 3201\n",
            "gni-12-87.txt : 36731 5985\n",
            "gni-12-98.txt : 16224 2590\n",
            "gni-13-1.txt : 1626 192\n",
            "gni-13-102.txt : 30878 4894\n",
            "gni-13-112.txt : 19159 2947\n",
            "gni-13-119.txt : 18953 2896\n",
            "gni-13-126.txt : 13340 2080\n",
            "gni-13-132.txt : 10794 1676\n",
            "gni-13-137.txt : 19029 2983\n",
            "gni-13-146.txt : 16305 2640\n",
            "gni-13-15.txt : 23937 3860\n",
            "gni-13-152.txt : 8180 1284\n",
            "gni-13-156.txt : 1375 135\n",
            "gni-13-2.txt : 13291 2094\n",
            "gni-13-25.txt : 2080 288\n",
            "gni-13-26.txt : 15269 2520\n",
            "gni-13-31.txt : 18432 3015\n",
            "gni-13-40.txt : 12151 1891\n",
            "gni-13-45.txt : 14357 2246\n",
            "gni-13-53.txt : 16837 2709\n",
            "gni-13-60.txt : 15182 2430\n",
            "gni-13-69.txt : 2423 324\n",
            "gni-13-7.txt : 19150 3070\n",
            "gni-13-70.txt : 14805 2293\n",
            "gni-13-76.txt : 12350 1906\n",
            "gni-13-81.txt : 12275 1876\n",
            "gni-13-86.txt : 10173 1505\n",
            "gni-13-90.txt : 6190 928\n",
            "gni-13-93.txt : 2471 326\n",
            "gni-13-94.txt : 23970 3738\n",
            "gni-14-1.txt : 2792 333\n",
            "gni-14-104.txt : 18140 2957\n",
            "gni-14-112.txt : 21367 3403\n",
            "gni-14-12.txt : 20792 3050\n",
            "gni-14-125.txt : 26533 4188\n",
            "gni-14-137.txt : 3472 437\n",
            "gni-14-138.txt : 18807 3110\n",
            "gni-14-149.txt : 14400 2352\n",
            "gni-14-160.txt : 15693 2551\n",
            "gni-14-166.txt : 18315 3031\n",
            "gni-14-173.txt : 22462 3804\n",
            "gni-14-181.txt : 12928 2122\n",
            "gni-14-187.txt : 26040 4413\n",
            "gni-14-196.txt : 20256 3355\n",
            "gni-14-2.txt : 29109 4481\n",
            "gni-14-20.txt : 25972 4327\n",
            "gni-14-205.txt : 11523 1788\n",
            "gni-14-211.txt : 11200 1770\n",
            "gni-14-216.txt : 16340 2599\n",
            "gni-14-222.txt : 13990 2147\n",
            "gni-14-230.txt : 12510 2156\n",
            "gni-14-234.txt : 16311 2535\n",
            "gni-14-241.txt : 29260 4694\n",
            "gni-14-255.txt : 23078 3479\n",
            "gni-14-29.txt : 10918 1756\n",
            "gni-14-34.txt : 14532 2252\n",
            "gni-14-41.txt : 2276 266\n",
            "gni-14-42.txt : 10359 1563\n",
            "gni-6-1-1.txt : 14751 2132\n",
            "gni-6-1-14.txt : 31654 4600\n",
            "gni-6-1-18.txt : 39453 5800\n",
            "gni-6-1-29.txt : 43135 6329\n",
            "gni-6-1-32.txt : 44119 6478\n",
            "gni-6-1-36.txt : 45449 6678\n",
            "gni-6-1-44.txt : 46954 6902\n",
            "gni-6-1-50.txt : 48171 7088\n",
            "gni-6-1-54.txt : 49498 7275\n",
            "gni-6-1-8.txt : 22845 3321\n",
            "gni-6-2-57.txt : 50711 7457\n",
            "gni-6-2-64.txt : 51762 7629\n",
            "gni-6-2-68.txt : 52908 7813\n",
            "gni-6-2-72.txt : 53765 7948\n",
            "gni-6-2-77.txt : 55208 8172\n",
            "gni-6-2-84.txt : 56642 8393\n",
            "gni-6-2-87.txt : 57921 8582\n",
            "gni-6-2-91.txt : 59226 8775\n",
            "gni-6-2-95.txt : 59226 8775\n",
            "gni-6-3-110.txt : 61365 9114\n",
            "gni-6-3-117.txt : 62587 9292\n",
            "gni-6-3-126.txt : 63739 9454\n",
            "gni-6-3-130.txt : 65032 9651\n",
            "gni-6-3-136.txt : 66285 9829\n",
            "gni-6-3-142.txt : 67756 10057\n",
            "gni-6-3-147.txt : 68880 10229\n",
            "gni-6-3-153.txt : 70191 10423\n",
            "gni-6-3-157.txt : 71277 10600\n",
            "gni-6-3-99.txt : 60298 8958\n",
            "gni-6-4-161.txt : 72508 10789\n",
            "gni-6-4-166.txt : 73615 10945\n",
            "gni-6-4-173.txt : 74785 11118\n",
            "gni-6-4-181.txt : 76050 11308\n",
            "gni-6-4-192.txt : 77061 11476\n",
            "gni-6-4-202.txt : 78328 11668\n",
            "gni-6-4-210.txt : 79694 11856\n",
            "gni-6-4-223.txt : 80867 12024\n",
            "gni-6-4-227.txt : 82187 12204\n",
            "gni-6-4-231.txt : 83525 12416\n",
            "gni-7-1-1.txt : 84611 12574\n",
            "gni-7-1-13.txt : 85701 12743\n",
            "gni-7-1-20.txt : 86783 12910\n",
            "gni-7-1-26.txt : 88259 13134\n",
            "gni-7-1-32.txt : 89600 13336\n",
            "gni-7-1-38.txt : 90705 13495\n",
            "gni-7-1-41.txt : 91761 13619\n",
            "gni-7-1-46.txt : 92293 13686\n",
            "gni-7-1-49.txt : 92293 13686\n",
            "gni-7-2-107.txt : 99279 14734\n",
            "gni-7-2-111.txt : 100539 14922\n",
            "gni-7-2-122.txt : 101891 15140\n",
            "gni-7-2-131.txt : 103127 15329\n",
            "gni-7-2-136.txt : 104465 15534\n",
            "gni-7-2-53.txt : 93237 13831\n",
            "gni-7-2-57.txt : 94241 13979\n",
            "gni-7-2-65.txt : 95520 14192\n",
            "gni-7-2-85.txt : 96866 14387\n",
            "gni-7-2-97.txt : 98003 14549\n",
            "gni-7-3-141.txt : 105730 15737\n",
            "gni-7-3-148.txt : 106161 15815\n",
            "gni-7-3-152.txt : 107421 16011\n",
            "gni-7-3-163.txt : 108461 16166\n",
            "gni-7-3-168.txt : 108754 16215\n",
            "gni-7-3-171.txt : 109854 16380\n",
            "gni-7-3-175.txt : 109854 16380\n",
            "gni-7-3-178.txt : 109854 16380\n",
            "gni-7-4-181.txt : 111300 16598\n",
            "gni-7-4-187.txt : 112779 16812\n",
            "gni-7-4-195.txt : 113967 16999\n",
            "gni-7-4-203.txt : 115313 17205\n",
            "gni-7-4-208.txt : 116479 17374\n",
            "gni-7-4-212.txt : 117683 17542\n",
            "gni-7-4-217.txt : 119092 17773\n",
            "gni-8-1-1 (1).txt : 120517 18018\n",
            "gni-8-1-1.txt : 120517 18018\n",
            "gni-8-1-19 (1).txt : 122901 18385\n",
            "gni-8-1-19.txt : 122901 18385\n",
            "gni-8-1-28 (1).txt : 124275 18625\n",
            "gni-8-1-28.txt : 124275 18625\n",
            "gni-8-1-34 (1).txt : 125603 18863\n",
            "gni-8-1-34.txt : 125603 18863\n",
            "gni-8-1-41 (1).txt : 127032 19080\n",
            "gni-8-1-41.txt : 127032 19080\n",
            "gni-8-1-50 (1).txt : 128360 19287\n",
            "gni-8-1-50.txt : 128360 19287\n",
            "gni-8-1-58.txt : 129503 19473\n",
            "gni-8-1-9.txt : 121837 18217\n",
            "gni-8-2-63.txt : 130726 19659\n",
            "gni-8-2-70.txt : 132127 19883\n",
            "gni-8-2-76.txt : 133212 20039\n",
            "gni-8-2-81.txt : 134225 20191\n",
            "gni-8-2-86.txt : 135397 20340\n",
            "gni-8-2-90.txt : 136700 20516\n",
            "gni-8-2-94.txt : 136898 20546\n",
            "gni-8-3-101.txt : 138183 20751\n",
            "gni-8-3-103.txt : 139357 20923\n",
            "gni-8-3-108.txt : 139958 21012\n",
            "gni-8-3-116.txt : 140301 21060\n",
            "gni-8-3-122.txt : 140651 21114\n",
            "gni-8-3-131.txt : 141048 21174\n",
            "gni-8-3-138.txt : 141469 21242\n",
            "gni-8-3-142.txt : 142237 21360\n",
            "gni-8-3-150.txt : 142967 21479\n",
            "gni-8-3-159.txt : 142967 21479\n",
            "gni-8-4-165.txt : 143135 21503\n",
            "gni-8-4-170.txt : 143767 21604\n",
            "gni-8-4-177.txt : 144448 21704\n",
            "gni-8-4-185.txt : 145094 21804\n",
            "gni-8-4-194.txt : 145534 21872\n",
            "gni-8-4-201.txt : 145820 21911\n",
            "gni-8-4-206.txt : 146444 22014\n",
            "gni-8-4-212.txt : 146998 22103\n",
            "gni-9-1-1.txt : 147660 22202\n",
            "gni-9-1-12.txt : 148553 22336\n",
            "gni-9-1-19.txt : 149035 22408\n",
            "gni-9-1-28.txt : 149565 22491\n",
            "gni-9-1-37.txt : 149565 22491\n",
            "gni-9-1-39.txt : 150100 22568\n",
            "gni-9-1-5.txt : 147996 22250\n",
            "gni-9-2-45.txt : 150901 22673\n",
            "gni-9-2-52.txt : 151436 22749\n",
            "gni-9-2-59.txt : 151761 22795\n",
            "gni-9-2-64.txt : 152197 22865\n",
            "gni-9-2-69.txt : 152749 22952\n",
            "gni-9-2-74.txt : 153601 23073\n",
            "gni-9-2-79.txt : 153929 23118\n",
            "gni-9-2-85.txt : 154268 23160\n",
            "gni-9-2-89.txt : 154632 23208\n",
            "gni-9-3-102.txt : 154632 23208\n",
            "gni-9-3-114.txt : 155197 23289\n",
            "gni-9-3-121.txt : 155695 23371\n",
            "gni-9-3-127.txt : 156415 23486\n",
            "gni-9-3-134.txt : 156857 23549\n",
            "gni-9-3-136.txt : 157439 23652\n",
            "gni-9-3-138.txt : 157889 23723\n",
            "gni-9-3-93.txt : 154632 23208\n",
            "gni-9-4-143.txt : 158611 23825\n",
            "gni-9-4-152.txt : 159073 23896\n",
            "gni-9-4-161.txt : 159679 23990\n",
            "gni-9-4-173.txt : 159708 23994\n",
            "gni-9-4-181.txt : 160206 24077\n",
            "gni-9-4-189.txt : 160887 24181\n",
            "gni-9-4-194.txt : 161336 24244\n",
            "gni-9-4-197.txt : 161685 24288\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "5VqjtHatB7x8",
        "colab_type": "code",
        "outputId": "53c8e183-4c16-4fc3-bead-1e64e9faf6fb",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "giRaws={}\n",
        "for fileid in GNI.fileids():\n",
        "  giRaws[fileid] = GNI.raw(fileid)\n",
        "  giRaws[fileid] = re.sub(r'[\\n\\r.,()]', '', giRaws[fileid])\n",
        "  words=nltk.word_tokenize(giRaws[fileid])\n",
        "  words = [x for x in words if not (x.lower() in stopwords.words('english'))]\n",
        "  fd = nltk.FreqDist(words)\n",
        "  fd = sorted([(freq, word) for (word, freq) in fd.items()], reverse=True)\n",
        "  print(fileid, fd[0:10])"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "gi-2017-15-3-81.txt [(6, ':'), (5, '’'), (4, 'group'), (3, 'patients'), (3, 'number'), (3, 'events'), (3, 'chromosomal'), (3, 'Dr'), (3, '&'), (2, 'tool')]\n",
            "gi-2017-15-3-82.txt [(17, 'CMA'), (17, ':'), (17, '%'), (15, 'syndrome'), (14, ']'), (14, '['), (13, ';'), (11, 'CNVs'), (10, 'clinical'), (10, 'MCAs')]\n",
            "gi-2017-15-4-113.txt [(6, ':'), (3, '’'), (3, 'issue'), (3, 'University'), (3, 'Korea'), (3, '&'), (2, 'reviewed'), (2, 'group'), (2, 'genes'), (2, 'gene')]\n",
            "gi-2017-15-4-114.txt [(105, ']'), (105, '['), (51, 'CTCF'), (38, 'cohesin'), (30, 'chromatin'), (24, 'sites'), (23, 'gene'), (21, 'DNA'), (19, 'complex'), (19, ':')]\n",
            "gi-2017-15-4-123.txt [(33, 'codons'), (32, ']'), (32, '['), (29, 'synonymous'), (16, 'genes'), (15, ':'), (12, 'mutations'), (12, 'codon'), (11, 'amino'), (9, 'translational')]\n",
            "gi-2017-15-4-128.txt [(26, ':'), (22, 'rapa'), (22, 'B'), (18, ']'), (18, '['), (17, 'genome'), (16, ';'), (15, 'genes'), (15, 'block'), (14, 'GBS')]\n",
            "gi-2017-15-4-136.txt [(33, 'mutations'), (33, '%'), (23, 'OncoChase-AS'), (20, 'NGS'), (20, ':'), (17, 'genes'), (16, 'analysis'), (14, 'detected'), (14, ']'), (14, '[')]\n",
            "gi-2017-15-4-142.txt [(27, ':'), (22, ']'), (22, '['), (17, 'models'), (15, 'proinsulin'), (14, 'structures'), (14, 'leader'), (12, 'predicted'), (12, 'insulin'), (10, ';')]\n",
            "gi-2017-15-4-147.txt [(38, ':'), (36, ';'), (32, 'pharmacophore'), (30, 'APE1'), (25, 'hits'), (25, 'compounds'), (23, ']'), (23, '['), (16, '1'), (13, 'inhibitors')]\n",
            "gi-2017-15-4-156.txt [(33, 'genes'), (20, 'cancer'), (19, 'class'), (19, ':'), (17, 'classes'), (16, ']'), (16, '['), (15, 'breast'), (15, 'age'), (14, 'expression')]\n",
            "gi-2017-15-4-162.txt [(83, ']'), (83, '['), (32, 'amino'), (31, 'protein'), (28, 'metal'), (23, 'proteins'), (20, 'variables'), (19, 'prediction'), (19, 'acid'), (16, 'ion')]\n",
            "gi-2017-15-4-170.txt [(44, 'HOTAIR'), (34, ']'), (34, '['), (31, ':'), (27, 'gene'), (22, 'different'), (20, 'sequences'), (19, 'CpGIs'), (17, 'Table'), (16, 'methylation')]\n",
            "gi-2017-15-4-178.txt [(27, 'analysis'), (25, ':'), (20, ']'), (20, '['), (20, 'GWAS'), (20, '>'), (20, '<'), (17, 'data'), (15, 'IVAG'), (12, 'DEG')]\n",
            "gi-2018-16-1-1.txt [(6, ':'), (4, 'data'), (3, '’'), (3, 'substitution'), (3, 'rate'), (3, 'genetic'), (3, 'SNP'), (3, 'MS'), (3, 'Hanwoo'), (3, '&')]\n",
            "gi-2018-16-1-10.txt [(39, ':'), (22, 'imputation'), (21, 'markers'), (21, 'MS'), (18, 'SNP'), (17, 'accuracy'), (16, 'reference'), (15, 'used'), (14, 'genotypes'), (14, 'data')]\n",
            "gi-2018-16-1-14.txt [(50, 'substitution'), (47, 'Hanwoo'), (42, 'Holstein'), (32, 'rates'), (32, 'number'), (30, 'HSF'), (23, 'rate'), (23, ':'), (21, 'genes'), (20, 'SNPs')]\n",
            "gi-2018-16-1-2.txt [(77, ']'), (77, '['), (40, 'ORs'), (24, 'expressed'), (21, 'expression'), (20, 'human'), (18, 'cells'), (14, 'reported'), (13, 'mouse'), (13, 'et')]\n",
            "gni-10-1.txt [(15, 'DNA'), (10, 'genome'), (9, 'sequencing'), (8, 'Seoul'), (7, 'sequence'), (6, 'sequences'), (6, 'novo'), (6, 'de'), (6, 'assembly'), (5, 'expression')]\n",
            "gni-10-106.txt [(36, 'rate'), (29, 'pulse'), (19, ']'), (19, '['), (14, 'study'), (12, 'SNPs'), (12, '='), (11, '5'), (11, '3'), (10, 'near')]\n",
            "gni-10-110.txt [(28, 'variations'), (27, ']'), (27, '['), (22, 'SCN5A'), (21, 'AVB'), (16, 'study'), (14, '2'), (13, 'patients'), (13, 'genetic'), (13, 'gene')]\n",
            "gni-10-117.txt [(48, '%'), (38, 'sample'), (36, 'size'), (33, 'study'), (33, 'power'), (27, 'statistical'), (25, 'disease'), (24, '5'), (21, 'cases'), (21, 'SNP')]\n",
            "gni-10-123.txt [(59, 'gene'), (37, 'genes'), (27, 'set'), (23, 'p-values'), (22, 'best'), (22, 'GSA-SNP'), (21, 'scores'), (21, 'one'), (21, 'i-GSEA4GWAS'), (19, 'imputed')]\n",
            "gni-10-128.txt [(19, 'molecular'), (16, 'histamine'), (10, 'receptor'), (9, 'vibrational'), (9, 'molecule'), (9, ']'), (9, '['), (8, 'frequency'), (8, 'Tanimoto'), (7, 'ligand')]\n",
            "gni-10-133.txt [(84, 'populations'), (82, 'admixture'), (61, ']'), (61, '['), (45, 'population'), (35, 'genetic'), (31, 'studies'), (31, 'admixed'), (29, 'ancestry'), (21, 'history')]\n",
            "gni-10-145.txt [(49, ']'), (49, '['), (37, 'histone'), (32, 'modifications'), (27, 'FAIRE'), (26, 'regulatory'), (26, 'gene'), (26, 'elements'), (25, 'sites'), (25, 'chromatin')]\n",
            "gni-10-153.txt [(70, 'genes'), (58, 'KDO'), (52, ']'), (52, '['), (44, 'gene'), (41, 'IFN-β'), (36, '8-Br'), (31, 'expression'), (30, 'pathway'), (26, 'transcription')]\n",
            "gni-10-16.txt [(15, 'PKD2'), (10, 'renal'), (10, 'ADPKD'), (8, 'cell'), (8, 'PKD1'), (6, 'mutations'), (6, 'kidney'), (6, 'genes'), (6, ']'), (6, '[')]\n",
            "gni-10-167.txt [(32, 'polymorphisms'), (32, 'cancer'), (28, 'lung'), (25, 'AKT2'), (24, 'AKT3'), (23, '%'), (22, ']'), (22, '['), (12, 'promoter'), (11, 'using')]\n",
            "gni-10-175.txt [(99, 'CRC'), (81, 'GRS'), (71, 'risk'), (54, ']'), (54, '['), (50, 'study'), (38, 'history'), (36, 'women'), (35, 'men'), (34, 'family')]\n",
            "gni-10-184.txt [(52, 'cancer'), (38, ']'), (38, '['), (36, 'gastric'), (30, '%'), (27, 'risk'), (25, 'folate'), (23, ';'), (13, 'pathway'), (13, 'association')]\n",
            "gni-10-194.txt [(57, 'CNV'), (51, 'CNVs'), (42, 'algorithms'), (24, 'calls'), (24, 'SNP'), (23, '3'), (22, 'data'), (21, '2'), (19, 'detected'), (19, '%')]\n",
            "gni-10-200.txt [(45, 'cattle'), (30, '%'), (29, 'populations'), (25, 'Ethiopian'), (23, 'breeds'), (20, ']'), (20, '['), (19, 'Hanwoo'), (18, 'SNPs'), (15, 'markers')]\n",
            "gni-10-206.txt [(48, '%'), (33, 'species'), (18, 'Korea'), (17, 'shellfish'), (16, 'fish'), (13, ';'), (12, ']'), (12, '['), (12, ':'), (11, 'sequences')]\n",
            "gni-10-213.txt [(7, '’'), (6, 'University'), (5, 'application'), (4, 'review'), (4, 'group'), (4, 'Prof'), (4, 'Korea'), (4, ':'), (3, 'diseases'), (3, 'articles')]\n",
            "gni-10-214.txt [(38, 'variants'), (28, 'WES'), (16, ']'), (16, '['), (15, 'patients'), (14, 'genetic'), (13, 'variant'), (13, 'analysis'), (12, 'human'), (12, 'disease')]\n",
            "gni-10-220.txt [(27, ']'), (27, '['), (21, 'genome'), (21, 'GWAS'), (16, ':'), (15, 'sequence'), (15, 'SNPs'), (14, 'study'), (14, 'gene'), (13, 'many')]\n",
            "gni-10-226.txt [(70, ']'), (70, '['), (62, 'human'), (47, 'TEs'), (39, 'genome'), (36, 'Alu'), (35, 'elements'), (25, 'genomic'), (24, 'L1'), (20, 'could')]\n",
            "gni-10-23.txt [(17, 'datasets'), (14, 'genes'), (7, 'two'), (7, 'squamous'), (7, 'significant'), (7, 'oral'), (7, 'different'), (7, 'combined'), (7, 'cell'), (6, 'detecting')]\n",
            "gni-10-234.txt [(43, 'expression'), (32, ']'), (32, '['), (20, 'gene'), (18, 'eQTLs'), (16, 'genes'), (15, 'may'), (14, 'levels'), (14, 'genetic'), (14, 'et')]\n",
            "gni-10-239.txt [(70, 'GDM'), (41, 'T2DM'), (34, ']'), (34, '['), (29, 'women'), (25, 'variants'), (22, 'genetic'), (21, 'risk'), (16, 'insulin'), (16, 'increased')]\n",
            "gni-10-244.txt [(36, 'hypertension'), (36, 'genes'), (28, ']'), (28, '['), (17, ';'), (14, 'analysis'), (12, 'p'), (12, 'gene'), (12, 'association'), (12, 'associated')]\n",
            "gni-10-249.txt [(92, 'DNA'), (39, 'water'), (38, 'samples'), (30, 'amplification'), (26, ']'), (26, '['), (26, 'Tank'), (25, 'Prebiocide'), (19, 'microbial'), (17, 'fosmid')]\n",
            "gni-10-256.txt [(54, 'interactions'), (36, 'analysis'), (32, 'SNPs'), (28, ']'), (28, '['), (27, 'biological'), (24, 'graph'), (21, 'network'), (18, 'GxG'), (17, 'GMDR')]\n",
            "gni-10-263.txt [(32, 'data'), (15, 'clustering'), (12, 'QCanvas'), (8, 'heatmaps'), (8, ':'), (7, 'provides'), (6, 'program'), (6, 'patterns'), (6, 'experimental'), (6, 'November')]\n",
            "gni-10-266.txt [(18, 'compression'), (15, 'sequences'), (13, 'grammar'), (12, 'symbols'), (11, 'e'), (10, 'string'), (10, ':'), (9, 'new'), (9, 'JSequitur'), (8, 'symbol')]\n",
            "gni-10-271.txt [(25, 'patents'), (25, 'bioinformatics'), (12, 'patent'), (10, 'groups'), (9, 'subclasses'), (9, 'database'), (9, 'KIPRIS'), (8, 'G06F'), (8, ':'), (7, 'relevant')]\n",
            "gni-10-33.txt [(25, 'cancer'), (13, ':'), (11, 'genes'), (11, 'Cancer'), (9, 'mutations'), (9, 'gene'), (7, 'candidates'), (7, 'analysis'), (6, 'information'), (6, 'genomic')]\n",
            "gni-10-40.txt [(17, 'pressure'), (16, 'blood'), (12, 'siRNA'), (12, ']'), (12, '['), (12, 'CASZ1'), (7, 'Casz1'), (6, 'mRNA'), (5, 'vivo'), (5, 'mice')]\n",
            "gni-10-44.txt [(16, 'pattern'), (14, 'patterns'), (8, 'interesting'), (8, ']'), (8, '['), (6, 'sequences'), (6, 'mining'), (6, 'DNA'), (5, 'measure'), (5, 'information')]\n",
            "gni-10-51.txt [(17, 'sequences'), (15, 'frequent'), (9, 'DNA'), (8, 'sequence'), (8, 'contiguous'), (7, 'mining'), (5, 'patterns'), (5, 'pattern'), (5, 'maximal'), (5, ']')]\n",
            "gni-10-58.txt [(15, 'intron'), (14, 'sequences'), (10, 'introns'), (9, ':'), (8, '3n'), (7, 'tools'), (7, 'BLAT'), (6, 'three'), (6, 'plant'), (6, ']')]\n",
            "gni-10-65.txt [(10, 'CNV'), (9, 'CNVs'), (8, ':'), (7, ']'), (7, '['), (5, 'variation'), (5, 'number'), (5, 'disease'), (5, 'Database'), (4, 'studies')]\n",
            "gni-10-69.txt [(38, 'cancer'), (37, ']'), (37, '['), (19, 'sequencing'), (17, 'data'), (16, 'tumor'), (16, 'genome'), (14, 'could'), (13, 'heterogeneity'), (13, 'biomarkers')]\n",
            "gni-10-74.txt [(53, ']'), (53, '['), (40, 'splicing'), (36, 'cancer'), (33, 'alternative'), (29, 'splice'), (16, 'sites'), (16, 'exon'), (14, 'could'), (12, 'transcripts')]\n",
            "gni-10-81.txt [(58, ']'), (58, '['), (51, 'F'), (40, 'selection'), (31, 'population'), (28, 'locus'), (27, 'model'), (25, 'loci'), (25, 'estimates'), (22, 'Bayesian')]\n",
            "gni-10-88.txt [(65, 'levels'), (52, 'SNPs'), (49, 'associated'), (49, 'TG'), (49, 'HDLC'), (42, 'LPL'), (42, '2'), (41, ']'), (41, '['), (37, '=')]\n",
            "gni-10-9.txt [(12, ']'), (12, '['), (9, 'parvum'), (7, 'gene'), (7, 'C'), (6, 'transfer'), (6, 'CysQ'), (5, 'role'), (5, 'genes'), (5, 'enzymes')]\n",
            "gni-10-99.txt [(26, ']'), (26, '['), (21, 'study'), (19, 'risk'), (18, 'HDL-C'), (16, 'genetic'), (14, 'lipid'), (13, 'variants'), (12, 'using'), (12, 'TG')]\n",
            "gni-11-1.txt [(6, '’'), (5, 'Dr'), (4, 'articles'), (4, 'University'), (4, 'Korea'), (4, ':'), (3, 'reviews'), (3, 'issue'), (3, 'group'), (3, 'analysis')]\n",
            "gni-11-101.txt [(8, 'Korea'), (5, 'University'), (5, 'Prof'), (4, '’'), (4, 'review'), (4, 'metagenomics'), (4, ':'), (3, 'articles'), (3, 'Seoul'), (2, 'reports')]\n",
            "gni-11-102.txt [(118, ']'), (118, '['), (53, 'sequences'), (39, 'reads'), (35, 'sequencing'), (28, 'data'), (27, 'sequence'), (27, 'information'), (25, 'using'), (25, 'taxonomic')]\n",
            "gni-11-114.txt [(47, 'soil'), (40, 'microbial'), (39, 'enzymes'), (35, ']'), (35, '['), (33, 'novel'), (26, 'metagenome'), (23, 'metagenomic'), (19, 'DNA'), (18, 'genes')]\n",
            "gni-11-121.txt [(97, 'viral'), (65, ']'), (65, '['), (55, 'viruses'), (23, 'sequences'), (22, 'environmental'), (20, 'metagenomics'), (17, 'gut'), (16, 'ssDNA'), (16, 'DNA')]\n",
            "gni-11-129.txt [(32, 'OH'), (31, ']'), (31, '['), (19, 'BP'), (12, 'SNPs'), (11, 'Ansung'), (11, 'Ansan'), (10, '2'), (10, '1'), (10, '%')]\n",
            "gni-11-135.txt [(22, 'traits'), (22, 'gene'), (14, 'terms'), (14, 'GO'), (13, 'similarity'), (13, 'sets'), (13, ']'), (13, '['), (11, 'GWA'), (9, 'set')]\n",
            "gni-11-142.txt [(41, 'SVA'), (35, ']'), (35, '['), (26, ';'), (23, 'elements'), (21, 'human'), (20, 'element'), (19, 'gene'), (18, 'genes'), (16, 'SVA-F')]\n",
            "gni-11-149.txt [(30, 'levels'), (27, 'liver'), (24, ']'), (24, '['), (24, 'ALT'), (23, 'AST'), (20, 'association'), (14, 'enzyme'), (14, 'associated'), (10, 'Korea')]\n",
            "gni-11-15.txt [(38, ';'), (35, 'IFI6'), (34, 'HBV'), (29, 'rs2808426'), (29, ']'), (29, '['), (27, '%'), (22, 'SR'), (22, 'CLD'), (21, 'patients')]\n",
            "gni-11-155.txt [(42, 'structure'), (21, 'torsion'), (21, 'backbone'), (20, 'secondary'), (20, 'protein'), (18, 'angles'), (18, 'alignment'), (17, 'structural'), (16, 'comparison'), (15, 'logPr')]\n",
            "gni-11-161.txt [(27, 'Seoul'), (16, 'Korea'), (12, 'University'), (12, 'National'), (12, 'Medicine'), (9, 'Institute'), (9, '110-799'), (6, 'School'), (6, 'Research'), (6, 'Health')]\n",
            "gni-11-163.txt [(7, 'articles'), (5, 'review'), (5, 'genomics'), (4, 'cancer'), (4, ':'), (3, 'two'), (2, 'year'), (2, 'next'), (2, 'issue'), (2, 'informatics')]\n",
            "gni-11-164.txt [(70, 'DNA'), (59, ']'), (59, '['), (47, 'cancer'), (46, 'HP1'), (40, 'histone'), (34, 'epigenetic'), (25, 'tumor'), (24, 'genes'), (24, 'gene')]\n",
            "gni-11-174.txt [(55, 'cancer'), (37, ']'), (37, '['), (26, 'clinical'), (25, 'tumor'), (22, 'genomics'), (21, 'treatment'), (15, 'lung'), (14, 'patients'), (14, 'patient')]\n",
            "gni-11-180.txt [(49, 'cancer'), (27, 'liver'), (24, ']'), (24, '['), (23, 'data'), (18, 'expression'), (16, 'genes'), (14, 'number'), (13, 'genomic'), (13, 'gene')]\n",
            "gni-11-186.txt [(36, 'data'), (30, 'clinical'), (22, ']'), (22, '['), (17, 'research'), (16, 'information'), (14, 'EMR'), (13, 'systems'), (10, 'informatics'), (10, 'database')]\n",
            "gni-11-191.txt [(66, ']'), (66, '['), (51, 'variants'), (50, 'variant'), (33, 'data'), (30, ':'), (28, 'used'), (28, 'tools'), (24, 'sequence'), (24, 'quality')]\n",
            "gni-11-2.txt [(32, ']'), (32, '['), (27, 'ENCODE'), (17, 'chromatin'), (16, 'regions'), (15, 'project'), (14, 'DNA'), (12, 'histone'), (12, 'genome'), (12, 'data')]\n",
            "gni-11-200.txt [(107, ']'), (107, '['), (60, 'methods'), (50, 'network'), (27, 'genes'), (25, 'gene'), (25, 'data'), (24, 'networks'), (24, 'method'), (23, 'matrix')]\n",
            "gni-11-211.txt [(37, 'biobanks'), (36, ']'), (36, '['), (30, 'biospecimens'), (28, 'research'), (17, 'Korea'), (16, 'samples'), (16, 'researchers'), (16, 'data'), (14, 'genetic')]\n",
            "gni-11-218.txt [(49, 'genetic'), (44, 'human'), (28, ';'), (28, ':'), (26, 'Declaration'), (16, 'research'), (14, 'ethical'), (14, 'Article'), (13, 'testing'), (13, 'genome')]\n",
            "gni-11-224.txt [(56, ']'), (56, '['), (56, 'RAGE'), (30, 'ligands'), (19, 'renal'), (18, 'diseases'), (18, 'AGEs'), (15, 'including'), (13, 'cell'), (11, 'kidney')]\n",
            "gni-11-230.txt [(85, 'humans'), (71, ']'), (71, '['), (63, 'Neanderthals'), (60, 'modern'), (44, 'genome'), (38, 'human'), (33, 'sequence'), (22, 'species'), (22, 'Neanderthal')]\n",
            "gni-11-239.txt [(46, 'samples'), (38, 'cancer'), (28, 'mutations'), (25, 'genes'), (24, 'mutation'), (20, 'data'), (18, 'metastatic'), (16, 'frequency'), (15, '’'), (15, 'mutant')]\n",
            "gni-11-24.txt [(38, 'somatic'), (31, 'mutations'), (26, ':'), (24, 'genes'), (21, 'samples'), (21, 'LOH'), (18, 'used'), (18, 'network'), (16, 'mutation'), (14, 'sample')]\n",
            "gni-11-245.txt [(113, 'cells'), (44, 'H460R'), (38, 'IR'), (33, ']'), (33, '['), (25, 'SNTB2'), (24, 'parental'), (24, '%'), (19, 'cell'), (17, 'response')]\n",
            "gni-11-254.txt [(57, 'g'), (50, 'MRP2'), (32, ']'), (32, '['), (29, 'patients'), (29, 'ADRs'), (29, '%'), (25, ';'), (23, 'cells'), (23, 'VPA')]\n",
            "gni-11-263.txt [(67, 'BMR'), (41, ']'), (41, '['), (39, 'children'), (30, 'KLF5'), (28, 'genotype'), (24, 'muscle'), (20, 'TT'), (19, 'group'), (19, '=')]\n",
            "gni-11-272.txt [(29, 'Flavobacterium'), (23, 'genes'), (21, ']'), (21, '['), (20, 'gene'), (19, 'housekeeping'), (17, 'species'), (15, 'F'), (14, '16S'), (13, 'rRNA')]\n",
            "gni-11-277.txt [(41, 'body'), (28, 'markers'), (27, 'fluid'), (17, ']'), (17, '['), (17, 'NanoString'), (16, 'using'), (16, 'identification'), (12, 'mRNA'), (12, 'RNA')]\n",
            "gni-11-282.txt [(35, 'molecular'), (23, ']'), (23, '['), (14, 'EVA'), (13, 'vibrational'), (13, 'ligands'), (12, 'descriptor'), (12, 'CIMVF'), (11, 'AdoR'), (10, 'similarity')]\n",
            "gni-11-289.txt [(15, 'information'), (15, 'HPV'), (14, 'protein'), (13, ']'), (13, '['), (12, ';'), (11, ':'), (9, 'strains'), (7, 'genome'), (6, 'using')]\n",
            "gni-11-34.txt [(18, 'AML'), (15, 'somatic'), (12, 'using'), (12, 'patients'), (11, 'mutations'), (11, 'KRAS'), (9, 'samples'), (9, 'mutation'), (8, 'disease'), (8, 'detected')]\n",
            "gni-11-38.txt [(29, 'genes'), (25, ']'), (25, '['), (20, 'samples'), (20, 'concept'), (17, 'mutated'), (15, 'mutations'), (15, 'AML'), (13, 'sample'), (12, 'gene')]\n",
            "gni-11-46.txt [(27, ']'), (27, '['), (19, 'NK-AML'), (17, 'SNVs'), (15, '%'), (14, 'AML'), (14, ';'), (11, 'risk'), (11, 'gene'), (10, 'study')]\n",
            "gni-11-52.txt [(14, ']'), (14, '['), (11, '’'), (11, '‘'), (11, 'domain'), (11, 'DNAJ/K'), (10, 'DNAJ'), (9, 'EDD'), (8, 'proteins'), (8, 'Seoul')]\n",
            "gni-11-55.txt [(15, 'siMacro'), (13, 'siRNA'), (10, 'plate'), (10, 'data'), (8, 'wells'), (8, 'Excel'), (7, 'screens'), (7, 'cell'), (7, 'bad'), (7, ']')]\n",
            "gni-11-58.txt [(3, 'Vibrational'), (3, 'Patterns'), (3, 'Molecular'), (3, 'Histamine'), (3, 'Frequency'), (3, 'Figure'), (3, 'Antagonists'), (3, 'Agonists'), (3, ':'), (2, 'ReceptorsS')]\n",
            "gni-11-59.txt [(7, 'issue'), (6, '’'), (5, ':'), (3, 'Informatics'), (3, '&'), (2, 'review'), (2, 'retraction'), (2, 'group'), (2, 'facial'), (2, 'chromatin')]\n",
            "gni-11-60.txt [(58, ']'), (58, '['), (30, 'ENCODE'), (27, 'data'), (24, 'chromatin'), (24, 'RNA'), (19, 'gene'), (17, 'used'), (17, 'expression'), (17, 'RNAs')]\n",
            "gni-11-68.txt [(84, 'ATC'), (66, ']'), (66, '['), (41, 'thyroid'), (36, '%'), (32, 'cancer'), (26, 'et'), (26, 'al'), (22, 'genes'), (19, 'found')]\n",
            "gni-11-7.txt [(41, 'IHEC'), (39, 'epigenome'), (33, ']'), (33, '['), (29, 'cells'), (26, 'data'), (18, 'reference'), (18, 'epigenomes'), (18, 'disease'), (18, 'cell')]\n",
            "gni-11-76.txt [(66, ']'), (66, '['), (32, 'bacterial'), (22, 'RNA'), (20, 'transcription'), (19, 'transcriptome'), (19, 'sRNAs'), (17, 'RNA-seq'), (15, 'transcripts'), (15, 'sequencing')]\n",
            "gni-11-83.txt [(60, 'facial'), (45, ']'), (45, '['), (42, 'genetic'), (35, 'width'), (35, '='), (33, 'measurements'), (30, 'study'), (26, 'height'), (25, 'face')]\n",
            "gni-11-93.txt [(37, 'SNPs'), (27, ']'), (27, '['), (19, 'color'), (18, 'hair'), (11, 'study'), (10, 'forensic'), (10, 'eye'), (9, 'used'), (9, 'prediction')]\n",
            "gni-11-97.txt [(3, 'Technology'), (3, ':'), (2, 'Widest'), (2, 'Water'), (2, 'Tara'), (2, 'Stuecker3'), (2, 'Strategy'), (2, 'Station'), (2, 'Spectrum'), (2, 'Space')]\n",
            "gni-11-98.txt [(6, 'Korea'), (6, ':'), (4, 'grant'), (4, 'Exome'), (3, 'Sequencing'), (3, 'Research'), (3, 'Interpretation'), (3, 'Blood'), (2, 'nos'), (2, 'Yun-Gyeong')]\n",
            "gni-11-99.txt [(5, 'National'), (4, 'Informatics'), (3, 'Whole-Exome'), (3, 'Using'), (3, 'Somatic'), (3, 'Sequencing'), (3, 'Seoul'), (3, 'Myeloid'), (3, 'Mutations'), (3, 'Leukemia')]\n",
            "gni-12-1.txt [(5, 'group'), (4, '’'), (4, 'Informatics'), (4, ':'), (4, '&'), (3, 'reports'), (3, 'issue'), (3, 'genetic'), (3, 'University'), (3, 'Genomics')]\n",
            "gni-12-105.txt [(109, 'Peg3'), (73, 'genes'), (65, 'domain'), (60, ']'), (60, '['), (35, 'imprinted'), (27, 'also'), (24, 'gene'), (23, 'Peg3-DMR'), (22, 'genomic')]\n",
            "gni-12-114.txt [(41, 'BMD'), (36, 'SNPs'), (35, ']'), (35, '['), (22, 'loci'), (16, 'associated'), (12, 'sites'), (12, 'previous'), (12, 'analysis'), (11, 'study')]\n",
            "gni-12-12.txt [(124, ']'), (124, '['), (56, 'EMT'), (43, 'TMPRSS4'), (42, 'cancer'), (40, 'expression'), (40, 'TM4SF5'), (30, 'cells'), (29, 'cell'), (23, 'invasion')]\n",
            "gni-12-121.txt [(40, 'medication'), (38, 'adherence'), (32, ']'), (32, '['), (19, 'personality'), (16, 'patients'), (16, 'disease'), (15, 'study'), (15, 'genetic'), (14, 'chronic')]\n",
            "gni-12-127.txt [(29, 'gastritis'), (25, ']'), (25, '['), (22, 'patients'), (18, 'factors'), (17, 'cancer'), (14, 'gastric'), (14, 'SNPs'), (12, 'study'), (12, 'males')]\n",
            "gni-12-135.txt [(4, '’'), (4, 'reviewed'), (4, ':'), (3, 'University'), (3, 'Informatics'), (3, 'Dr'), (3, 'Award'), (3, '&'), (2, 'variants'), (2, 'treatment')]\n",
            "gni-12-136.txt [(67, ']'), (67, '['), (65, 'replication'), (45, 'DNA'), (43, 'human'), (42, 'genomic'), (42, 'CNV'), (37, 'timing'), (34, 'genome'), (24, 'CNVs')]\n",
            "gni-12-145.txt [(30, 'state'), (29, ']'), (29, '['), (28, 'HMM'), (18, 'HMMs'), (17, '’'), (17, '‘'), (16, 'states'), (14, 'hidden'), (14, 'epigenetic')]\n",
            "gni-12-151.txt [(44, 'FDA'), (38, '’'), (38, ']'), (38, '['), (35, 'genetic'), (28, 'testing'), (22, 'DTC'), (18, 'tests'), (18, '23andMe'), (16, 'clinical')]\n",
            "gni-12-156.txt [(82, ']'), (82, '['), (53, 'cell'), (49, 'cancer'), (46, 'nimbolide'), (44, 'cells'), (22, 'expression'), (22, 'Nimbolide'), (21, 'apoptosis'), (19, 'protein')]\n",
            "gni-12-165.txt [(29, 'SNPs'), (28, 'haplotype'), (26, 'LD'), (22, 'blocks'), (21, 'significant'), (21, ']'), (21, '['), (18, 'data'), (17, 'variants'), (17, 'BMI')]\n",
            "gni-12-171.txt [(28, 'methylation'), (25, 'DNA'), (23, 'expression'), (22, 'genes'), (18, ']'), (18, '['), (18, 'SFRP1'), (16, 'survival'), (16, 'samples'), (16, 'gene')]\n",
            "gni-12-181.txt [(20, 'using'), (19, 'genes'), (18, 'epistatic'), (16, ']'), (16, '['), (15, 'DM'), (14, 'SNPs'), (14, 'HT'), (14, 'CAD'), (13, ';')]\n",
            "gni-12-187.txt [(53, 'SNPs'), (51, ']'), (51, '['), (32, 'METS'), (30, 'GWAS'), (16, 'reported'), (16, 'functional'), (16, ':'), (16, '1'), (13, '×')]\n",
            "gni-12-195.txt [(58, ']'), (58, '['), (34, 'MetS'), (27, 'associated'), (21, 'pathways'), (19, 'gene'), (18, 'analysis'), (17, 'signaling'), (15, 'protein'), (14, 'obesity')]\n",
            "gni-12-2.txt [(91, ']'), (91, '['), (44, 'DNA'), (32, ':'), (26, 'regions'), (25, 'chromatin'), (24, 'specific'), (24, 'cell'), (23, 'sequencing'), (22, 'genome')]\n",
            "gni-12-203.txt [(26, 'hepatitis'), (13, ']'), (13, '['), (9, 'Table'), (9, 'SNPs'), (9, ';'), (8, '%'), (6, 'variables'), (6, 'study'), (6, 'dataset')]\n",
            "gni-12-208.txt [(102, 'population'), (80, 'N'), (54, ']'), (54, '['), (52, 'current'), (38, 'estimates'), (37, 'effective'), (35, 'recent'), (33, 'data'), (31, 'size')]\n",
            "gni-12-21.txt [(73, 'patients'), (46, 'data'), (32, 'visualization'), (31, ']'), (31, '['), (29, 'patient'), (28, 'Fig'), (27, 'genes'), (26, 'treatment'), (24, 'analysis')]\n",
            "gni-12-216.txt [(34, 'network'), (33, ']'), (33, '['), (21, 'SNP'), (20, 'gastritis'), (20, 'SNPs'), (18, 'pairs'), (17, 'interactions'), (17, 'disease'), (15, ';')]\n",
            "gni-12-225.txt [(30, 'platelet'), (30, ']'), (30, '['), (18, 'count'), (12, 'counts'), (11, 'associated'), (10, 'study'), (10, 'SNPs'), (10, '='), (9, 'known')]\n",
            "gni-12-231.txt [(30, 'uric'), (30, 'acid'), (29, 'levels'), (24, 'ABCG2'), (20, 'association'), (20, ']'), (20, '['), (15, 'subjects'), (15, 'study'), (13, 'individuals')]\n",
            "gni-12-236.txt [(23, 'BMI'), (20, ']'), (20, '['), (20, 'SNPs'), (15, 'study'), (13, 'subjects'), (13, 'glucose'), (11, 'fasting'), (11, 'MAGIC'), (9, 'insulin')]\n",
            "gni-12-240.txt [(35, ']'), (35, '['), (27, 'HNF1B'), (26, 'GLUT2'), (22, 'mutation'), (17, 'insulin'), (16, 'diabetes'), (16, 'cells'), (15, 'expression'), (14, 'wild-type')]\n",
            "gni-12-247.txt [(31, 'COX-2'), (30, 'cell'), (22, 'osteosarcoma'), (22, 'cells'), (22, ']'), (22, '['), (16, 'genes'), (14, 'proliferation'), (13, '2'), (12, '#')]\n",
            "gni-12-254.txt [(43, 'matrix'), (38, 'SNPs'), (37, 'effects'), (26, ']'), (26, '['), (26, 'BLUP'), (24, 'SNP-SNP'), (23, 'genetic'), (22, 'used'), (19, 'values')]\n",
            "gni-12-261.txt [(96, 'miRNAs'), (45, ']'), (45, '['), (41, 'sequences'), (39, 'miRNA'), (30, 'REs'), (27, 'precursor'), (24, 'type'), (18, 'identified'), (17, 'form')]\n",
            "gni-12-268.txt [(38, 'protein'), (34, 'mutase'), (33, 'phosphoglucosamine'), (31, ']'), (31, '['), (28, 'structure'), (27, 'model'), (18, 'analysis'), (13, 'residues'), (13, 'proteins')]\n",
            "gni-12-276.txt [(48, 'rpoB'), (45, ']'), (45, '['), (26, 'RIF'), (16, 'protein'), (15, '%'), (13, 'using'), (13, 'model'), (13, 'MTB'), (13, 'Fig')]\n",
            "gni-12-283.txt [(48, 'kir62'), (33, 'model'), (27, 'models'), (20, 'mutant'), (18, 'Supplementary'), (18, 'Fig'), (17, ']'), (17, '['), (16, 'phytochemicals'), (15, '1')]\n",
            "gni-12-289.txt [(23, 'mutations'), (18, 'MAP'), (13, '‘'), (13, 'software'), (13, ']'), (13, '['), (12, 'information'), (10, '’'), (10, 'tab'), (10, 'recurrent')]\n",
            "gni-12-35.txt [(42, '”'), (42, '“'), (40, ']'), (40, '['), (25, 'SNPs'), (18, 'ethnicity-specific'), (17, 'semantic'), (17, 'records'), (16, ':'), (15, 'genes')]\n",
            "gni-12-42.txt [(34, 'ethnic'), (30, 'SNPs'), (29, ']'), (29, '['), (27, 'groups'), (27, ':'), (23, 'SNP'), (20, 'Pan-Asian'), (19, 'populations'), (17, 'genetic')]\n",
            "gni-12-49.txt [(7, '’'), (5, 'group'), (5, 'Dr'), (5, '&'), (4, 'NGS'), (4, ':'), (3, 'sequencing'), (3, 'lung'), (3, 'genetic'), (3, 'cancer')]\n",
            "gni-12-50.txt [(29, 'cancer'), (29, '%'), (28, 'sequencing'), (27, 'target'), (23, 'DNA'), (21, 'using'), (20, 'samples'), (20, 'sample'), (19, 'L'), (18, ';')]\n",
            "gni-12-58.txt [(25, 'AERD'), (19, 'TEC'), (13, 'subjects'), (12, 'polymorphisms'), (11, 'aspirin'), (11, ':'), (10, 'SNPs'), (9, 'study'), (9, ']'), (9, '[')]\n",
            "gni-12-64.txt [(39, 'E6'), (36, ']'), (36, '['), (20, 'HPV'), (19, 'protein'), (16, 'study'), (15, 'cervical'), (15, 'cancer'), (15, 'HPV-16'), (14, 'docking')]\n",
            "gni-12-71.txt [(111, 'tRNA'), (40, ']'), (40, '['), (27, 'G'), (25, 'tRNAs'), (25, 'position'), (22, 'residue'), (20, 'modification'), (20, 'N'), (17, '18th')]\n",
            "gni-12-76.txt [(16, 'proteins'), (14, ']'), (14, '['), (13, 'putative'), (9, '”'), (9, '“'), (9, 'annotation'), (9, ':'), (8, 'databases'), (8, 'SFannotation')]\n",
            "gni-12-79.txt [(9, 'TEs'), (7, 'genomic'), (6, '’'), (5, 'University'), (5, 'DNA'), (5, '&'), (4, 'Kim'), (4, 'FBI'), (4, 'Dr'), (4, ':')]\n",
            "gni-12-80.txt [(64, 'DNA'), (48, ']'), (48, '['), (27, 'transposition'), (25, 'target'), (25, 'TE'), (25, 'FBI'), (23, 'replication'), (20, 'ends'), (17, 'may')]\n",
            "gni-12-87.txt [(116, ']'), (116, '['), (86, 'genome'), (61, 'genomes'), (48, 'retrotransposons'), (44, 'species'), (40, 'DNA'), (39, 'elements'), (37, 'size'), (34, 'plant')]\n",
            "gni-12-98.txt [(60, ']'), (60, '['), (31, 'genome'), (28, 'human'), (24, 'TEs'), (21, 'insertions'), (21, 'evolution'), (20, 'Alu'), (18, 'may'), (18, 'elements')]\n",
            "gni-13-1.txt [(4, ':'), (3, '’'), (3, 'important'), (3, 'genome'), (3, 'baumannii'), (3, 'Acinetobacter'), (2, 'silico'), (2, 'pathogen'), (2, 'one'), (2, 'iron')]\n",
            "gni-13-102.txt [(79, 'C'), (72, ']'), (72, '['), (67, 'species'), (56, 'Crocodylus'), (44, 'chromosome'), (43, 'chromosomes'), (32, 'karyotype'), (32, 'crocodilians'), (31, '1')]\n",
            "gni-13-112.txt [(83, 'introns'), (56, ']'), (56, '['), (32, 'genes'), (19, 'intron'), (19, 'gene'), (18, '’'), (15, 'splicing'), (15, 'expression'), (14, 'regions')]\n",
            "gni-13-119.txt [(62, ']'), (62, '['), (28, 'reference'), (27, 'data'), (27, 'RNA-seq'), (25, 'transcripts'), (25, 'reads'), (22, 'transcriptome'), (21, 'read'), (20, 'used')]\n",
            "gni-13-126.txt [(48, 'T1DM'), (36, 'fulminant'), (24, ']'), (24, '['), (20, 'HLA'), (19, 'position'), (19, 'haplotype'), (16, 'cases'), (14, 'associated'), (13, 'acid')]\n",
            "gni-13-132.txt [(27, 'G64'), (26, 'cancers'), (15, 'samples'), (14, 'expression'), (14, 'cancer'), (13, 'groups'), (13, 'different'), (13, ':'), (12, 'survival'), (12, 'patient')]\n",
            "gni-13-137.txt [(47, 'selection'), (43, ']'), (43, '['), (33, 'positive'), (33, 'genes'), (26, 'using'), (20, 'XPEHH'), (19, 'XPCLR'), (18, ':'), (16, 'SNP')]\n",
            "gni-13-146.txt [(34, 'SNPs'), (32, 'traits'), (30, 'milk'), (29, 'h'), (21, 'variance'), (18, '%'), (16, '='), (15, 'genetic'), (15, 'chromosome'), (15, 'Holstein')]\n",
            "gni-13-15.txt [(44, 'ER'), (41, 'KR'), (33, 'using'), (33, ']'), (33, '['), (26, 'structures'), (25, 'FASN'), (24, 'domains'), (24, ':'), (21, 'models')]\n",
            "gni-13-152.txt [(32, 'promoter'), (16, 'elements'), (11, ']'), (11, '['), (11, ':'), (9, 'states'), (9, 'chromatin'), (9, 'blocks'), (9, 'ChromHMM'), (8, 'regions')]\n",
            "gni-13-156.txt [(5, 'Korea'), (4, 'miR-27a'), (4, 'Potential'), (4, 'Kim12'), (4, 'Gastric'), (4, 'Diagnostic'), (4, 'Biomarker'), (3, 'andPrognostic'), (3, 'Research'), (3, 'Daejeon')]\n",
            "gni-13-2.txt [(43, 'baumannii'), (34, ']'), (34, '['), (28, 'gene'), (26, 'acinetobactin'), (23, 'iron'), (21, 'cluster'), (12, 'entA'), (11, 'biosynthesis'), (11, 'acquisition')]\n",
            "gni-13-25.txt [(4, 'sequencing'), (4, ':'), (3, '’'), (3, 'human'), (3, 'group'), (3, 'genome'), (3, 'Dr'), (2, 'variants'), (2, 'substantial'), (2, 'next-generation')]\n",
            "gni-13-26.txt [(73, 'nc886'), (48, ']'), (48, '['), (30, 'PKR'), (27, 'cancer'), (19, ':'), (15, 'expression'), (13, 'cells'), (12, 'data'), (12, 'Pol')]\n",
            "gni-13-31.txt [(51, 'depth'), (49, 'sequencing'), (34, ']'), (34, '['), (32, 'SNPs'), (30, 'using'), (22, 'variants'), (22, 'exome'), (22, 'Fig'), (22, '120×')]\n",
            "gni-13-40.txt [(37, 'chromatin'), (30, ']'), (30, '['), (26, 'expression'), (17, 'gene'), (16, 'regulatory'), (16, 'genetic'), (15, 'number'), (14, 'yeast'), (13, 'human')]\n",
            "gni-13-45.txt [(42, 'miRNAs'), (31, 'gene'), (26, 'HOXA9'), (22, 'binding'), (22, 'AGO'), (22, ';'), (21, 'LEU'), (19, 'protein'), (18, 'molecular'), (18, 'mir-126')]\n",
            "gni-13-53.txt [(37, 'proteins'), (35, 'protein'), (32, ']'), (32, '['), (28, 'HPs'), (15, ':'), (14, 'cholerae'), (13, 'V'), (12, 'used'), (11, 'functional')]\n",
            "gni-13-60.txt [(49, ']'), (49, '['), (38, 'E6'), (32, 'protein'), (23, 'cancer'), (23, 'HPV'), (18, 'cervical'), (18, 'HPV18'), (17, 'binding'), (16, 'structure')]\n",
            "gni-13-69.txt [(7, '’'), (6, ':'), (5, 'miRNAs'), (4, 'gene'), (4, 'expression'), (3, '‘'), (3, 'regulatory'), (3, 'plasma'), (3, 'network'), (3, 'issue')]\n",
            "gni-13-7.txt [(52, 'data'), (32, ']'), (32, '['), (28, 'application'), (27, 'genomic'), (22, 'genome'), (22, 'GIS'), (18, 'ontology'), (18, 'gene'), (16, 'users')]\n",
            "gni-13-70.txt [(40, 'cancer'), (39, 'gastric'), (36, 'expression'), (35, 'miR-27a'), (29, 'plasma'), (27, ']'), (27, '['), (18, 'samples'), (15, 'patients'), (15, 'miRNAs')]\n",
            "gni-13-76.txt [(49, 'genes'), (33, 'expression'), (31, 'regulatory'), (26, 'target'), (25, 'insulin'), (21, 'network'), (20, 'TFs'), (18, 'gene'), (18, ']'), (18, '[')]\n",
            "gni-13-81.txt [(21, 'transgenic'), (21, 'rice'), (20, 'NGS'), (18, 'DNA'), (16, 'vector'), (16, ']'), (16, '['), (15, 'reads'), (15, 'GM'), (12, 'using')]\n",
            "gni-13-86.txt [(24, 'cancer'), (17, 'data'), (17, 'Korean'), (15, 'database'), (14, ':'), (12, 'patients'), (11, 'search'), (10, 'system'), (10, 'n'), (10, 'Korea')]\n",
            "gni-13-90.txt [(19, 'species'), (18, ']'), (18, '['), (10, 'biodiversity'), (8, 'richness'), (8, 'index'), (8, 'VBioindex'), (7, 'indices'), (7, 'evenness'), (7, ':')]\n",
            "gni-13-93.txt [(6, '’'), (6, 'University'), (5, 'group'), (5, ':'), (4, 'non-coding'), (4, 'introns'), (4, 'Dr'), (3, 'transcriptome'), (3, 'tRFs'), (3, 'sequencing')]\n",
            "gni-13-94.txt [(74, ']'), (74, '['), (61, 'tRFs'), (47, 'tRNA'), (24, 'series'), (19, 'tRF-5'), (16, 'tRNAs'), (16, 'cells'), (14, 'RNA'), (13, 'ncRNAs')]\n",
            "gni-14-1.txt [(6, 'Korea'), (6, ':'), (5, '’'), (5, 'Dr'), (4, 'protein'), (4, 'group'), (3, 'platforms'), (3, 'University'), (2, 'spectrometry-based'), (2, 'role')]\n",
            "gni-14-104.txt [(51, ']'), (51, '['), (43, 'NS3'), (40, 'protein'), (39, 'ZIKV'), (22, ';'), (16, 'structure'), (13, 'drugs'), (12, 'residues'), (12, 'drug')]\n",
            "gni-14-112.txt [(49, 'miRNAs'), (44, 'ALA'), (40, 'genes'), (40, 'LEU'), (34, ':'), (31, 'AGO'), (29, 'protein'), (29, 'VAL'), (28, 'cancer'), (25, 'TYR')]\n",
            "gni-14-12.txt [(51, 'neuropeptides'), (45, ']'), (45, '['), (30, 'peptide'), (26, 'peptides'), (17, 'MS'), (16, 'neuropeptide'), (13, 'including'), (12, 'identification'), (12, 'also')]\n",
            "gni-14-125.txt [(66, 'HPs'), (59, 'proteins'), (55, ']'), (55, '['), (31, 'protein'), (25, 'function'), (24, 'virulence'), (24, 'predicted'), (21, 'showing'), (20, 'used')]\n",
            "gni-14-137.txt [(7, 'genetic'), (7, 'data'), (6, 'University'), (6, ':'), (5, 'analysis'), (5, 'Dr'), (4, '’'), (4, 'diseases'), (4, 'common'), (4, 'Seoul')]\n",
            "gni-14-138.txt [(53, 'prediction'), (34, 'KARE'), (32, 'set'), (32, ']'), (32, '['), (32, 'SNPs'), (24, 'risk'), (24, '='), (21, 'models'), (21, 'dataset')]\n",
            "gni-14-149.txt [(44, 'prediction'), (43, 'SNPs'), (28, ']'), (28, '['), (27, 'using'), (22, 'model'), (17, 'quantitative'), (17, 'models'), (17, 'SLR'), (15, 'risk')]\n",
            "gni-14-160.txt [(25, ']'), (25, '['), (24, 'BOOST'), (23, 'interactions'), (21, 'interaction'), (19, 'SNPs'), (17, 'model'), (16, 'pairs'), (16, 'gene-gene'), (16, 'T2D')]\n",
            "gni-14-166.txt [(27, 'AFT-MDR'), (26, 'effect'), (24, 'model'), (23, 'method'), (22, 'power'), (20, 'survival'), (20, 'proposed'), (20, 'gene-gene'), (19, 'time'), (18, 'main')]\n",
            "gni-14-173.txt [(33, 'two'), (31, 'size'), (30, 'SZ'), (29, 'studies'), (29, 'methods'), (26, 'model'), (26, 'effect'), (26, 'IVW'), (24, 'sample'), (24, ']')]\n",
            "gni-14-181.txt [(25, 'association'), (25, ']'), (25, '['), (22, 'phenotype'), (19, 'GIDS'), (17, '2'), (16, 'SNPs'), (14, '1'), (13, 'composite'), (13, 'classes')]\n",
            "gni-14-187.txt [(97, 'genes'), (80, 'group'), (72, 'genetic'), (52, 'sites'), (52, 'data'), (45, 'test'), (37, 'lasso'), (36, 'gene'), (34, 'selection'), (33, 'testing')]\n",
            "gni-14-196.txt [(70, 'haplotype'), (64, 'blocks'), (50, 'block'), (44, ']'), (44, '['), (40, '%'), (31, 'markers'), (30, 'LD'), (28, 'methods'), (27, 'dataset')]\n",
            "gni-14-2.txt [(68, ']'), (68, '['), (48, 'DEPs'), (34, 'protein'), (33, 'data'), (30, 'analysis'), (29, 'network'), (29, 'cellular'), (28, 'selected'), (25, 'database')]\n",
            "gni-14-20.txt [(73, 'authorization'), (59, 'access'), (55, 'data'), (43, 'information'), (43, 'IABio'), (38, 'server'), (35, 'database'), (29, 'service'), (25, 'token'), (24, 'used')]\n",
            "gni-14-205.txt [(19, 'NHL'), (17, 'lymphoma'), (16, 'IL10'), (15, 'polymorphisms'), (14, ']'), (14, '['), (13, 'patients'), (11, 'study'), (10, 'Cancer'), (9, 'using')]\n",
            "gni-14-211.txt [(37, 'sQTLs'), (29, ']'), (29, '['), (21, 'trait-associated'), (20, 'Arabidopsis'), (17, 'thaliana'), (17, 'exon'), (15, 'SNPs'), (12, 'result'), (12, 'genotype')]\n",
            "gni-14-216.txt [(50, 'CNV'), (28, ']'), (28, '['), (22, 'regions'), (20, 'study'), (18, 'region'), (16, 'identified'), (16, ':'), (13, 'risk'), (13, 'associated')]\n",
            "gni-14-222.txt [(32, 'kimchi'), (27, 'data'), (15, ']'), (15, '['), (14, 'information'), (13, 'knowledge'), (12, 'network'), (11, 'semantic'), (11, 'biological'), (11, 'Fig')]\n",
            "gni-14-230.txt [(35, 'cattle'), (33, 'breeds'), (24, 'LD'), (24, 'Korean'), (21, ']'), (21, '['), (20, 'study'), (20, 'population'), (20, 'Ne'), (17, 'size')]\n",
            "gni-14-234.txt [(50, 'expression'), (38, 'genes'), (32, 'gene'), (29, 'tissues'), (29, ']'), (29, '['), (26, 'features'), (24, 'significant'), (24, 'codon'), (23, 'level')]\n",
            "gni-14-241.txt [(63, ']'), (63, '['), (57, 'petrii'), (56, 'drug'), (55, 'B'), (50, 'proteins'), (34, 'target'), (34, 'lpxD'), (32, 'protein'), (23, ':')]\n",
            "gni-14-255.txt [(42, 'proteins'), (39, ']'), (39, '['), (26, 'nucleatum'), (25, 'protein'), (22, ':'), (20, 'F'), (17, 'essential'), (16, 'http'), (16, 'drug')]\n",
            "gni-14-29.txt [(36, 'sequences'), (31, 'msDNA'), (13, 'palindrome'), (13, ']'), (13, '['), (12, ':'), (10, '2'), (9, 'using'), (9, 'set'), (9, 'hits')]\n",
            "gni-14-34.txt [(37, 'samples'), (29, 'methylation'), (27, 'markers'), (26, 'data'), (24, ']'), (24, '['), (21, 'datasets'), (18, 'chromosome'), (16, 'patterns'), (16, 'age')]\n",
            "gni-14-41.txt [(6, ':'), (5, '’'), (4, 'group'), (4, 'diseases'), (4, 'University'), (4, 'Korea'), (4, 'Dr'), (3, 'reported'), (3, 'methylation'), (3, 'Seoul')]\n",
            "gni-14-42.txt [(22, ']'), (22, '['), (19, 'NGS'), (13, 'ultra-rare'), (13, 'diseases'), (11, 'mutations'), (8, 'medicine'), (8, ':'), (7, 'treatment'), (7, 'precision')]\n",
            "gni-6-1-1.txt [(41, 'et'), (41, 'al'), (33, 'CNVs'), (24, 'genome'), (18, 'CNV'), (17, 'variants'), (17, 'number'), (16, 'platforms'), (15, ';'), (14, '2006')]\n",
            "gni-6-1-14.txt [(64, 'et'), (64, 'al'), (33, 'CNVs'), (26, 'genome'), (23, ';'), (22, 'markers'), (22, 'genetic'), (21, 'cholesterol'), (21, 'HDL'), (20, 'used')]\n",
            "gni-6-1-18.txt [(82, 'et'), (82, 'al'), (35, ';'), (33, 'CNVs'), (29, 'genome'), (26, 'studies'), (26, 'markers'), (25, 'using'), (24, 'genetic'), (23, 'used')]\n",
            "gni-6-1-29.txt [(91, 'et'), (91, 'al'), (36, ';'), (33, 'CNVs'), (30, 'studies'), (29, 'genome'), (27, 'using'), (26, 'markers'), (24, 'used'), (24, 'genetic')]\n",
            "gni-6-1-32.txt [(93, 'et'), (93, 'al'), (36, ';'), (33, 'CNVs'), (30, 'studies'), (29, 'genome'), (28, 'using'), (26, 'markers'), (24, 'used'), (24, 'study')]\n",
            "gni-6-1-36.txt [(101, 'et'), (101, 'al'), (39, ';'), (33, 'CNVs'), (31, 'studies'), (29, 'using'), (29, 'genome'), (27, 'markers'), (26, '2004'), (25, 'used')]\n",
            "gni-6-1-44.txt [(104, 'et'), (104, 'al'), (39, ';'), (33, 'CNVs'), (31, 'studies'), (30, 'genome'), (29, 'using'), (27, 'markers'), (26, 'study'), (26, '2004')]\n",
            "gni-6-1-50.txt [(104, 'et'), (104, 'al'), (39, ';'), (33, 'CNVs'), (31, 'studies'), (30, 'genome'), (29, 'using'), (27, 'markers'), (26, 'study'), (26, '2004')]\n",
            "gni-6-1-54.txt [(104, 'et'), (104, 'al'), (39, ';'), (33, 'CNVs'), (31, 'studies'), (31, 'genome'), (29, 'using'), (27, 'markers'), (27, '2004'), (26, 'used')]\n",
            "gni-6-1-8.txt [(51, 'et'), (51, 'al'), (33, 'CNVs'), (26, 'genome'), (21, 'cholesterol'), (21, 'HDL'), (19, ';'), (18, 'CNV'), (17, 'variants'), (17, 'used')]\n",
            "gni-6-2-57.txt [(107, 'et'), (107, 'al'), (39, ';'), (33, 'CNVs'), (32, 'studies'), (31, 'genome'), (30, 'using'), (27, 'used'), (27, 'markers'), (27, '2004')]\n",
            "gni-6-2-64.txt [(107, 'et'), (107, 'al'), (43, ';'), (33, 'using'), (33, 'CNVs'), (32, 'studies'), (32, 'genome'), (27, 'used'), (27, 'markers'), (27, '2004')]\n",
            "gni-6-2-68.txt [(108, 'et'), (108, 'al'), (43, ';'), (33, 'using'), (33, 'CNVs'), (32, 'studies'), (32, 'genome'), (27, 'used'), (27, 'markers'), (27, '2004')]\n",
            "gni-6-2-72.txt [(111, 'et'), (111, 'al'), (44, ';'), (33, 'using'), (33, 'CNVs'), (32, 'studies'), (32, 'genome'), (28, '2004'), (27, 'used'), (27, 'markers')]\n",
            "gni-6-2-77.txt [(115, 'et'), (115, 'al'), (46, ';'), (33, 'using'), (33, 'studies'), (33, 'CNVs'), (32, 'genome'), (28, '2004'), (27, 'used'), (27, 'markers')]\n",
            "gni-6-2-84.txt [(117, 'et'), (117, 'al'), (47, ';'), (33, 'using'), (33, 'studies'), (33, 'CNVs'), (32, 'genome'), (28, '2004'), (27, 'used'), (27, 'study')]\n",
            "gni-6-2-87.txt [(119, 'et'), (119, 'al'), (48, ';'), (34, 'using'), (33, 'studies'), (33, 'CNVs'), (32, 'genome'), (28, 'used'), (28, '2004'), (27, 'study')]\n",
            "gni-6-2-91.txt [(121, 'et'), (121, 'al'), (50, ';'), (34, 'using'), (34, 'studies'), (33, 'CNVs'), (32, 'genome'), (28, 'used'), (28, 'study'), (28, '2004')]\n",
            "gni-6-2-95.txt [(121, 'et'), (121, 'al'), (50, ';'), (34, 'using'), (34, 'studies'), (33, 'CNVs'), (32, 'genome'), (28, 'used'), (28, 'study'), (28, '2004')]\n",
            "gni-6-3-110.txt [(137, 'et'), (137, 'al'), (57, ';'), (36, 'studies'), (34, 'using'), (33, 'CNVs'), (32, 'genome'), (30, 'study'), (30, '2004'), (28, 'used')]\n",
            "gni-6-3-117.txt [(143, 'et'), (143, 'al'), (59, ';'), (37, 'studies'), (34, 'using'), (33, 'CNVs'), (32, 'genome'), (31, 'study'), (30, '2004'), (28, 'used')]\n",
            "gni-6-3-126.txt [(143, 'et'), (143, 'al'), (59, ';'), (37, 'studies'), (35, 'using'), (33, 'genome'), (33, 'CNVs'), (31, 'study'), (30, '2004'), (28, 'used')]\n",
            "gni-6-3-130.txt [(144, 'al'), (143, 'et'), (59, ';'), (38, 'studies'), (37, 'using'), (34, 'genome'), (33, 'CNVs'), (32, 'study'), (30, '2004'), (29, '2005')]\n",
            "gni-6-3-136.txt [(147, 'al'), (146, 'et'), (60, ';'), (41, 'using'), (38, 'studies'), (34, 'genome'), (33, 'study'), (33, 'CNVs'), (30, 'used'), (30, '2004')]\n",
            "gni-6-3-142.txt [(151, 'al'), (150, 'et'), (60, ';'), (41, 'using'), (39, 'studies'), (34, 'study'), (34, 'genome'), (33, 'CNVs'), (31, '2004'), (30, 'used')]\n",
            "gni-6-3-147.txt [(154, 'al'), (153, 'et'), (62, ';'), (42, 'using'), (39, 'studies'), (34, 'study'), (34, 'genome'), (33, 'CNVs'), (32, '2004'), (30, 'used')]\n",
            "gni-6-3-153.txt [(155, 'al'), (154, 'et'), (63, ';'), (42, 'using'), (39, 'studies'), (35, 'study'), (34, 'genome'), (33, 'CNVs'), (32, '2004'), (30, 'used')]\n",
            "gni-6-3-157.txt [(155, 'al'), (154, 'et'), (64, ';'), (42, 'using'), (39, 'studies'), (35, 'study'), (34, 'genome'), (33, 'CNVs'), (32, '2004'), (30, 'used')]\n",
            "gni-6-3-99.txt [(131, 'et'), (131, 'al'), (53, ';'), (34, 'using'), (34, 'studies'), (33, 'CNVs'), (32, 'genome'), (29, 'study'), (29, '2004'), (28, 'used')]\n",
            "gni-6-4-161.txt [(155, 'al'), (154, 'et'), (64, ';'), (42, 'using'), (39, 'studies'), (39, 'genome'), (35, 'study'), (33, 'CNVs'), (32, '2004'), (31, 'used')]\n",
            "gni-6-4-166.txt [(158, 'al'), (157, 'et'), (65, ';'), (42, 'using'), (39, 'studies'), (39, 'genome'), (35, 'study'), (33, 'CNVs'), (32, '2004'), (31, 'used')]\n",
            "gni-6-4-173.txt [(161, 'al'), (160, 'et'), (66, ';'), (42, 'using'), (39, 'studies'), (39, 'genome'), (36, 'study'), (33, 'CNVs'), (32, 'used'), (32, '2004')]\n",
            "gni-6-4-181.txt [(169, 'al'), (168, 'et'), (67, ';'), (42, 'using'), (41, 'studies'), (39, 'genome'), (36, 'study'), (35, '2005'), (33, 'CNVs'), (33, '2004')]\n",
            "gni-6-4-192.txt [(172, 'al'), (171, 'et'), (70, ';'), (42, 'using'), (41, 'studies'), (39, 'genome'), (36, 'study'), (36, '2005'), (35, '2004'), (33, 'used')]\n",
            "gni-6-4-202.txt [(177, 'al'), (176, 'et'), (71, ';'), (43, 'using'), (41, 'studies'), (40, 'genome'), (37, '2004'), (36, 'study'), (36, '2005'), (34, 'genes')]\n",
            "gni-6-4-210.txt [(180, 'al'), (179, 'et'), (72, ';'), (43, 'using'), (41, 'studies'), (40, 'genome'), (38, 'genes'), (38, '2004'), (37, 'study'), (36, '2005')]\n",
            "gni-6-4-223.txt [(181, 'al'), (180, 'et'), (72, ';'), (43, 'using'), (41, 'studies'), (40, 'genome'), (38, 'study'), (38, 'genes'), (38, '2004'), (36, '2005')]\n",
            "gni-6-4-227.txt [(183, 'al'), (182, 'et'), (73, ';'), (43, 'using'), (41, 'studies'), (40, 'genome'), (38, 'study'), (38, 'genes'), (38, '2004'), (36, '2005')]\n",
            "gni-6-4-231.txt [(188, 'al'), (187, 'et'), (75, ';'), (43, 'using'), (42, 'genome'), (41, 'studies'), (40, 'study'), (40, '2004'), (39, 'CNVs'), (38, 'genes')]\n",
            "gni-7-1-1.txt [(188, 'al'), (187, 'et'), (75, ';'), (45, 'using'), (42, 'genome'), (41, 'study'), (41, 'studies'), (40, '2004'), (39, 'CNVs'), (38, 'genes')]\n",
            "gni-7-1-13.txt [(204, 'al'), (203, 'et'), (85, ';'), (45, 'using'), (44, '2004'), (42, 'genome'), (41, 'study'), (41, 'studies'), (39, 'genes'), (39, 'CNVs')]\n",
            "gni-7-1-20.txt [(207, 'al'), (206, 'et'), (86, ';'), (45, 'using'), (45, '2004'), (43, 'study'), (42, 'genome'), (41, 'studies'), (40, 'genes'), (39, 'CNVs')]\n",
            "gni-7-1-26.txt [(214, 'al'), (213, 'et'), (88, ';'), (47, '2004'), (45, 'using'), (44, 'study'), (42, 'genome'), (41, 'studies'), (40, 'genes'), (40, '2005')]\n",
            "gni-7-1-32.txt [(214, 'al'), (213, 'et'), (89, ';'), (47, '2004'), (45, 'using'), (44, 'study'), (42, 'studies'), (42, 'genome'), (40, 'genes'), (40, '2005')]\n",
            "gni-7-1-38.txt [(214, 'al'), (213, 'et'), (90, ';'), (47, 'using'), (47, '2004'), (44, 'study'), (44, 'genome'), (42, 'studies'), (40, 'genes'), (40, 'data')]\n",
            "gni-7-1-41.txt [(214, 'al'), (213, 'et'), (90, ';'), (47, 'using'), (47, '2004'), (44, 'study'), (44, 'genome'), (42, 'studies'), (40, 'genes'), (40, 'data')]\n",
            "gni-7-1-46.txt [(215, 'al'), (214, 'et'), (90, ';'), (47, 'using'), (47, '2004'), (44, 'study'), (44, 'genome'), (42, 'studies'), (40, 'genes'), (40, 'data')]\n",
            "gni-7-1-49.txt [(215, 'al'), (214, 'et'), (90, ';'), (47, 'using'), (47, '2004'), (44, 'study'), (44, 'genome'), (42, 'studies'), (40, 'genes'), (40, 'data')]\n",
            "gni-7-2-107.txt [(248, 'al'), (247, 'et'), (114, ';'), (50, '2004'), (47, 'using'), (46, 'study'), (44, 'genome'), (44, '2005'), (43, 'studies'), (42, 'data')]\n",
            "gni-7-2-111.txt [(251, 'al'), (250, 'et'), (116, ';'), (50, '2004'), (47, 'using'), (46, 'study'), (45, 'genome'), (45, '2005'), (44, 'studies'), (42, 'data')]\n",
            "gni-7-2-122.txt [(252, 'al'), (251, 'et'), (116, ';'), (51, '2004'), (47, 'using'), (46, 'study'), (45, 'genome'), (45, '2005'), (44, 'studies'), (42, 'genes')]\n",
            "gni-7-2-131.txt [(253, 'al'), (252, 'et'), (116, ';'), (51, '2004'), (47, 'using'), (47, 'study'), (45, 'genome'), (45, '2005'), (44, 'studies'), (42, 'genes')]\n",
            "gni-7-2-136.txt [(255, 'al'), (254, 'et'), (116, ';'), (51, '2004'), (48, 'study'), (47, 'using'), (46, 'genome'), (46, '2005'), (44, 'studies'), (43, 'data')]\n",
            "gni-7-2-53.txt [(218, 'al'), (217, 'et'), (91, ';'), (47, 'using'), (47, '2004'), (45, 'study'), (44, 'genome'), (42, 'studies'), (41, 'genes'), (40, 'data')]\n",
            "gni-7-2-57.txt [(220, 'al'), (219, 'et'), (93, ';'), (47, 'using'), (47, '2004'), (45, 'study'), (44, 'genome'), (42, 'studies'), (41, 'genes'), (40, 'data')]\n",
            "gni-7-2-65.txt [(238, 'al'), (237, 'et'), (108, ';'), (50, '2004'), (47, 'using'), (45, 'study'), (44, 'genome'), (42, 'studies'), (42, 'data'), (42, '2005')]\n",
            "gni-7-2-85.txt [(242, 'al'), (241, 'et'), (111, ';'), (50, '2004'), (47, 'using'), (46, 'study'), (44, 'genome'), (43, '2005'), (42, 'studies'), (42, 'data')]\n",
            "gni-7-2-97.txt [(246, 'al'), (245, 'et'), (112, ';'), (50, '2004'), (47, 'using'), (46, 'study'), (44, 'genome'), (44, '2005'), (43, 'studies'), (42, 'data')]\n",
            "gni-7-3-141.txt [(258, 'al'), (257, 'et'), (116, ';'), (52, '2004'), (49, 'using'), (48, 'study'), (46, 'genome'), (46, '2005'), (44, 'studies'), (44, 'data')]\n",
            "gni-7-3-148.txt [(258, 'al'), (257, 'et'), (116, ';'), (52, '2004'), (49, 'using'), (48, 'study'), (46, 'genome'), (46, '2005'), (45, 'data'), (44, 'studies')]\n",
            "gni-7-3-152.txt [(261, 'al'), (260, 'et'), (118, ';'), (52, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, '2005'), (45, 'genes')]\n",
            "gni-7-3-163.txt [(262, 'al'), (261, 'et'), (118, ';'), (52, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, '2005')]\n",
            "gni-7-3-168.txt [(262, 'al'), (261, 'et'), (118, ';'), (52, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, '2005')]\n",
            "gni-7-3-171.txt [(264, 'al'), (263, 'et'), (119, ';'), (53, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-3-175.txt [(264, 'al'), (263, 'et'), (119, ';'), (53, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-3-178.txt [(264, 'al'), (263, 'et'), (119, ';'), (53, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-4-181.txt [(270, 'al'), (269, 'et'), (124, ';'), (53, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-4-187.txt [(279, 'al'), (278, 'et'), (131, ';'), (53, '2004'), (49, 'using'), (49, 'study'), (49, 'studies'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-4-195.txt [(281, 'al'), (280, 'et'), (132, ';'), (54, '2004'), (51, 'study'), (50, 'studies'), (49, 'using'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-4-203.txt [(282, 'al'), (281, 'et'), (132, ';'), (54, '2004'), (53, 'study'), (50, 'studies'), (49, 'using'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-4-208.txt [(282, 'al'), (281, 'et'), (132, ';'), (54, '2004'), (53, 'study'), (50, 'studies'), (49, 'using'), (47, 'genome'), (47, 'genes'), (47, 'data')]\n",
            "gni-7-4-212.txt [(288, 'al'), (287, 'et'), (132, ';'), (58, 'data'), (55, '2004'), (53, 'study'), (50, 'studies'), (49, 'using'), (47, 'genome'), (47, 'genes')]\n",
            "gni-7-4-217.txt [(288, 'al'), (287, 'et'), (132, ';'), (58, 'data'), (55, '2004'), (53, 'study'), (50, 'studies'), (49, 'using'), (49, 'genes'), (47, 'genome')]\n",
            "gni-8-1-1 (1).txt [(300, 'al'), (299, 'et'), (140, ';'), (58, 'data'), (55, '2004'), (53, 'study'), (50, 'studies'), (49, 'using'), (49, 'genes'), (47, 'genome')]\n",
            "gni-8-1-1.txt [(300, 'al'), (299, 'et'), (140, ';'), (58, 'data'), (55, '2004'), (53, 'study'), (50, 'studies'), (49, 'using'), (49, 'genes'), (47, 'genome')]\n",
            "gni-8-1-19 (1).txt [(314, 'al'), (313, 'et'), (151, ';'), (58, 'data'), (57, '2004'), (54, 'study'), (51, 'studies'), (49, 'using'), (49, 'genes'), (48, '2006')]\n",
            "gni-8-1-19.txt [(314, 'al'), (313, 'et'), (151, ';'), (58, 'data'), (57, '2004'), (54, 'study'), (51, 'studies'), (49, 'using'), (49, 'genes'), (48, '2006')]\n",
            "gni-8-1-28 (1).txt [(329, 'al'), (328, 'et'), (161, ';'), (58, 'data'), (57, '2004'), (55, '2006'), (54, 'study'), (51, 'studies'), (51, 'genes'), (49, 'using')]\n",
            "gni-8-1-28.txt [(329, 'al'), (328, 'et'), (161, ';'), (58, 'data'), (57, '2004'), (55, '2006'), (54, 'study'), (51, 'studies'), (51, 'genes'), (49, 'using')]\n",
            "gni-8-1-34 (1).txt [(337, 'al'), (336, 'et'), (162, ';'), (58, 'data'), (57, '2006'), (57, '2004'), (54, 'study'), (53, 'studies'), (51, 'genes'), (49, 'using')]\n",
            "gni-8-1-34.txt [(337, 'al'), (336, 'et'), (162, ';'), (58, 'data'), (57, '2006'), (57, '2004'), (54, 'study'), (53, 'studies'), (51, 'genes'), (49, 'using')]\n",
            "gni-8-1-41 (1).txt [(342, 'al'), (340, 'et'), (162, ';'), (58, 'data'), (57, '2006'), (57, '2004'), (55, 'study'), (53, 'studies'), (52, '2005'), (51, 'genes')]\n",
            "gni-8-1-41.txt [(342, 'al'), (340, 'et'), (162, ';'), (58, 'data'), (57, '2006'), (57, '2004'), (55, 'study'), (53, 'studies'), (52, '2005'), (51, 'genes')]\n",
            "gni-8-1-50 (1).txt [(346, 'al'), (344, 'et'), (165, ';'), (60, '2004'), (58, 'data'), (57, 'studies'), (57, '2006'), (56, 'study'), (52, '2005'), (51, 'genes')]\n",
            "gni-8-1-50.txt [(346, 'al'), (344, 'et'), (165, ';'), (60, '2004'), (58, 'data'), (57, 'studies'), (57, '2006'), (56, 'study'), (52, '2005'), (51, 'genes')]\n",
            "gni-8-1-58.txt [(347, 'al'), (345, 'et'), (165, ';'), (60, '2004'), (58, 'data'), (57, 'studies'), (57, '2006'), (56, 'study'), (52, 'genes'), (52, '2005')]\n",
            "gni-8-1-9.txt [(306, 'al'), (305, 'et'), (144, ';'), (58, 'data'), (55, '2004'), (53, 'study'), (51, 'studies'), (49, 'using'), (49, 'genes'), (48, '2005')]\n",
            "gni-8-2-63.txt [(351, 'al'), (349, 'et'), (166, ';'), (60, '2004'), (58, 'data'), (57, 'studies'), (57, '2006'), (56, 'study'), (53, 'gene'), (52, 'genes')]\n",
            "gni-8-2-70.txt [(358, 'al'), (356, 'et'), (168, ';'), (61, '2004'), (58, 'data'), (57, 'studies'), (57, '2006'), (56, 'study'), (55, 'gene'), (54, '2005')]\n",
            "gni-8-2-76.txt [(362, 'al'), (360, 'et'), (170, ';'), (61, '2004'), (58, 'data'), (57, 'study'), (57, 'studies'), (57, '2006'), (56, 'gene'), (56, '2005')]\n",
            "gni-8-2-81.txt [(364, 'al'), (362, 'et'), (171, ';'), (61, '2004'), (58, 'study'), (58, 'data'), (57, 'studies'), (57, '2006'), (56, 'genes'), (56, 'gene')]\n",
            "gni-8-2-86.txt [(364, 'al'), (362, 'et'), (171, ';'), (61, '2004'), (59, 'data'), (58, 'study'), (57, 'studies'), (57, '2006'), (56, 'genes'), (56, 'gene')]\n",
            "gni-8-2-90.txt [(365, 'al'), (363, 'et'), (171, ';'), (61, '2004'), (60, 'data'), (59, 'study'), (57, 'studies'), (57, '2006'), (56, 'genes'), (56, 'gene')]\n",
            "gni-8-2-94.txt [(366, 'al'), (363, 'et'), (171, ';'), (61, '2004'), (60, 'data'), (59, 'study'), (57, 'studies'), (57, '2006'), (57, '2005'), (56, 'genes')]\n",
            "gni-8-3-101.txt [(373, 'al'), (370, 'et'), (171, ';'), (61, 'data'), (61, '2004'), (60, 'study'), (57, 'studies'), (57, 'gene'), (57, '2006'), (57, '2005')]\n",
            "gni-8-3-103.txt [(375, 'al'), (372, 'et'), (172, ';'), (64, 'study'), (61, 'data'), (61, '2004'), (57, 'using'), (57, 'studies'), (57, 'gene'), (57, '2006')]\n",
            "gni-8-3-108.txt [(379, 'al'), (376, 'et'), (173, ';'), (64, 'study'), (61, 'data'), (61, '2004'), (59, 'studies'), (59, '2006'), (57, 'using'), (57, 'genes')]\n",
            "gni-8-3-116.txt [(379, 'al'), (376, 'et'), (173, ';'), (64, 'study'), (61, 'data'), (61, '2004'), (60, 'studies'), (59, '2006'), (57, 'using'), (57, 'genes')]\n",
            "gni-8-3-122.txt [(379, 'al'), (376, 'et'), (173, ';'), (65, 'study'), (62, 'data'), (61, '2004'), (60, 'studies'), (59, '2006'), (57, 'using'), (57, 'genes')]\n",
            "gni-8-3-131.txt [(381, 'al'), (378, 'et'), (173, ';'), (65, 'study'), (62, 'data'), (62, '2004'), (60, 'studies'), (59, '2006'), (58, 'using'), (57, 'genes')]\n",
            "gni-8-3-138.txt [(382, 'al'), (379, 'et'), (176, ';'), (66, 'study'), (62, 'data'), (62, '2004'), (60, 'studies'), (59, '2006'), (58, 'using'), (57, 'genes')]\n",
            "gni-8-3-142.txt [(390, 'al'), (387, 'et'), (176, ';'), (67, 'study'), (62, 'data'), (62, '2004'), (61, 'studies'), (60, 'gene'), (59, '2006'), (58, 'using')]\n",
            "gni-8-3-150.txt [(390, 'al'), (387, 'et'), (176, ';'), (68, 'study'), (62, 'data'), (62, '2004'), (61, 'studies'), (60, 'gene'), (59, '2006'), (58, 'using')]\n",
            "gni-8-3-159.txt [(390, 'al'), (387, 'et'), (176, ';'), (68, 'study'), (62, 'data'), (62, '2004'), (61, 'studies'), (60, 'gene'), (59, '2006'), (58, 'using')]\n",
            "gni-8-4-165.txt [(390, 'al'), (387, 'et'), (176, ';'), (68, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (59, '2006'), (58, 'using')]\n",
            "gni-8-4-170.txt [(390, 'al'), (387, 'et'), (176, ';'), (68, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (59, 'using'), (59, '2006')]\n",
            "gni-8-4-177.txt [(393, 'al'), (390, 'et'), (178, ';'), (68, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (59, 'using'), (59, '2006')]\n",
            "gni-8-4-185.txt [(398, 'al'), (395, 'et'), (181, ';'), (68, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (60, '2006'), (59, 'using')]\n",
            "gni-8-4-194.txt [(398, 'al'), (395, 'et'), (181, ';'), (69, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (60, '2006'), (59, 'using')]\n",
            "gni-8-4-201.txt [(398, 'al'), (395, 'et'), (181, ';'), (69, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (60, '2006'), (59, 'using')]\n",
            "gni-8-4-206.txt [(403, 'al'), (400, 'et'), (181, ';'), (69, 'study'), (63, 'data'), (62, 'gene'), (62, '2004'), (61, 'studies'), (60, '2006'), (59, 'using')]\n",
            "gni-8-4-212.txt [(417, 'al'), (414, 'et'), (188, ';'), (69, 'study'), (63, 'data'), (63, '2005'), (62, 'gene'), (62, '2006'), (62, '2004'), (61, 'studies')]\n",
            "gni-9-1-1.txt [(417, 'al'), (414, 'et'), (188, ';'), (70, 'study'), (63, 'data'), (63, '2005'), (62, 'gene'), (62, '2006'), (62, '2004'), (61, 'studies')]\n",
            "gni-9-1-12.txt [(420, 'al'), (417, 'et'), (191, ';'), (70, 'study'), (64, 'gene'), (63, 'data'), (63, '2005'), (62, '2006'), (62, '2004'), (61, 'studies')]\n",
            "gni-9-1-19.txt [(422, 'al'), (419, 'et'), (191, ';'), (70, 'study'), (67, 'data'), (64, 'gene'), (63, '2006'), (63, '2005'), (62, '2004'), (61, 'studies')]\n",
            "gni-9-1-28.txt [(422, 'al'), (419, 'et'), (191, ';'), (70, 'study'), (68, 'data'), (64, 'gene'), (63, 'genes'), (63, '2006'), (63, '2005'), (62, '2004')]\n",
            "gni-9-1-37.txt [(422, 'al'), (419, 'et'), (191, ';'), (70, 'study'), (68, 'data'), (64, 'gene'), (63, 'genes'), (63, '2006'), (63, '2005'), (62, '2004')]\n",
            "gni-9-1-39.txt [(422, 'al'), (419, 'et'), (191, ';'), (70, 'study'), (68, 'data'), (64, 'gene'), (63, 'genes'), (63, '2006'), (63, '2005'), (63, '2004')]\n",
            "gni-9-1-5.txt [(417, 'al'), (414, 'et'), (188, ';'), (70, 'study'), (63, 'data'), (63, '2005'), (62, 'gene'), (62, '2006'), (62, '2004'), (61, 'studies')]\n",
            "gni-9-2-45.txt [(423, 'al'), (420, 'et'), (191, ';'), (70, 'study'), (68, 'data'), (65, 'gene'), (64, 'genes'), (63, '2006'), (63, '2005'), (63, '2004')]\n",
            "gni-9-2-52.txt [(429, 'al'), (426, 'et'), (195, ';'), (70, 'study'), (69, 'data'), (65, 'gene'), (64, 'genes'), (64, '2006'), (64, '2004'), (63, '2005')]\n",
            "gni-9-2-59.txt [(429, 'al'), (426, 'et'), (195, ';'), (70, 'study'), (69, 'data'), (65, 'gene'), (64, 'genes'), (64, '2006'), (64, '2004'), (63, '2005')]\n",
            "gni-9-2-64.txt [(430, 'al'), (427, 'et'), (195, ';'), (71, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-2-69.txt [(432, 'al'), (429, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-2-74.txt [(432, 'al'), (429, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-2-79.txt [(432, 'al'), (429, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-2-85.txt [(432, 'al'), (429, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-2-89.txt [(433, 'al'), (430, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-3-102.txt [(433, 'al'), (430, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-3-114.txt [(434, 'al'), (431, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-3-121.txt [(438, 'al'), (435, 'et'), (199, ';'), (73, 'study'), (69, 'data'), (66, '2006'), (65, 'genes'), (65, 'gene'), (65, '2005'), (64, '2004')]\n",
            "gni-9-3-127.txt [(446, 'al'), (443, 'et'), (205, ';'), (73, 'study'), (69, 'data'), (67, 'gene'), (67, '2005'), (66, 'genes'), (66, '2006'), (65, '2004')]\n",
            "gni-9-3-134.txt [(446, 'al'), (443, 'et'), (205, ';'), (73, 'study'), (69, 'data'), (67, 'gene'), (67, '2005'), (66, 'genes'), (66, '2006'), (65, '2004')]\n",
            "gni-9-3-136.txt [(446, 'al'), (443, 'et'), (205, ';'), (73, 'study'), (69, 'gene'), (69, 'data'), (67, 'genes'), (67, '2005'), (66, '2006'), (65, '2004')]\n",
            "gni-9-3-138.txt [(446, 'al'), (443, 'et'), (205, ';'), (73, 'study'), (69, 'gene'), (69, 'data'), (67, 'genes'), (67, '2005'), (66, '2006'), (65, '2004')]\n",
            "gni-9-3-93.txt [(433, 'al'), (430, 'et'), (196, ';'), (72, 'study'), (69, 'data'), (65, 'genes'), (65, 'gene'), (64, '2006'), (64, '2005'), (64, '2004')]\n",
            "gni-9-4-143.txt [(447, 'al'), (444, 'et'), (207, ';'), (73, 'study'), (71, 'gene'), (69, 'data'), (67, 'genes'), (67, '2005'), (66, '2006'), (65, '2004')]\n",
            "gni-9-4-152.txt [(447, 'al'), (444, 'et'), (207, ';'), (73, 'study'), (71, 'gene'), (69, 'data'), (67, 'genes'), (67, '2005'), (66, '2006'), (65, '2004')]\n",
            "gni-9-4-161.txt [(451, 'al'), (448, 'et'), (207, ';'), (73, 'study'), (71, 'gene'), (69, 'data'), (68, '2006'), (67, 'genes'), (67, '2005'), (66, '2004')]\n",
            "gni-9-4-173.txt [(451, 'al'), (448, 'et'), (207, ';'), (73, 'study'), (71, 'gene'), (69, 'data'), (68, '2006'), (67, 'genes'), (67, '2005'), (66, '2004')]\n",
            "gni-9-4-181.txt [(451, 'al'), (448, 'et'), (207, ';'), (73, 'study'), (71, 'gene'), (69, 'data'), (68, '2006'), (67, 'genes'), (67, '2005'), (66, '2004')]\n",
            "gni-9-4-189.txt [(451, 'al'), (448, 'et'), (208, ';'), (73, 'study'), (71, 'gene'), (70, 'data'), (68, '2006'), (67, 'genes'), (67, '2005'), (66, 'studies')]\n",
            "gni-9-4-194.txt [(451, 'al'), (448, 'et'), (210, ';'), (73, 'study'), (71, 'gene'), (70, 'data'), (68, '2006'), (67, 'genes'), (67, '2005'), (67, '2004')]\n",
            "gni-9-4-197.txt [(451, 'al'), (448, 'et'), (210, ';'), (73, 'study'), (71, 'gene'), (70, 'data'), (68, '2006'), (67, 'genes'), (67, '2005'), (67, '2004')]\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "w2p0hPIKDB4A",
        "colab_type": "code",
        "outputId": "06211fdb-64a2-4744-bc16-e4caa9b94747",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "raw = '''I'm feeling \"hot-tempered\".'''\n",
        "re.split(r' ',raw)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[\"I'm\", 'feeling', '\"hot-tempered\".']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 50
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "yL1MSA45Dj4K",
        "colab_type": "code",
        "outputId": "c0b38035-c75a-4f9d-cf18-1507cf602d18",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "re.findall(r\"\\w+(?:[-']\\w+)*|'|[-.(]+|\\S\\w*\", raw)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[\"I'm\", 'feeling', '\"hot', '-', 'tempered', '\"', '.']"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 51
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ei2HcBPcEaPf",
        "colab_type": "text"
      },
      "source": [
        "* Regular Expression\n",
        "\n",
        "`(?x)` : set flag to allow verbose regexps\n",
        "\n",
        "`([A-Z]\\.)+` : abbreviations (ex U.S.A)\n",
        "\n",
        "`\\w+(-\\w+)*` : words with optional internal hyphens (ex hot-water) to one word\n",
        "\n",
        "`\\$?\\d+(\\.\\d+)?%?` : currency percentages (ex $12.40, 82%)\n",
        "\n",
        "`\\.\\.\\.` : ...\n",
        "\n",
        "[][.,;\"'?():-_`] : seperate tokens include [ , ]"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "HTGfZeseEUSi",
        "colab_type": "code",
        "outputId": "e744c9c1-8ea1-4d79-d871-c95b48d06e5e",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 148
        }
      },
      "source": [
        "text = 'That U.S.A poster-print costs $12.40...'\n",
        "pattern = r'''(?x)\n",
        "    ([A-Z]\\.)+\n",
        "  | \\w+(-\\w+)*\n",
        "  | \\$?\\d+(\\.\\d+)?%?\n",
        "  | \\.\\.\\.\n",
        "  | [][.,;\"'?():-_`]'''\n",
        "\n",
        "nltk.regexp_tokenize(text, pattern)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[('', '', ''),\n",
              " ('S.', '', ''),\n",
              " ('', '', ''),\n",
              " ('', '-print', ''),\n",
              " ('', '', ''),\n",
              " ('', '', '.40'),\n",
              " ('', '', '')]"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 57
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "zIkXJ048Hp1r",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "giRaw = GNI.raw('gni-9-3-127.txt')\n",
        "giWords = GNI.words('gni-9-3-127.txt')\n",
        "giWords2 = re.split(r'[ \\n\\t]+', giRaw)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "9UamSPFfIAOk",
        "colab_type": "code",
        "outputId": "2d742247-4969-4db0-e1c8-307437f9dce1",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(giWords)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "28968"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 62
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "HQEMjM4BIB5j",
        "colab_type": "code",
        "outputId": "7eb5ae34-052c-41d0-990b-4c7e4cccb3ca",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(giWords2)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "23389"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 63
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "UAngEg7cIPjm",
        "colab_type": "code",
        "outputId": "f7a5e4ea-8260-4d40-c6f7-c5ccb2603c69",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 56
        }
      },
      "source": [
        "#sentence segmentation\n",
        "text = nltk.corpus.gutenberg.raw('chesterton-thursday.txt')\n",
        "sents = nltk.sent_tokenize(text)\n",
        "print(sents[79:89])"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "['\"Nonsense!\"', 'said Gregory, who was very rational when anyone else\\nattempted paradox.', '\"Why do all the clerks and navvies in the\\nrailway trains look so sad and tired, so very sad and tired?', 'I will\\ntell you.', 'It is because they know that the train is going right.', 'It\\nis because they know that whatever place they have taken a ticket\\nfor that place they will reach.', 'It is because after they have\\npassed Sloane Square they know that the next station must be\\nVictoria, and nothing but Victoria.', 'Oh, their wild rapture!', 'oh,\\ntheir eyes like stars and their souls again in Eden, if the next\\nstation were unaccountably Baker Street!\"', '\"It is you who are unpoetical,\" replied the poet Syme.']\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "zSXPneawJrOJ",
        "colab_type": "text"
      },
      "source": [
        "\n",
        "\n",
        "---\n",
        "\n",
        "학습활동"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "vWE9SZRAJGRE",
        "colab_type": "code",
        "colab": {}
      },
      "source": [
        "giRaw = GNI.raw('gni-9-3-127.txt')\n",
        "sents = nltk.sent_tokenize(giRaw)"
      ],
      "execution_count": 0,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "e3JSkjEGJ_uw",
        "colab_type": "code",
        "outputId": "5e99d0fb-b57d-4688-9f20-ad11b55b7811",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 36
        }
      },
      "source": [
        "len(sents)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1007"
            ]
          },
          "metadata": {
            "tags": []
          },
          "execution_count": 70
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "ffaAeu7DKTaq",
        "colab_type": "code",
        "outputId": "54d66907-6264-42ef-c423-a963f720914f",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "for sent in sents[:50]:\n",
        "  print(\"*\"+sent)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "*\n",
            "=============Title==========\n",
            "Copy Number Variations in the Human Genome: Potential Source for Individual Diversity and Disease Association Studies.\n",
            "*=============Cor Author==========\n",
            "*Corresponding author: E-mail yejun@catholic.ac.krTel +82-2-590-1214, Fax +82-2-596-8969 Accepted 11 March 2008\n",
            "===========Author==========\n",
            "Tae-Min Kim1, Seon-Hee Yim2 and Yeun-Jun Chung1,2*1Department of Microbiology, 2Integrated Research Center for Genome Polymorphism, The Catholic University of Korea, Seoul 137-701, Korea−5); the re-maining 12 (ARHGEF12, CSMD1, ARSG, SGSM1, GJA8, ALX4, 2q31.1, 15q22.1, LIPC, CSK, 7q21.13, 7q31.1, 10q25.2) were suggestive loci (10−5＜p-value＜10−4).\n",
            "*To enhance the original association study, a total of 1,827,004 imputed SNPs were reanalyzed for their asso-ciation with blood pressure in this study.\n",
            "*===========Keywords==========\n",
            "Keywords: array-CGH, Copy number variation (CNV), Genome-wide association study (GWAS)Keywords: chromosome, genome-wide linkage search, heritability, HDL cholesterolKeywords: inbreeding coefficient, Mengolian population, STR, HWE, PICKeywords: haplotype, HapMap, Korean, LD, populations, SNP\n",
            "===========Sub Heading==========\n",
            "Abstract\tIntroduction\tThe definition of CNV\tThe identification of CNVs using differ-ent platforms\tClinical implications of CNVs and dis-ease association study\tConclusion\tIntroduction\tMethods\tResults and Discussion\tIntroduction\tMethods\tResults\tDiscussion\tIntroduction\tMethods\tMethods\tConstruction of Adenoviral Vector for hTERT-specific Group I Intron \tMajor Functionalities of CGHscape\tMethods\tFeatures and Results\tPersonal Genomics\tPolymorphism and Mutation Databases\tMethods\tFeatures and Results\tMethods \tFeatures and Results\tMethodology\tJ2.5dPathway Browser Architecture\tMethods\tData and processing\tMethods\tMethods\tMethods\tMethods\tMethods\tMethods\tMethods\tSelf-Reported Personality Measures\tMethods\tParticle Swarm Optimization\t\n",
            "==========Minor Heading===========\n",
            "ASubjects, medical histories, genotyping, and measurement of HDL cholesterol\tStatistical analyses, heritability estimation, and variance component linkage analysis\tParticipants\tGenotyping\tEstimating Hardy-Weinberg Equilibrium (HWE), Information Contents and Inbreeding Coefficients\tASNP Selection\tDNA Samples\tGenotyping\tStatistical Analysis \tADatasets\tThe dataset\tSubjects\tMethod\tDataset\tStudy samples\tAnimals \tCataloging syntenic regions\tData acquisition\tStudy samples\tStudy subjects\tSubjects \tMolecular Modeling\tModel design \t\n",
            "===========Main Text==========\n",
            "Abstract.\n",
            "*The widespread presence of large-scale genomic variations, termed copy number variation (CNVs), has been recently recognized in phenotypically normal individuals.\n",
            "*Judging by the growing number of reports on CNVs, it is now evident that these variants contribute significantly to genetic diversity in the human genome.\n",
            "*Like single nucleotide polymorphisms (SNPs), CNVs are expected to serve as potential biomarkers for disease susceptibility or drug responses.\n",
            "*However, the technical and practical concerns still remain to be tackled.\n",
            "*In this review, we examine the current status of CNV DBs and research, including the ongoing efforts of CNV screening in the human genome.\n",
            "*We also discuss the characteristics of platforms that are available at the moment and suggest the potential of CNVs in clinical research and application.\n",
            "*IIntroduction.\n",
            "*Traditionally, large-scale genomic variants that are visible in conventional karyotyping have been thought to be associated with early-onset, highly penetrant genetic disorders, while they are incompatible in normal, disease-free individuals (Lupski, 1998; Stankiewicz and Lupski, 2002).\n",
            "*The construction of the 'reference genome' by the human genome sequencing project is based on the belief that human genome sequences are virtually identical, even in different individuals, except for well-known single nucleotide polymorphisms (SNP) or size-variants of tandem repeats such as mini- or microsatellites (variable number of tandem repeats or VNTR) (Przeworski et al., 2000).\n",
            "*This traditional concept has been recently challenged by the discovery that large structural variations are more prevalent than previously presumed (Check, 2005).\n",
            "*Using high-resolution whole- genome scanning technologies such as array-based comparative genomic hybridization (array-CGH), two groups of pioneering scientists have identified widespread copy number variations (CNVs) in apparently healthy, normal individuals (Iafrate et al., 2004; Sebat et al., 2004).\n",
            "*It proposes that our genome is more diverse than has ever been recognized, and subsequent studies have identified up to 11,000 CNVs across the whole genome (Tuzun et al., 2005; Hinds et al., 2006; Mills et al., 2006; McCarroll et al., 2006; Conrad et al., 2006; Sharp et al., 2005; Wong et al., 2007; de Smith et al., 2007).\n",
            "*Although the current understanding of CNVs is still limited for practical use and technical challenges still remain to be tackled, recent studies already have demonstrated the potential association of CNVs with various diseases, suggesting plausible functional significances and highlighting the promising utility of CNVs.\n",
            "*The current coverage of CNVs in the human genome already has exceeded that of SNPs (approximately 600 Mb comprising 12% of human genome) and is still increasing (Cooper et al., 2007).\n",
            "*These large-scale structural variants, in addition to SNPs, will serve as powerful sources to help our understanding of human genetic variation and of differences in disease susceptibility for various diseases.\n",
            "*This paper reviews the current knowledge and future perspectives of CNVs.\n",
            "*The definition of CNV.\n",
            "*Structural variations that involve large DNA segments can take various forms, such as duplication, deletion, insertion, inversion, and translocation.\n",
            "*Among them, DNA copy number variations larger than 1 kb are collectively termed CNVs.\n",
            "*Fig.\n",
            "*1 illustrates the concept of CNV.\n",
            "*Although the CNV can include large, microscopically visible genomic variations, it generally indicates a submicroscopic structural variation that is hardly detectable by conventional karyotyping (35 Mb) (Freeman et al., 2006).\n",
            "*Smaller variations such as small insertional- deletion (indel) polymorphisms are not included in CNVs, while they comprise another large collection of over  400,000 variants in the human genome (Mills et al., 2006), and  neither is the insertional polymorphism of mobile elements such as Alus or L1 elements considered a CNV.\n",
            "*At the beginning stages of CNV discovery, a number of terms were proposed to define them e.g., large-scale copy number variants (LCV) (Iafrate et al., 2004), copy number polymorphism (CNP) (Sebat et al., 2004), and intermediate-sized variants (ISV) (Tuzun et al., 2005).\n",
            "*The current definition of CNV is also operational and can be modified with the advance of scanning resolution and coverage, and availability of allele frequency in a determined population.The identification of CNVs using differ-ent platforms.\n",
            "*Various scanning platforms and quality control methods have been used to identify CNV calls.\n",
            "*Because the choice of platforms has a great effect on the results, it is worth reviewing the characteristics of platforms to improve the understanding of CNVs.\n",
            "*The presence of CNVs in normal individuals was reported for the first time in 2004 independently by two groups led by Lee C. and Wigler M. (Iafrate et al., 2004; Sebat et al., 2004).\n",
            "*Both studies used two-dye array-CGH techniques that used clones of bacterial artificial chromosomes (BAC) or oligonucleotides (representational oligonucleotide microarray analysis, or ROMA).\n",
            "*Theyindependently reported about 250 and 80 loci as changes in copy number from 39 and 20 normal individuals, respectively.\n",
            "*Fig.\n",
            "*2 illustrates the general concept of CNV detection based on two-dye array-CGH.\n",
            "*Although the average numbers of CNVs per individual genome were similar in two studies (about 12 CNVs per genome), it should be noted that there was little overlap between the results.\n",
            "*This discrepancy between studies was possibly due to the use of different platforms and experimental conditions in different populations.\n",
            "*However, it is also probable that there are still large numbers of structural variants that have yet to be discovered (Buckley et al., 2005; Eichler, 2006).\n",
            "*One following study that provided evidence on the widespread presence of large-scale structural variations in the human genome was based solely on in silico analysis (Tuzun et al., 2005).\n",
            "*The sequence-level comparison of two independent genome sequences, i.e., one derived from a human genome reference assembly  and the other from fosmid clones of a genomic library, revealed about 300 structural variations, including inversions.\n",
            "*This method can detect various types of structural variants, including inversion, which is not detectable by conventional array-CGH platforms.\n",
            "*Indeed, the results by Tuzun et al.\n",
            "*(2005) can be used as validated control for primary verification or for parameter tuning for the development of CNV-detection platforms or algorithms.\n",
            "*Although the use of this method is currently limited by the unavailability of sequence data, ongoing efforts to sequence the individual human genome and to develop cost-effective sequencing platforms (Bennett et al., 2005) will be able to facilitate sequence-level genome comparisons and the identification of highly qualified structural variants in the near future.\n",
            "*Two studies by McCarroll et al.\n",
            "*and Conrad et al., which focused on the identification of deletion variants (McCarroll et al., 2006; Conrad et al., 2006), used 1.2 million SNP genotyping data from The International HapMap Consortium (International HapMap Consortium.\n",
            "*2005).\n",
            "*They assumed that allelic deletion causes the discard of probes in SNP genotyping.\n",
            "*For example, the runs of consecutive probes with null genotype calls or runs of SNP genotypes whose allelic frequencies deviate from expected Hardy-Weinberg equilibrium ratios or expected Mendelian inheritance patterns might represent the presence of deleted loci.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "x87O4aUvMRAR",
        "colab_type": "code",
        "outputId": "8327af9c-8c32-4d2a-fe68-f259e1e52076",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "for sent in sents[200:300]:\n",
        "  print(\"*\"+sent)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "*The information contents in terms of PIC for single marker, range between 0.2 and 0.9, as shown in Fig.\n",
            "*1.\n",
            "*Average PIC was 0.72 and 0.71 for Orkhontuul and Dashbalbar populations, respectively which are relatively high for single marker information contents.\n",
            "*There was no significant difference in PIC across the chromosomes or populations.\n",
            "*The high PIC level enabled accurate estimation of other population genetic parameters.\n",
            "*HWE was satisfied among 88.6 % and 94.2%, respectively, of all markers in Orkhontuul and Dashbalbar populations (p-value 0.05).\n",
            "*If we apply the criteria of p-value 0.01, 90.5% and 95.3% of all markers were in HWE status All the markers including those which were not in HWE, were used for estimating the inbreeding coefficients,.\n",
            "*Inbreeding coefficient was estimated to be 0.0023 and 0.0021 in Orkhontuul and Dashbalbar populations.\n",
            "*Discussion.\n",
            "*Population isolates are generally considered to be one of the most ideal populations for genetic study (Pajukanta et al., 2003; rcos-Burgos et al., 2002; Escamilla et al., 2001).\n",
            "*However, possible inbreeding can cause deviation from general assumptions on which most analyses depend.\n",
            "*Presence of inbreeding can be problematic, because, if exits, l the genetic relationships between unrelated as well as related persons could be underestimated.\n",
            "*This underestimation of IBD can result in inflation of type I errors for linkage analysis (Hossjer et al., 2006 Nomura et al., 2005), linkage disequilibrium estimations and haplotype reconstructions (Zhang et al., 2004).\n",
            "*The inbreeding coefficient found in this study (about 0.2% in each population), does not necessitate any adjustment for genetic analyses such as IBD calculation, classic or non-parametric linkage analysis, and variance component-based linkage analysis.\n",
            "*By estimating the last common ancestor, 0.2% of inbreeding coefficient corresponds to 10 or 11 generations (Jensen- Seaman et al., 2001; Santos-Lopes et al., 2007).\n",
            "*In this study, both ABI and deCODE STR markers were genotyped with standardized procedure and any markers with more than 1% of genotype errors were discarded.\n",
            "*The genotype errors were confirmed within the pedigree structure.\n",
            "*Any Mendelian inconsistency was deleted and markers with possible double-recombination were also deleted.\n",
            "*Generally, genotyping in family-based study is more accurate than in studies using individuals only.\n",
            "*Thus, It is not likely that any genotype error could have been biased our findings.\n",
            "*In conclusion, we have estimated inbreeding coefficients in two population isolates in Mongolia.,.\n",
            "*We found that they fall in negligible range, allowing related genetic studies to be performed without any modification or adjustment for possible inbreeding effects.\n",
            "*This finding validates the ability of The GENDISCAN study to add to the growing body of evidence which associates specific genetic variations with complex disorders.% (6.4 of 34.5 Mb) of chromosome 22 with 757 tagSNPs and 815 haplotypes (frequency 5.0%).\n",
            "*Of 3430 common SNPs genotyped in all five populations, 514 were monomorphic in Koreans.\n",
            "*The CHB + JPT samples have more than a 72% overlap with the monomorphic SNPs in Koreans, while the CEU + YRI samples have less than a 38% overlap.\n",
            "*The patterns of hot spots and LD blocks were dispersed throughout chromosome 22, with some common blocks among populations, highly concordant between the three Asian samples.\n",
            "*Analysis of the distribution of chimpanzee-derived allele frequency (DAF), a measure of genetic differentiation, Fst levels, and allele frequency difference (AFD) among Koreans and the HapMap samples showed a strong correlation between the Asians, while the CEU and YRI samples showed a very weak correlation with Korean samples.\n",
            "*Relative distance as a quantitative measurement based upon DAF, Fst, and AFD indicated that all three Asian samples are very proximate, while CEU and YRI are significantly remote from the Asian samples.\n",
            "*Comparative genome-wide LD studies provide useful information on the association studies of complex diseases.\n",
            "*IIntroduction.\n",
            "*Vast amounts of information on single nucleotide polymorphisms (SNPs) and progress in high-throughput genotyping technology have generated a great deal of interest in establishing genome-wide linkage disequilibrium (LD) maps for genetic studies of complex traits (Chakravarti 2001; The International HapMap Consortium 2003; Myers and Bottolo 2005).\n",
            "*LD is known to occur in a block-like structure across the genome, with conserved haplotype blocks of tens to hundreds of kilobases punctuated by \"hot spots\" of recombination (Daly et al.\n",
            "*2001).\n",
            "*Since the concept of whole genome association studies using SNPs was introduced (Risch and Merikangas 1996), an optimal number of SNPs required for association studies has been center of extensive debate (Kruglyak 1999).\n",
            "*Initial studies have focused on average LD levels and the variability in processes that generate LD (Cardon and Abecasis 2003).\n",
            "*Although a single chromosome could carry many haplotypes in LD blocks, recent studies suggest that haplotypic variation may be much lower than previously imagined (Jeffreys et al.\n",
            "*2001; Patil et al.\n",
            "*2001; Gabriel et al.\n",
            "*2002).\n",
            "*Patil's group identified haplotype blocks on chromosome 21 for which over 80% of chromosomes were represented by a few common haplotypes (Patil et al.\n",
            "*2001).\n",
            "*In the analysis of human chromosome 22 with a marker density of one SNP per 15 kb, Dawson's group reported a highly variable pattern of LD along the chromosome, in which extensive regions of complete LD of up to 804 kb in length were interspersed with regions of no detectable LD (Dawson et al.\n",
            "*2002).\n",
            "*Although differences of LD patterns between populations have been reported (Abecasis et al.\n",
            "*2002; Reich et al.\n",
            "*2001, Zavattari et al.\n",
            "*2002), little information is available on the haplotype structure in different populations other than the recent study by S.B.\n",
            "*Gabriel, et al.\n",
            "*(Gabriel et al.\n",
            "*2002).\n",
            "*On the other hand, haplotype analysis has been widely employed in linkage studies for narrowing down the location of disease susceptibility genes (Zhang et al.\n",
            "*2004; Park 2007).\n",
            "*The International HapMap Project was launched to develop a haplotype map of the human genome, the HapMap, which will describe the common patterns of human DNA sequence variation among four population samples: 30 trios from Yoruba in Ibadan, Nigeria (YRI), 45 unrelated Japanese in Tokyo, Japan (JPT), 45 unrelated Han Chinese in Beijing, China (CHB), and 30 trios in a Utah, US population with Northern and Western European ancestry (CEU) from the CEPH collection (The International HapMap Consortium 2003; 2004; 2007).\n",
            "*As the International HapMap Project releases a validated SNP map of 1 marker per kb for the HapMap samples, the general applicability of the HapMap data needs to be confirmed in samples from related populations.\n",
            "*Recent comparative studies of LD patterns have shown a high degree of concordance among various populations (Gabriel et al.\n",
            "*2002; Shifman et al.\n",
            "*2003; Stenzel et al.\n",
            "*2004; Mueller et al.\n",
            "*2005).\n",
            "*As the HapMap samples include Japanese and Chinese, it was our interest to test whether significant differences in LD exist between Koreans and the two other Asian samples.\n",
            "*In this paper, we measured the LD pattern along chromosome 22 in Korean samples and compared the Korean data with those of the four HapMap samples.\n",
            "*We were interested in exploring how the HapMap data could be used to estimate the genomic structure of Koreans.\n",
            "*We expect that this study will contribute to the development of proper strategies for association studies of common complex diseases in Koreans using the HapMap data.\n",
            "*Methods.\n",
            "*A total of 111,448 reference SNPs from chromosome 22 in the dbSNP (http://www.ncbi.nlm.nih.gov/SNP, build 116) were collected.\n",
            "*To maximize cost effectiveness of genotyping, SNPs were selected based on the following criteria: 1) markers with even spacing, 2) verified SNPs, 3) coding SNPs.\n",
            "*The SNPs were scored for the selection of the study using the following strategies.\n",
            "*First, it was most important in mapping chromosomal LD blocks to have relatively equal spaces between SNP markers.\n",
            "*Second, verified SNP markers (validation status was scored as 0 to 4 in the dbSNP) that had higher scores were chosen to prevent or reduce genotyping failure.\n",
            "*Also, repeated sequence regions were excluded by repeat masking with Primer3 software (Rozen and Skaletsky 2000).\n",
            "*Third, to be useful for a further study, protein coding SNPs had higher scores.\n",
            "*A total of 12,674 genotyping experiments were conducted by four Genotyping Centers, and a final set of 4681 markers passed the stringent quality control procedure (The International HapMap Consortium 2003).\n",
            "*Genomic DNA from 90 unrelated Korean individuals without family histories of major diseases was obtained from the Genomic Research Center in the Korean National Institute of Health (KNIH).\n",
            "*The KNIH samples were collected as part of an epidemiological project and represent urban and rural regions in the south of Seoul.\n",
            "*The sex ratio was 0.5 and the mean age was 50.\n",
            "*Informed consent from all participating subjects was obtained through KNIH, and research approval came from the relevant ethical committees.\n",
            "*DNA was isolated from peripheral blood leukocytes according to standard procedures with proteinase K-RNase digestion, followed by phenol-chloroform extraction.For each SNP, we chose a set of three primers: two PCR primers to amplify a product of 100-200 bps under standard conditions and an optimized extension primer to be complementary to the sequence immediately to a SNP site.\n",
            "*For genotyping, we employed three platforms-6063 SNP genotypings were done using the Orchid Bioscience SNP-IT assay (Princeton, NJ), 984 SNP genotypings using the PerkinElmer Life Sciences FP-TDI assay (Boston, MA), and 5627 SNP genotypings using the Sequenom MassARRAY (San Diego, CA).\n",
            "*A genotype frequency for each SNP was checked for consistency between the observed values and those expected from the Hardy-Weinberg equilibrium test in each assay.\n",
            "*Haploview version 3.2 (Barrett et al.\n",
            "*2004), based on the expectation-maximization (EM) method (Excoffier and Slatkin 1995), was used to infer haplotype phase and population frequency and to estimate the Lewontin's coefficients D' (Lewontin 1998), LOD, and correlation coefficient r (Hill and Robertson 1968).\n",
            "*PHASE v2.1 was used to estimate the recombination parameters (Li and Stephens 2003; Crawford et al.\n",
            "*2004) and assess the statistical significance of haplotype profile differences and individual haplotype fre-2006).\n",
            "*Because it has been suggested that the functional significance of IL-1B-3737 might depend on a broader haplotype, we used the three SNPs for haplotype analysis.\n",
            "*Haplotypes were reconstructed by PHASE version 2.1, using previously produced genotype data (Lee et al., 2004).\n",
            "*Of the possible eight haplotypes, three common ones accounted for 98% of the estimated haplotypes in the Korean population.\n",
            "*Table 1 shows the haplotype frequency estimation in each population.\n",
            "*The potentially more inflammatory IL-1B-511T/-31C haplotype represented 53.5% of the Korean haplotypes, compared with 33.7% of the Caucasian haplotypes.\n",
            "*So far, in many previous association studies, the individual SNP approach, most frequently using IL-1B-511 and IL-1B-31, has been adopted.\n",
            "*To our knowledge, we reported first that the IL-1B-1464 polymorphism has allele-specific differences in nuclear protein binding and is associated with a clinical disease (Lee et al., 2004).\n",
            "*The biological implication of this polymorphism was supported by in vivo studies by Chen et al.\n",
            "*that showed that the IL-1B-1464 polymorphism has substantial allele-specific effects when both IL-1B-511 and IL-1B-31 were alleles T and C, respectively (Chen et al., 2006).\n",
            "*The more informative haplotype 1 (GTC), containing the IL-1B-1464 polymorphism, which shows the highest transcriptional activity, represents 9.3% and 6.0% of Korean and Caucasian haplotypes, respectively, whereas haplotype 3 (GCT), with the lowest activity, had a higher frequency in Caucasians (64.8%) when compared with Koreans (44.2%) (Table 1).\n",
            "*The difference in IL-1B promoter haplotype frequency between the Korean and Caucasian populations was statistically significant (=20.6, p=0.000), and the allele frequencies of the IL-1B-1464 polymorphism (rs#1143623) were also significantly different between the two populations (IL-1B-1464 G allele frequencies for Korean and Hapmap European=0.548 and 0.672, respectively) (=6.38, p=0.01).\n",
            "*It has been suggested that genes that are involved in immune function may be under selective pressure in direct interaction with the environment (Sawyer et al., 2004; Kim et al., 2005).\n",
            "*The genes that influence a phenotypic variation between populations are expected to show high Fst values.\n",
            "*Compared with the Fst value for the Caucasian-vs-Asian comparison, the Fst values for the African-vs-Asian or -Caucasian comparisons were remarkably high (Fig.\n",
            "*1).\n",
            "*Previously, we reported that the IL-1B-1464 polymorphism contributes to the development of intestinal-type gastric cancer among Koreans (Lee et al., 2004).\n",
            "*As a curious finding in our report, the editor pointed out that carriers of IL-1B-1464 G tend to have a decreased risk of diffuse-type of gastric cancer, which is the opposite of intestinal-type gastric cancer, although both intestinal and diffuse types of gastric cancer are related to Helicobacter pylori-induced gastritis (Furuta et al., 2004).\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "amChPq8xNN73",
        "colab_type": "code",
        "outputId": "36acdf1e-c969-4853-98ce-fe5330c1232d",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        }
      },
      "source": [
        "for sent in sents[300:400]:\n",
        "  print(\"*\"+sent)"
      ],
      "execution_count": 0,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "*Our results showed that most IL-1B-1464 C alleles are linked to the IL-1B-511T/-31C haplotype (Table 1).\n",
            "*Considering the level of promoter activity of haplotype 2 (CTC), we cannot exclude the possible association between this haplotype and the risk of diffuse-type gastric cancer, especially depending on interactions with other regulatory factors (Lee et al., 2007).\n",
            "*Association studies that use individual SNPs appear to be insufficient, and the understanding of functional haplotype structure of populations could provide potential explanations for IL-1B-related controversies and ethnic-specific associations.\n",
            "*Therefore, we believe that these Korean haplotype data will be useful for future association studies between IL-1B SNPs and disease risk.nted domains, including the human imprinted gene cluster that contains IGF2, H19, KCNQ1, ASCL2, and CDKN1C (Rapkins et al., 2006).\n",
            "*If, as has been suggested, imprinted genes are intimately connected with the acquisition of parental resources, we would not anticipate the existence of such genes in chicken, which leave their offspring to their own heritance after conception.\n",
            "*Phylogenetic analyses expose that the relationship between human and mouse is closer than that between human, mouse, and chicken.\n",
            "*Similarly, the relationship between zebrafish and chicken is quite distant (Shah et al., 2004).\n",
            "*Nonetheless, we assumed that chicken have imprinted genes due to the existence of common ancestral genomic regions that have evolved on a similar basis in each of the aforementioned species.\n",
            "*The purpose of this study was to identify candidate imprinted genes in chicken based on an analysis of orthologous genes in human, mouse, zebrafish, and chicken using the HomoloGene database.ols for the clinical oncology to determine the prognosis of patients (Lossos et al., 2004; Pomeroy et al., 2002), the molecular diagnosis (Golub et al., 1999) as well as the responsiveness to therapeutics (Snyder and Morgan, 2004).\n",
            "*There have been many reports on the molecular pattern analysis using microarray to understand the chemo- and radio-resistance in cervical cancer (Achary et al., 2000; Tewari et al., 2005; Wong et al., 2006), rectal cancer (Kim et al., 2007) and esophageal cancer (Fukuda et al., 2004).\n",
            "*Most of the studies are to identify differentially expressed genes in patients with different clinical outcomes, which can be applied to the evaluation of prognosis more accurately.\n",
            "*Although the conventional parameters like tumor stage and grade can be used to decide optimal cancer therapy, molecular markers would provide valuable information to make clinical decisions (Klopp and Eifel, 2006).\n",
            "*Genome-wide analysis on gene expression can predict the clinical consequences more accurately.\n",
            "*In addition, the information from gene expression profiling can facilitate the development of biological target for therapeutics by identifying pathways and determining steps contributing to the phenotype.\n",
            "*In this study, we examined the expression profiles of two lung cancer cell lines, which showed differential re- 1995).\n",
            "*In the inactive form, the pseudosubstrate domain is bound to the catalytic domain of PKC (Orr et al, 1994).\n",
            "*Upon stimulation, PKC translocates to the plasma membrane where the C1 and C2 domains interact with DAG and phosphatidylserine, respectively.\n",
            "*This interaction causes the pseudosubstrate domain to dissociate from the catalytic domain, which results in activation of PKC.\n",
            "*Inactive PKC is not freely distributed throughout the cytoplasm but appears to be localized to specific sites within the cell.\n",
            "*Association of PKC with scaffolding proteins such as AKAP79 (A Kinase-Anchoring Protein 79) (Klauck et al, 1996) and Gravin (Nauert et al, 1996) facilitates localization.\n",
            "*Streptomycetes are ubiquitous soil bacteria, and they play a key role in the global carbon cycle by degrading the insoluble remains of other organisms.\n",
            "*More clues to the development of the PKC super family come from the study of the bacterium Streptomyces coelicolor.\n",
            "*S. coelicolor has a large collection of enzymes and can metabolize many diverse nutrients.\n",
            "*This extremely simple organism contains approximately 8,667,507bp, yet has complex life cycle exhibiting mycelial growth and spore formation (Bentley, 2002) and notable for production of pharmaceutically useful anti-tumor compounds.\n",
            "*Of the predicted genes, an unprecedented proportion carries out regulatory functions in the cell (Winstead, 2002).\n",
            "*More than twelve percent of the genome is involved in facilitating biological processes, such as the bacterium's s reduce implementation time and increase the likelihood of eliminating bugs and localizing code modifications when a change in implementation is required.\n",
            "*In the initial version of the interface, all of the classes got tangled with each other and corrupted the concept of object-oriented programming.\n",
            "*However, they have been completely redesigned, as shown in Table 1.\n",
            "*This table summarizes the recent modifications of our system, and the interfaces for each class are documented, similar to Fig.\n",
            "*2.\n",
            "*The refactored version is now composed of 3851 lines, compared with the initial version, which was composed of 2765 lines of code.\n",
            "*By importing the five packages, an exemplary software system called J3dPSV 1.0, shown in Fig.\n",
            "*3, has been developed for viewing 3D structures of proteins from the Protein Data Bank for demonstrational purposes.\n",
            "*J3dPSV supports visualization of proteins for educational purposes by simulating simple molecular graphics.\n",
            "*In addition, J3dPSV interactively displays a molecule on the screen in a variety of color schemes, molecular representations, and animation features.\n",
            "*The molecular model can be changed by selecting the list (cartoon tubes, backbone, protein, cylinder, or line) in  useful suggestions for genotype information.\n",
            "*Compared to the current genotyping tools, GTVseq has several unique and useful features in the following aspects:  * GTVseq uses two different scoring schemes and the results are reported separately.\n",
            "*One of the scoring schemes is similar to that of NCBI, while the other is particularly useful for viral sequences with new or complicated genotypes (vide infra).\n",
            "** GTVseq offers an easy and interactive web-based user interface, with intuitive reports for genotyping results.\n",
            "** GTVseq can be used for genotyping many important viruses such as HIV-1, HIV-2, HBV, HCV, HTLV-1, HTLV-2, poliovirus, enterovirus, flavivirus, Hantavirus, and rotavirus, thus permitting the most comprehensive genotyping of viral genomes to date.Methods.\n",
            "*For genotyping of viral genome sequences, we need to establish 'reference sequences' for each genotype.\n",
            "*We have downloaded the reference sequence database collections from NCBI (http://www.ncbi.nlm.nih.gov/projects/ genotyping), for HIV-1, HIV-2, HBV, HCV, HTLV-1, HTLV-2, and poliovirus.\n",
            "*For HIV-1 reference sequences, GTVseq also provides several different collections of reference databases such as HIV-1 (2004) & CRF, HIV-1 (2005), HIV-1 (2005) & CRF.\n",
            "*For enterovirus, flavivirus, Hantavirus, and rotavirus, the reference sequences were combination of databases and interactive web pages for manipulating and displaying annotations on genomes.\n",
            "*In other words, GBrowse is a web-based application tool that is developed for navigating and visualizing the genomic features and annotations interactively for users.\n",
            "*Through it, users can view a certain region of the desired genomes and search for genetic biomarkers.\n",
            "*They may conduct a full-text search for most features of the genomes.\n",
            "*They also can download SNP assay, genotype, and allele frequency information and generate customized sets of tag-SNPs for their association studies (Thorisson et al., 2005).\n",
            "*GBrowse utilizes a web-based display that can be used to show arbitrary features of a nucleotide or protein sequence and can accommodate genome-scale sequences that are megabases in length.\n",
            "*The GBrowse system consists of various kinds of software modules and systems, such as web servers, database systems, and Perl libraries.\n",
            "*At present, many biological websites that provide genomic variants or portal services have been developed using GBrowse, including the following: the UCSC Genome Browser (Kuhn et al., 2007), the International HapMap Project (Thorisson et al., 2005), PlasmoDB (The is database is free for non-commercial purposes.\n",
            "*The KRDD is visualized using a web-based graphical view, and anonymous users can query and browse the data using the search function.\n",
            "*The KRDD homepage is shown in Fig.\n",
            "*1, and the stored data are visualized using a web-based graphical view.\n",
            "*It has four major menus of web pages: (i) a Blast Search of a mutant line; Blast from rice Ds-tagging mutant lines; (ii) a primer design tool to identify genotypes of Ds insertion lines; (iii) a Phenotype menu for Ds lines, searching by gene name and phenotype characteristics among specific Ds lines; and (iv) a Management menu for Ds lines.\n",
            "*The Blast Search is searchable by selecting specific databases, consisting of DS Sequence, Indica Core, Japonica Core, Indica EST, Japonica EST, Indica Genome, Japonica Genome, Indica GSS, and Japonica GSS in Oryza sativa.\n",
            "*The KRDD uses several reference databases to facilitate a comprehensive analysis of the genome sequence.\n",
            "*These include the Entrez nucleotide database of the National Center for Biotechnology entative biological pathway database, now provides the KEGG Metabolism Atlas (Okuda et al., 2008) by manually combining about 120 existing metabolic pathway maps, as shown in Fig.\n",
            "*1.\n",
            "*However, the static approach to representing metabolic pathway diagrams offers no flexibility.\n",
            "*On the other hand, our initial attempts to visualize all information automatically in a single atlas map resulted in a confusing diagram that was difficult to interpret, as shown in Fig.\n",
            "*2.\n",
            "*It should be noted that Fig.\n",
            "*2 differs in many aspects from Fig.\n",
            "*1 or conventional drawings in biochemistry textbooks.\n",
            "*For this reason, we designed a new metabolic atlas viewing tool called J2dpathway, which has node-abstracting features.\n",
            "*When J2dpathway is initially executed, a window frame appears, as shown in Fig.\n",
            "*3.\n",
            "*The screen consists of views and editors.\n",
            "*The tool-bar menu at the top lists various tool icons, including zoom-in, zoom-out, cliques, highly connected nodes, and obtaining cycles.\n",
            "*The Map Repository View on the left side lists a preinstalled data source that has many example pathways to explore, which are arranged in a tree view of the components of of the HIF1ODD domain with multiple partner proteins, such as ARD1 (Jeong et al., 2002), prolyl hydroxylase (PHD) (Schofield and Ratcliffe, 2004), and p53 (Fels and Koumenis, 2005; Sanchez-Puig et al., 2005), also have been reported.\n",
            "*However, the molecular basis for the multiple binding specificity of the HIF1ODD domain has not been understood yet.\n",
            "*The detailed characterization of the correlation between the binding sequence motifs in the ODD domain and its binding to multiple target proteins is necessary for understanding the versatile function of the HIF1ODD domain.\n",
            "*Two functionally independent sequence motifs, the N-terminal and C-terminal ODD (NODD and CODD), in the HIF1ODD domain were shown to bind to the DNA-binding domain (DBD) of p53 (Hansson et al., 2002).\n",
            "*The crystal structure of the CODD motif in complex with pVHL was determined to ncluding 1p36, 5q31, and 21q22, by whole-genome linkage analysis (genome-wide association studies), and many polymorphisms also have been identified at these loci (Suzuki et al., 2003; Tokuhiro et al., 2003).\n",
            "*Human leucine-rich alpha-2-glycoprotein 1 (LRG1) was first identified as a trace protein in human serum (Haupt & Baudner, 1977).\n",
            "*The LRG1 gene is located on chromosome 19p13.3, and the primary sequence of LRG includes repeated leucine residues and also has putative membrane-binding domains.\n",
            "*Serum LRG1 is the first extracellular ligand for cytochrome c (Cyt c).\n",
            "*Cyt c is a ubiquitous, heme-containing protein that normally resides in the space between the inner and outer mitochondrial membranes (Newmeyer et al., 2003).\n",
            "*Extracellular Cyt c may play a role in inflammation, as it has been reported to cause arthritis when it is injected into mice.\n",
            "*Its levels in RA patients' sera are significantly lower than those of healthy controls (Pullerits et al., 2005).\n",
            "*At least eight repeating 24-amino acid segments that have a notable consensus sequence were identified in a large family of LRG proteins.\n",
            "*The function of LRG has not been elucidated, although the functions of many of the other members of the LRR (leucine-rich repeat)-containing superfamily are known (Kobe & Deisenhofer, 1994; Buchanan & Gay, 1996).\n",
            "*Plasma LRG expression levels are lower in liver cancer patients who are treated with radiofrequency ablation (Kawakami eutic agent for cancer is the in vivo specificity of cancer cell regression.\n",
            "*For such a specificity, target RNA-independent and nonspecific transgene induction by the group I intron should be avoided.\n",
            "*In other words, mis-spliced products should not be generated by the group I intron.\n",
            "*In this study, in order to evaluate the therapeutic feasibility of the hTERT-specific group I intron, we assessed the target RNA specificity of the trans-splicing phenomenon by the intron in mice that have been intraperitoneally xenografted with human cancer cells.Construction of Adenoviral Vector for hTERT-specific Group I Intron .\n",
            "*The expression vector that encodes for the hTERT-specific trans-splicing group I intron was constructed as previously described (Kwon et al., 2005; Song et al., 2006).\n",
            "*In brief, the Rib21AS group I intron, which recognizes uridine at position 21 (U21) of hTERT RNA, was generated to harbor an extended internal guide sequence, which includes an internal guide sequence (IGS, 5'-GGCAGG-3'), an extension of the P1 helix, an additional 6-nt-long P10 helix, and a 325-nt-long antisense sequence that is complementary to the downstream region (30 to 354 residues) of the targeted U21 of hTERT RNA.\n",
            "*In addition, cDNA, as a 3' exon that encodes for the lacZ gene, was inserted downstream of the modified group I intron expression construct (Fig.\n",
            "*1A).\n",
            "*using PHRAP (http:// www.phrap.org/), it does not ensure correct assembly because the quality scores that are generated from 454 data are not compatible with those from Sanger reads.\n",
            "*Further, PHRAP has problems with handling massive reads (usually hundreds of thousands from an SFF file).\n",
            "*A recent report has demonstrated that GS assembler programs (gsAssembler for de novo assembly and gsMapping for reference-guided assembly; http://www.\n",
            "*454.com/enabling-technology/the-software.asp) that are supplied by Roche Applied Science are ideal for correct assembly of 454 data that are short and inherently error-rich (Chaisson and Pevzner, 2008).\n",
            "*Recent versions (1.1.02.15 and later) of GS assembler programs support mixed assembly with Sanger-type reads, but their performance is not well known at present.\n",
            "*Moreover, because pre-existing assembly software such as PHRAP and CelAsm (Huson et al., 2001) do not directly support data that are produced by 454 machines, 454-derived contigs (GS contigs) should be used as if they were individual reads or be shredded to generate many overlapping 'pseudoreads' (Goldberg et al., 2006).\n",
            "*Pseudoreads, made from GS contigs to emulate the read size of standard Sanger data (ca.\n",
            "*600 bp), are virtual reads whose stepping between consecutive dertaken as a collaboration between Korean funding agencies (Ministry of Education, Science and Technology and Korean National Institute of Health), experimental academia (Ulsan Medical Institute, SungKyunKwan Medical Institute, and Korea Advanced Institute of Science and Technology), and corporations (DNA Link, SNP-Genetics, and Samsung Advanced Institute of Technology) (Yoo et al., 2006; Lee et al., 2008).\n",
            "*Resulting from the project, a Korean SNP and haplotype database system was developed to help those researchers who study high-frequency, complex Korean diseases and changes in ethnic global migratory variants.\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "P5W3ZjFRLaJ3",
        "colab_type": "text"
      },
      "source": [
        "1. seperator를 정의하여 \\n, \\t, : 등으로 문장 구분 필요\n",
        "2. Fig.2 등의 약어가 .을 기준으로 문장이 구분되기 때문에 이와 같은 약어에 대한 예외처리(사전) 필요\n",
        "3. Tuzun et al.(2005)등의 논문이 .을 기준으로 문장이 구분되기 때문에 논문에 대한 예외처리(사전) 필요\n",
        "4. 순서 번호를 매기는 '1.~'라는 문장에 대해 .을 기준으로 문장이 구분되기 때문에 삭제 혹은 예외처리(사전) 필요\n",
        "5. url에 대해 www.~ 에서 .을 기준으로 문장이 구분되기 때문에 url에 대한 예외처리(사전) 필요\n",
        "6. 데이터에 대한 'ca.600 bp'등의 단어에 대해 .으로 구분이 되기 때문에 데이터 단위에 대한 예외처리(사전) 필요"
      ]
    }
  ]
}